<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Pritam Kumar Panda">

<title>scrnaseq-analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="scrnaseq-analysis_files/libs/clipboard/clipboard.min.js"></script>
<script src="scrnaseq-analysis_files/libs/quarto-html/quarto.js"></script>
<script src="scrnaseq-analysis_files/libs/quarto-html/popper.min.js"></script>
<script src="scrnaseq-analysis_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="scrnaseq-analysis_files/libs/quarto-html/anchor.min.js"></script>
<link href="scrnaseq-analysis_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="scrnaseq-analysis_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="scrnaseq-analysis_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="scrnaseq-analysis_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="scrnaseq-analysis_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="scrnaseq-analysis_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="scrnaseq-analysis_files/libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="scrnaseq-analysis_files/libs/typedarray-0.1/typedarray.min.js"></script>
<script src="scrnaseq-analysis_files/libs/jquery-3.5.1/jquery.min.js"></script>
<link href="scrnaseq-analysis_files/libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="scrnaseq-analysis_files/libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="scrnaseq-analysis_files/libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="scrnaseq-analysis_files/libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">scrnaseq-analysis</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Pritam Kumar Panda </p>
          </div>
  </div>
    
  
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">status</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>The following package(s) are in an inconsistent state:

 package installed recorded used
 remotes y         y        n   

See ?renv::status() for advice on resolving these issues.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">snapshot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>The following package(s) will be updated in the lockfile:

# RSPM -----------------------------------------------------------------------
- remotes   [2.4.2.1 -&gt; *]

- Lockfile written to "~/Desktop/scRNAseq-analysis/renv.lock".</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">activate</span>()</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">timeout =</span> <span class="fu">max</span>(<span class="dv">1000</span>, <span class="fu">getOption</span>(<span class="st">"timeout"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("Seurat")</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("SeuratObject")</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("ggplot2")</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("patchwork")</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("tidyverse")</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("dplyr")</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("ggsci")</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("viridis")</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("pheatmap")</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("devtools")</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co">#install.packages("hdf5r")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#if (!require("BiocManager", quietly = TRUE))</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co">#    install.packages("BiocManager")</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install(version = "3.18")</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("BiocParallel")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co">#if old packages pops up to update a/s/n then just simply do</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co">#install() and press enter.</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("DESeq2")</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("scran")</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("celldex")</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("glmGamPoi")</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="co">#remotes::install_github('chris-mcginnis-ucsf/DoubletFinder')</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="co">#BiocManager::install("SingleR")</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SummarizedExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: MatrixGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: matrixStats</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'MatrixGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
    colWeightedMeans, colWeightedMedians, colWeightedSds,
    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
    rowWeightedSds, rowWeightedVars</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomicRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    findMatches</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biobase</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biobase'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:MatrixGenerics':

    rowMedians</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:matrixStats':

    anyMissing, rowMedians</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scran)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SingleCellExperiment</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: scuttle</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocParallel)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a><span class="fu">register</span>(<span class="fu">SnowParam</span>(<span class="at">workers =</span> <span class="dv">20</span>, <span class="at">progressbar =</span> <span class="cn">TRUE</span>))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: SeuratObject</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: sp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'sp'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    %over%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built under R 4.3.0 but the current version is
4.3.2; it is recomended that you reinstall 'SeuratObject' as the ABI
for R may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>'SeuratObject' was built with package 'Matrix' 1.6.4 but the current
version is 1.6.5; it is recomended that you reinstall 'SeuratObject' as
the ABI for 'Matrix' may have changed</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'SeuratObject'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomicRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:GenomeInfoDb':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:IRanges':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:S4Vectors':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:BiocGenerics':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    intersect</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Seurat'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:SummarizedExperiment':

    Assays</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ lubridate 1.9.3     ✔ tibble    3.2.1
✔ purrr     1.0.2     ✔ tidyr     1.3.1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ lubridate::%within%() masks IRanges::%within%()
✖ dplyr::collapse()     masks IRanges::collapse()
✖ dplyr::combine()      masks Biobase::combine(), BiocGenerics::combine()
✖ dplyr::count()        masks matrixStats::count()
✖ dplyr::desc()         masks IRanges::desc()
✖ tidyr::expand()       masks S4Vectors::expand()
✖ dplyr::filter()       masks stats::filter()
✖ dplyr::first()        masks S4Vectors::first()
✖ dplyr::lag()          masks stats::lag()
✖ ggplot2::Position()   masks BiocGenerics::Position(), base::Position()
✖ purrr::reduce()       masks GenomicRanges::reduce(), IRanges::reduce()
✖ dplyr::rename()       masks S4Vectors::rename()
✖ lubridate::second()   masks S4Vectors::second()
✖ lubridate::second&lt;-() masks S4Vectors::second&lt;-()
✖ dplyr::slice()        masks IRanges::slice()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(hdf5r)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'hdf5r'

The following object is masked from 'package:purrr':

    flatten_df

The following object is masked from 'package:SummarizedExperiment':

    values

The following object is masked from 'package:GenomicRanges':

    values

The following object is masked from 'package:S4Vectors':

    values</code></pre>
</div>
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggsci)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(celldex)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'celldex'

The following objects are masked from 'package:SingleR':

    BlueprintEncodeData, DatabaseImmuneCellExpressionData,
    HumanPrimaryCellAtlasData, ImmGenData, MonacoImmuneData,
    MouseRNAseqData, NovershternHematopoieticData</code></pre>
</div>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SingleCellExperiment)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmGamPoi)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'glmGamPoi'

The following object is masked from 'package:dplyr':

    vars

The following object is masked from 'package:ggplot2':

    vars</code></pre>
</div>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DoubletFinder)</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'

The following object is masked from 'package:lubridate':

    stamp

The following object is masked from 'package:patchwork':

    align_plots</code></pre>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: viridisLite</code></pre>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>To start the analysis, let’s read in the <code>10X</code>-corrected matrices. If starting from typical <code>Cell Ranger</code> output, it’s possible to choose if you want to use Ensemble ID or gene symbol for the count matrix. This is done using <code>gene.column</code> option; default is ‘2,’ which is gene symbol.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co">#1. check if the file exist</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a><span class="fu">file.exists</span>(<span class="st">"KICH_GSE159115_expression.h5"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">file.exists</span>(<span class="st">"KICH_GSE159115_CellMetainfo_table.tsv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>adj_matrix <span class="ot">&lt;-</span> Seurat<span class="sc">::</span><span class="fu">Read10X_h5</span>(<span class="at">filename =</span> <span class="st">"KICH_GSE159115_expression.h5"</span>, <span class="at">use.names =</span> T)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a><span class="co">#metadata   &lt;- as.data.frame(read_tsv("KICH_GSE159115_CellMetainfo_table.tsv",show_col_types = FALSE))</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>After this, we will make a <code>Seurat</code> object. Seurat object summary shows us that 1) number of cells (“samples”) approximately matches the description of each dataset; 2) genes (features) in the reference.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> adj_matrix ,<span class="at">project =</span> <span class="st">"TISCH2"</span>,<span class="at">min.cells =</span> <span class="dv">3</span>, <span class="at">min.features =</span> <span class="dv">200</span> )</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a>seurat_obj</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>An object of class Seurat 
16041 features across 2850 samples within 1 assay 
Active assay: RNA (16041 features, 0 variable features)
 1 layer present: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'Seurat' [package "SeuratObject"] with 13 slots
  ..@ assays      :List of 1
  .. ..$ RNA:Formal class 'Assay5' [package "SeuratObject"] with 8 slots
  .. .. .. ..@ layers    :List of 1
  .. .. .. .. ..$ counts:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. .. ..@ i       : int [1:8221142] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. .. ..@ p       : int [1:2851] 0 985 2102 3357 4410 5863 7197 8590 9957 11096 ...
  .. .. .. .. .. .. ..@ Dim     : int [1:2] 16041 2850
  .. .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. ..@ x       : num [1:8221142] 1.29 1.29 1.29 3.21 1.29 ...
  .. .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ cells     :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:2850, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:2850] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. .. .. ..$ dim     : int [1:2] 2850 1
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:2850] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. ..@ features  :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:16041, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. .. .. ..$ dim     : int [1:2] 16041 1
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. ..@ default   : int 1
  .. .. .. ..@ assay.orig: chr(0) 
  .. .. .. ..@ meta.data :'data.frame': 16041 obs. of  0 variables
  .. .. .. ..@ misc      :List of 1
  .. .. .. .. ..$ calcN: logi TRUE
  .. .. .. ..@ key       : chr "rna_"
  ..@ meta.data   :'data.frame':    2850 obs. of  3 variables:
  .. ..$ orig.ident  : Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..$ nCount_RNA  : num [1:2850] 1703 1776 2010 1748 2248 ...
  .. ..$ nFeature_RNA: int [1:2850] 985 1117 1255 1053 1453 1334 1393 1367 1139 1398 ...
  ..@ active.assay: chr "RNA"
  ..@ active.ident: Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..- attr(*, "names")= chr [1:2850] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  ..@ graphs      : list()
  ..@ neighbors   : list()
  ..@ reductions  : list()
  ..@ images      : list()
  ..@ project.name: chr "TISCH2"
  ..@ misc        : list()
  ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1
  .. ..$ : int [1:3] 5 0 1
  ..@ commands    : list()
  ..@ tools       : list()</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Assuming you have a Seurat object named seurat_object</span></span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb75-3"><a href="#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the Seurat object as an RDS file</span></span>
<span id="cb75-4"><a href="#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seurat_obj, <span class="at">file =</span> <span class="st">"seurat_object.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s erase <code>adj.matrix</code> from memory to save RAM, and look at the <code>Seurat</code> object a bit closer. <code>str</code> commant allows us to see all fields of the class:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>adj.matrix <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'Seurat' [package "SeuratObject"] with 13 slots
  ..@ assays      :List of 1
  .. ..$ RNA:Formal class 'Assay5' [package "SeuratObject"] with 8 slots
  .. .. .. ..@ layers    :List of 1
  .. .. .. .. ..$ counts:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. .. ..@ i       : int [1:8221142] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. .. ..@ p       : int [1:2851] 0 985 2102 3357 4410 5863 7197 8590 9957 11096 ...
  .. .. .. .. .. .. ..@ Dim     : int [1:2] 16041 2850
  .. .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. ..@ x       : num [1:8221142] 1.29 1.29 1.29 3.21 1.29 ...
  .. .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ cells     :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:2850, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:2850] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. .. .. ..$ dim     : int [1:2] 2850 1
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:2850] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. ..@ features  :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:16041, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. .. .. ..$ dim     : int [1:2] 16041 1
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. ..@ default   : int 1
  .. .. .. ..@ assay.orig: chr(0) 
  .. .. .. ..@ meta.data :'data.frame': 16041 obs. of  0 variables
  .. .. .. ..@ misc      :List of 1
  .. .. .. .. ..$ calcN: logi TRUE
  .. .. .. ..@ key       : chr "rna_"
  ..@ meta.data   :'data.frame':    2850 obs. of  3 variables:
  .. ..$ orig.ident  : Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..$ nCount_RNA  : num [1:2850] 1703 1776 2010 1748 2248 ...
  .. ..$ nFeature_RNA: int [1:2850] 985 1117 1255 1053 1453 1334 1393 1367 1139 1398 ...
  ..@ active.assay: chr "RNA"
  ..@ active.ident: Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..- attr(*, "names")= chr [1:2850] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  ..@ graphs      : list()
  ..@ neighbors   : list()
  ..@ reductions  : list()
  ..@ images      : list()
  ..@ project.name: chr "TISCH2"
  ..@ misc        : list()
  ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1
  .. ..$ : int [1:3] 5 0 1
  ..@ commands    : list()
  ..@ tools       : list()</code></pre>
</div>
</div>
<p>Meta.data is the most important field for next steps. It can be acessed using both @ and [[]] operators. Right now it has 3 fields per celL: dataset ID, number of UMI reads detected per cell (nCount_RNA), and the number of expressed (detected) genes per same cell (nFeature_RNA).</p>
<pre class="{#{r now add metadata to the seurat object}"><code>
#row.names(meta) &lt;- colnames(x=seurat_obj)
#or
##row.names(meta) &lt;- meta$Cell
##pacreas_meta_1 &lt;- AddMetaData(pan_seo, metadata = meta)
##head(pacreas_meta_1)

#head(colnames(x= seurat_obj))
#meta &lt;- AddMetaData(seurat_obj, meta, row.names("Celltype"))
#head(meta)</code></pre>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="co">#meta &lt;- seurat_obj@meta.data</span></span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dim(meta)</span></span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a><span class="co">#head(meta)</span></span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="co">#colnames(meta)</span></span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(meta$nCount_RNA)</span></span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a><span class="co">#summary(meta$nFeature_RNA)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s add several more values useful in diagnostics of cell quality. Michochondrial genes are useful indicators of cell state.</p>
<p>For <strong>mouse</strong> datasets, change pattern to “Mt-,” or explicitly list gene IDs with the <em>features = …</em> option.</p>
<p>Similarly, we can define ribosomal proteins (their names begin with <strong>RPS</strong> or <strong>RPL</strong>), which often take substantial fraction of reads:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>seurat_obj[[<span class="st">"percent.mt"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(seurat_obj, <span class="at">pattern =</span> <span class="st">"^MT"</span>)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>seurat_obj[[<span class="st">"percent.rb"</span>]] <span class="ot">&lt;-</span> <span class="fu">PercentageFeatureSet</span>(seurat_obj, <span class="at">pattern =</span> <span class="st">"^RP[SL]"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>A modeling framework for the normalization and variance stabilization of molecular count data from scRNA-seq experiments. This procedure omits the need for heuristic steps including pseudocount addition or log-transformation and improves common downstream analytical tasks such as variable gene selection, dimensional reduction, and differential expression <code>sctransform</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">SCTransform</span>(seurat_obj, <span class="at">vars.to.regress =</span> <span class="st">"percent.mt"</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Pearson correlation between the two features is displayed above the plot</p>
<p>High degree: If the coefficient value lies <strong>between ± 0.50 and ± 1</strong>, then it is said to be a strong correlation. Moderate degree: If the value lies between ± 0.30 and ± 0.49, then it is said to be a medium correlation.</p>
<p>The Pearson correlation measures the strength of the linear relationship between two variables. It has a value between -1 to 1, with a value of <strong>-1 meaning a total negative linear correlation</strong>, 0 being no correlation, and + 1 meaning a total positive correlation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Plots</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nFeature_RNA"</span>,<span class="st">"nCount_RNA"</span>,<span class="st">"percent.mt"</span>,<span class="st">"percent.rb"</span>),<span class="at">ncol =</span> <span class="dv">4</span>,<span class="at">pt.size =</span> <span class="fl">0.1</span>) <span class="sc">&amp;</span> </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeatureScatter</span>(seurat_obj, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"percent.mt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-12-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeatureScatter</span>(seurat_obj, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"nFeature_RNA"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-12-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeatureScatter</span>(seurat_obj, <span class="at">feature1 =</span> <span class="st">"nCount_RNA"</span>, <span class="at">feature2 =</span> <span class="st">"percent.rb"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-12-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeatureScatter</span>(seurat_obj, <span class="at">feature1 =</span> <span class="st">"percent.rb"</span>, <span class="at">feature2 =</span> <span class="st">"percent.mt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-12-5.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co">#*__________Quality Control And Filtering_________</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">subset</span>(seurat_obj, <span class="at">subset =</span> nFeature_RNA <span class="sc">&gt;</span> <span class="dv">200</span> <span class="sc">&amp;</span> nFeature_RNA <span class="sc">&lt;</span> <span class="dv">2500</span> <span class="sc">&amp;</span> percent.mt <span class="sc">&lt;</span> <span class="dv">5</span>)</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'Seurat' [package "SeuratObject"] with 13 slots
  ..@ assays      :List of 2
  .. ..$ RNA:Formal class 'Assay5' [package "SeuratObject"] with 8 slots
  .. .. .. ..@ layers    :List of 1
  .. .. .. .. ..$ counts:Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. .. ..@ i       : int [1:1501544] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. .. ..@ p       : int [1:810] 0 985 2102 3357 4410 5863 7197 8590 9957 11096 ...
  .. .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. ..@ x       : num [1:1501544] 1.29 1.29 1.29 3.21 1.29 ...
  .. .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ cells     :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:809, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. .. .. ..$ dim     : int [1:2] 809 1
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. ..@ features  :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:16041, 1] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. .. .. ..$ dim     : int [1:2] 16041 1
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr "counts"
  .. .. .. ..@ default   : int 1
  .. .. .. ..@ assay.orig: chr(0) 
  .. .. .. ..@ meta.data :'data.frame': 16041 obs. of  0 variables
  .. .. .. ..@ misc      :List of 1
  .. .. .. .. ..$ calcN: logi TRUE
  .. .. .. ..@ key       : chr "rna_"
  .. ..$ SCT:Formal class 'SCTAssay' [package "Seurat"] with 9 slots
  .. .. .. ..@ SCTModel.list:List of 1
  .. .. .. .. ..$ counts:Formal class 'SCTModel' [package "Seurat"] with 7 slots
  .. .. .. .. .. .. ..@ feature.attributes:'data.frame':    16041 obs. of  12 variables:
  .. .. .. .. .. .. .. ..$ detection_rate       : num [1:16041] 0.25404 0.17965 0.00632 0.03544 0.05649 ...
  .. .. .. .. .. .. .. ..$ gmean                : num [1:16041] 0.1263 0.08541 0.00269 0.01573 0.0251 ...
  .. .. .. .. .. .. .. ..$ variance             : num [1:16041] 0.08595 0.06204 0.00182 0.01234 0.01952 ...
  .. .. .. .. .. .. .. ..$ residual_mean        : num [1:16041] 3.19e-03 -5.03e-03 -3.55e-19 1.31e-04 2.64e-03 ...
  .. .. .. .. .. .. .. ..$ residual_variance    : num [1:16041] 0.591 0.572 0.09 0.585 0.663 ...
  .. .. .. .. .. .. .. ..$ theta                : num [1:16041] Inf Inf Inf Inf Inf ...
  .. .. .. .. .. .. .. ..$ (Intercept)          : num [1:16041] -9.76 -10.14 -13.6 -11.82 -11.35 ...
  .. .. .. .. .. .. .. ..$ log_umi              : num [1:16041] 2.3 2.3 2.3 2.3 2.3 ...
  .. .. .. .. .. .. .. ..$ genes_log_gmean_step1: logi [1:16041] FALSE FALSE FALSE FALSE FALSE FALSE ...
  .. .. .. .. .. .. .. ..$ step1_theta          : num [1:16041] NA NA NA NA NA ...
  .. .. .. .. .. .. .. ..$ step1_(Intercept)    : num [1:16041] NA NA NA NA NA ...
  .. .. .. .. .. .. .. ..$ step1_log_umi        : num [1:16041] NA NA NA NA NA ...
  .. .. .. .. .. .. ..@ cell.attributes   :'data.frame':    809 obs. of  3 variables:
  .. .. .. .. .. .. .. ..$ umi        : num [1:809] 1703 1776 2010 1748 2248 ...
  .. .. .. .. .. .. .. ..$ log_umi    : num [1:809] 3.23 3.25 3.3 3.24 3.35 ...
  .. .. .. .. .. .. .. ..$ cells_step1: logi [1:809] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..@ clips             :List of 2
  .. .. .. .. .. .. .. ..$ vst: num [1:2] -53.4 53.4
  .. .. .. .. .. .. .. ..$ sct: num [1:2] -9.75 9.75
  .. .. .. .. .. .. ..@ umi.assay         : chr "RNA"
  .. .. .. .. .. .. ..@ model             : chr "y ~ log_umi"
  .. .. .. .. .. .. ..@ arguments         :List of 33
  .. .. .. .. .. .. .. ..$ glmGamPoi_check          : logi TRUE
  .. .. .. .. .. .. .. ..$ latent_var               : chr "log_umi"
  .. .. .. .. .. .. .. ..$ batch_var                : NULL
  .. .. .. .. .. .. .. ..$ latent_var_nonreg        : NULL
  .. .. .. .. .. .. .. ..$ n_genes                  : num 2000
  .. .. .. .. .. .. .. ..$ n_cells                  : num 2850
  .. .. .. .. .. .. .. ..$ method                   : chr "glmGamPoi_offset"
  .. .. .. .. .. .. .. ..$ do_regularize            : logi TRUE
  .. .. .. .. .. .. .. ..$ theta_regularization     : chr "od_factor"
  .. .. .. .. .. .. .. ..$ res_clip_range           : num [1:2] -53.4 53.4
  .. .. .. .. .. .. .. ..$ bin_size                 : num 500
  .. .. .. .. .. .. .. ..$ min_cells                : num 5
  .. .. .. .. .. .. .. ..$ residual_type            : chr "pearson"
  .. .. .. .. .. .. .. ..$ return_cell_attr         : logi TRUE
  .. .. .. .. .. .. .. ..$ return_gene_attr         : logi TRUE
  .. .. .. .. .. .. .. ..$ return_corrected_umi     : logi TRUE
  .. .. .. .. .. .. .. ..$ min_variance             : chr "umi_median"
  .. .. .. .. .. .. .. ..$ bw_adjust                : num 3
  .. .. .. .. .. .. .. ..$ gmean_eps                : num 1
  .. .. .. .. .. .. .. ..$ theta_estimation_fun     : chr "theta.ml"
  .. .. .. .. .. .. .. ..$ theta_given              : NULL
  .. .. .. .. .. .. .. ..$ exclude_poisson          : logi TRUE
  .. .. .. .. .. .. .. ..$ use_geometric_mean       : logi TRUE
  .. .. .. .. .. .. .. ..$ use_geometric_mean_offset: logi FALSE
  .. .. .. .. .. .. .. ..$ fix_intercept            : logi FALSE
  .. .. .. .. .. .. .. ..$ fix_slope                : logi FALSE
  .. .. .. .. .. .. .. ..$ scale_factor             : logi NA
  .. .. .. .. .. .. .. ..$ vst.flavor               : chr "v2"
  .. .. .. .. .. .. .. ..$ verbosity                : num 0
  .. .. .. .. .. .. .. ..$ verbose                  : NULL
  .. .. .. .. .. .. .. ..$ show_progress            : NULL
  .. .. .. .. .. .. .. ..$ set_min_var              : num 0.0202
  .. .. .. .. .. .. .. ..$ sct.clip.range           : num [1:2] -9.75 9.75
  .. .. .. .. .. .. ..@ median_umi        : num 2813
  .. .. .. ..@ counts       :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. ..@ i       : int [1:1497441] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. ..@ p       : int [1:810] 0 994 2114 3366 4427 5875 7209 8602 9967 11109 ...
  .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..@ x       : num [1:1497441] 2 2 2 5 2 2 2 2 6 2 ...
  .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ data         :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. ..@ i       : int [1:1497441] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. ..@ p       : int [1:810] 0 994 2114 3366 4427 5875 7209 8602 9967 11109 ...
  .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..@ x       : num [1:1497441] 1.1 1.1 1.1 1.79 1.1 ...
  .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ scale.data   : num [1:3000, 1:809] -0.141 -0.285 -0.143 -0.139 6.186 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:3000] "LINC00115" "AGRN" "ACAP3" "DVL1" ...
  .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. ..@ assay.orig   : chr "RNA"
  .. .. .. ..@ var.features : chr [1:3000] "WAS" "LAT" "FMNL1" "SERPINA1" ...
  .. .. .. ..@ meta.features:'data.frame':  16041 obs. of  0 variables
  .. .. .. ..@ misc         : Named list()
  .. .. .. ..@ key          : chr "sct_"
  ..@ meta.data   :'data.frame':    809 obs. of  7 variables:
  .. ..$ orig.ident  : Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..$ nCount_RNA  : num [1:809] 1703 1776 2010 1748 2248 ...
  .. ..$ nFeature_RNA: int [1:809] 985 1117 1255 1053 1453 1334 1393 1367 1139 1398 ...
  .. ..$ percent.mt  : num [1:809] 2.25 2.15 1.54 2.7 1.79 ...
  .. ..$ percent.rb  : num [1:809] 18.4 17.8 14.9 17.2 12.3 ...
  .. ..$ nCount_SCT  : num [1:809] 2540 2441 2358 2643 2313 ...
  .. ..$ nFeature_SCT: int [1:809] 994 1120 1252 1061 1448 1334 1393 1365 1142 1397 ...
  ..@ active.assay: chr "SCT"
  ..@ active.ident: Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..- attr(*, "names")= chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  ..@ graphs      : list()
  ..@ neighbors   : list()
  ..@ reductions  : list()
  ..@ images      : list()
  ..@ project.name: chr "TISCH2"
  ..@ misc        : list()
  ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1
  .. ..$ : int [1:3] 5 0 1
  ..@ commands    :List of 1
  .. ..$ SCTransform.RNA:Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "SCTransform.RNA"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:48"
  .. .. .. ..@ assay.used : chr "RNA"
  .. .. .. ..@ call.string: chr [1:2] "SCTransform(seurat_obj, vars.to.regress = \"percent.mt\", " "    verbose = FALSE)"
  .. .. .. ..@ params     :List of 15
  .. .. .. .. ..$ assay                  : chr "RNA"
  .. .. .. .. ..$ new.assay.name         : chr "SCT"
  .. .. .. .. ..$ do.correct.umi         : logi TRUE
  .. .. .. .. ..$ ncells                 : num 5000
  .. .. .. .. ..$ variable.features.n    : num 3000
  .. .. .. .. ..$ variable.features.rv.th: num 1.3
  .. .. .. .. ..$ vars.to.regress        : chr "percent.mt"
  .. .. .. .. ..$ do.scale               : logi FALSE
  .. .. .. .. ..$ do.center              : logi TRUE
  .. .. .. .. ..$ clip.range             : num [1:2] -9.75 9.75
  .. .. .. .. ..$ vst.flavor             : chr "v2"
  .. .. .. .. ..$ conserve.memory        : logi FALSE
  .. .. .. .. ..$ return.only.var.genes  : logi TRUE
  .. .. .. .. ..$ seed.use               : num 1448145
  .. .. .. .. ..$ verbose                : logi FALSE
  ..@ tools       : list()</code></pre>
</div>
</div>
<p>In order to do further analysis, we need to normalize the data to account for sequencing depth. Conventional way is to scale it to 10,000 (as if all cells have 10k UMIs overall), and log2-transform the obtained values. Normalized data are stored in <code>srat[['RNA']]@data</code> of the ‘RNA’ assay.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="co">#___________Normalization and Feature Selection____________</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seurat_obj, <span class="at">normalization.method =</span> <span class="st">"LogNormalize"</span>, <span class="at">scale.factor =</span> <span class="dv">10000</span>)</span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seurat_obj, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Identify top 10 highly variable genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify the 10 most highly variable genes</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(<span class="fu">VariableFeatures</span>(seurat_obj), <span class="dv">10</span>)</span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>top10</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "WAS"      "LAT"      "FMNL1"    "SERPINA1" "POU2F2"   "FERMT3"  
 [7] "SNHG12"   "CXCL16"   "LCP2"     "TAGAP"   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot variable features with and without labels</span></span>
<span id="cb92-2"><a href="#cb92-2" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(seurat_obj)</span>
<span id="cb92-3"><a href="#cb92-3" aria-hidden="true" tabindex="-1"></a>plot2 <span class="ot">&lt;-</span> <span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot1, <span class="at">points =</span> top10, <span class="at">repel =</span> <span class="cn">TRUE</span>,<span class="at">xnudge=</span><span class="dv">0</span>,<span class="at">ynudge=</span><span class="dv">0</span>)</span>
<span id="cb92-4"><a href="#cb92-4" aria-hidden="true" tabindex="-1"></a>plot1 <span class="sc">+</span> plot2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 8 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a>plot1 <span class="ot">&lt;-</span> <span class="fu">VariableFeaturePlot</span>(seurat_obj)</span>
<span id="cb94-2"><a href="#cb94-2" aria-hidden="true" tabindex="-1"></a><span class="fu">LabelPoints</span>(<span class="at">plot =</span> plot1, <span class="at">points =</span> top10, <span class="at">repel =</span> <span class="cn">TRUE</span>, <span class="at">xnudge =</span> <span class="dv">0</span>, <span class="at">ynudge =</span> <span class="dv">0</span>, <span class="at">max.overlaps =</span> <span class="dv">20</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p><code>ScaleData</code> converts normalized gene expression to Z-score (values centered at 0 and with variance of 1). It’s stored in <code>srat[['RNA']]@scale.data</code> and used in following PCA. Default is to run scaling only on variable genes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co">#*************************************************</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="co">#__________Scaling the data__________</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a>all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seurat_obj)</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_obj, <span class="at">features =</span> all.genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
</div>
<p>PCA</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="co">#_________Linear Dimensional Reduction_____________</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(<span class="at">object =</span> seurat_obj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  TYROBP, FCER1G, AIF1, HLA-DQA1, CD14, C1QC, C1QA, C1QB, PYCARD, ARPC1B 
       CXCL16, SH3BGRL3, CCL3L3, HLA-E, RNASET2, GPR183, HLA-DMB, LCP2, RB1, WAS 
       PPT1, RNF130, CTSZ, PTPRE, NCF2, CNPY3, TNFSF12, FERMT3, PPP1R18, CAP1 
Negative:  SPINK1, RHCG, RGS16, GDF15, ADM, MT-ATP8, CRYAB, SLC26A7, KITLG, G0S2 
       RARRES2, ELF3, BNIP3, MIR222HG, DDIT3, RP11-2N1.2, TTR, MTRNR2L12, SLC4A9, EPB41L4A-AS1 
       PCSK1N, KCNQ1OT1, SPAG4, ABCA5, LINC00890, HIST1H1C, INSR, PCP4, RASD1, FABP4 
PC_ 2 
Positive:  HLA-DRA, AIF1, HLA-DMB, HLA-DQA1, TYROBP, MS4A7, HLA-DMA, FCER1G, C1QB, C1QC 
       CD14, C1QA, CXCL16, RNASET2, GPR183, CCL3L3, MAFB, WAS, ARRB2, FERMT3 
       NCKAP1L, LRRC25, CECR1, RHCG, IFI30, NCF2, ADAM28, FAM49B, HLA-DOA, BIN2 
Negative:  TINAGL1, KCNS3, ESAM, UACA, PTK2, DDR2, MYO1B, JAG1, PDLIM7, TMEM47 
       LOXL2, CALHM2, AKAP12, SYDE1, ENAH, RRAS, EDNRA, GRK5, TGFB3, COL18A1 
       PRSS23, ACTN1, SERPINH1, ADGRA2, SORBS3, NDN, LAMA4, TCEAL7, RAB23, RBMS3 
PC_ 3 
Positive:  SLC26A7, RHCG, MIR222HG, SLC4A1, RGS16, COL18A1, PTPRG, HSPA1B, PLA2G4F, LAMA5 
       RHBG, TFRC, SLC4A9, LAMB1, GRN, MTRNR2L12, COL6A1, PKDCC, ARAP1, NRCAM 
       IGF1, RP11-2N1.2, LTBP2, KIAA1109, HOMER1, OGT, CLASP2, COL4A4, MON2, NR0B2 
Negative:  EPB41L4A-AS1, CEBPG, TAF1D, ENO2, MTHFD2, SHMT2, BNIP3, UCHL1, CLGN, CTA-29F11.1 
       KCNMB3, STC2, TGIF1, SNHG12, PSAT1, TMEM45A, CEBPB, DLGAP1-AS2, LARP6, CHAC1 
       KLHDC7B, BNIP3L, LAMP3, TMIE, MAFF, ASNS, PRSS3, RP11-798M19.6, PPP1R15A, WARS 
PC_ 4 
Positive:  DDIT3, EGR1, GDF15, CEBPB, TGIF1, ATF3, XPOT, PSAT1, IDH1, CHAC1 
       CEBPG, STC2, KCNMB3, ADM, RHOB, KLF4, CLGN, RCAN1, DLGAP1-AS2, ENO2 
       KLHDC7B, PPP1R15A, CREG1, ASNS, SLC1A5, IGFBP5, ELF3, SCD, UCHL1, ELMSAN1 
Negative:  TRBC1, CD7, CD52, LTB, CD69, KLRB1, GZMA, LAT, TRDC, ITM2A 
       GNLY, CCL5, TRAC, EVL, KLRD1, TXK, APOBEC3G, RGCC, AP3M2, ADGRE5 
       SOCS2, SH3BGRL3, CLEC2D, CCND3, ALOX5AP, BIN2, RASGRP2, FLT3LG, PGAP1, FMNL1 
PC_ 5 
Positive:  TRBC1, CD7, CD52, LTB, KLRB1, CD69, FOSB, TRDC, GZMA, KLF6 
       LAT, CCL5, TXK, CTD-3252C9.4, FOSL2, GNLY, ITM2A, BIRC3, VPS37B, KDM6B 
       MCL1, IRF1, PPP1R15A, EGR1, NEDD9, ZFP36L2, KLF4, MIDN, REL, PRMT9 
Negative:  IFITM3, CD14, C1QC, GLMP, RBP1, MS4A7, STAB1, MAF, C1QB, SLC40A1 
       FABP4, C1QA, BLVRA, LRRC25, LTC4S, RASSF4, SRD5A3, EMP2, CREG1, NAGK 
       HLA-DMB, CMKLR1, PCP4, DAB2, NDN, HLA-DRA, CRYAB, IFI6, KCNS3, CTSZ </code></pre>
</div>
<div class="sourceCode cell-code" id="cb99"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb99-1"><a href="#cb99-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Examine PCA Results</span></span>
<span id="cb99-2"><a href="#cb99-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Examine and visualize PCA results a few different ways</span></span>
<span id="cb99-3"><a href="#cb99-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(seurat_obj[[<span class="st">"pca"</span>]], <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">nfeatures =</span> <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>PC_ 1 
Positive:  TYROBP, FCER1G, AIF1, HLA-DQA1, CD14 
Negative:  SPINK1, RHCG, RGS16, GDF15, ADM 
PC_ 2 
Positive:  HLA-DRA, AIF1, HLA-DMB, HLA-DQA1, TYROBP 
Negative:  TINAGL1, KCNS3, ESAM, UACA, PTK2 
PC_ 3 
Positive:  SLC26A7, RHCG, MIR222HG, SLC4A1, RGS16 
Negative:  EPB41L4A-AS1, CEBPG, TAF1D, ENO2, MTHFD2 
PC_ 4 
Positive:  DDIT3, EGR1, GDF15, CEBPB, TGIF1 
Negative:  TRBC1, CD7, CD52, LTB, CD69 
PC_ 5 
Positive:  TRBC1, CD7, CD52, LTB, KLRB1 
Negative:  IFITM3, CD14, C1QC, GLMP, RBP1 </code></pre>
</div>
</div>
<p>PCA plots</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb101"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb101-1"><a href="#cb101-1" aria-hidden="true" tabindex="-1"></a><span class="co">#PLOTS PCA</span></span>
<span id="cb101-2"><a href="#cb101-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VizDimLoadings</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">reduction =</span> <span class="st">"pca"</span>) <span class="sc">&amp;</span> </span>
<span id="cb101-3"><a href="#cb101-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">5</span>), <span class="at">axis.title=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">8</span>,<span class="at">face=</span><span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">"pca"</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb104"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb104-1"><a href="#cb104-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimHeatmap</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimHeatmap</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>, <span class="at">nfeatures =</span> <span class="dv">20</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimHeatmap</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>, <span class="at">nfeatures =</span> <span class="dv">20</span>, <span class="at">cells =</span> <span class="dv">500</span>, <span class="at">balanced =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb107"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb107-1"><a href="#cb107-1" aria-hidden="true" tabindex="-1"></a><span class="co">#________Determine Dimensionality Of The Dataset_________</span></span>
<span id="cb107-2"><a href="#cb107-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ElbowPlot</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-19-7.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We can now do clustering. Higher resolution leads to more clusters (default is 0.8). It would be very important to find the correct cluster resolution in the future, since cell type markers depends on cluster definition.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a><span class="co">#********************************************************</span></span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a><span class="co">#___________FindNeighbours For Clustering_________________</span></span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>) <span class="co">#Depending on Dimensionality,clustering will bet set accordingly</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing nearest neighbor graph</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Computing SNN</code></pre>
</div>
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(seurat_obj, <span class="at">resolution =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 809
Number of edges: 28716

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.8582
Number of communities: 8
Elapsed time: 0 seconds</code></pre>
</div>
<div class="sourceCode cell-code" id="cb113"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb113-1"><a href="#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">Idents</span>(seurat_obj), <span class="dv">5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>AAGTCTGAGACCACGA-1 ACTGCTCAGATGTGGC-1 AGCCTAACAGAGTGTG-1 ATCGAGTTCGGTTAAC-1 
                 7                  7                  7                  7 
ATTCTACAGGCGACAT-1 
                 7 
Levels: 0 1 2 3 4 5 6 7</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb115"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb115-1"><a href="#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="co">#************************************************************</span></span>
<span id="cb115-2"><a href="#cb115-2" aria-hidden="true" tabindex="-1"></a><span class="co">#__________Non-Linear Dimensional Reduction____________________</span></span>
<span id="cb115-3"><a href="#cb115-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj, <span class="at">verbose =</span> F)</span>
<span id="cb115-4"><a href="#cb115-4" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to 'umap-learn' and metric to 'correlation'
This message will be shown once per session</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:56:55 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:56:55 Read 809 rows and found 15 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:56:55 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:56:55 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
15:56:55 Writing NN index file to temp file /var/folders/k5/2sqf38tx6cl7vn0xqg7hckb40000gq/T//Rtmp4d6Vv1/filec6ba78b5d83c
15:56:55 Searching Annoy index using 1 thread, search_k = 3000
15:56:55 Annoy recall = 100%
15:56:55 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
15:56:56 Initializing from normalized Laplacian + noise (using RSpectra)
15:56:56 Commencing optimization for 500 epochs, with 31896 positive edges
15:56:57 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunTSNE</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Once clustering is done, active identity is reset to clusters (“seurat_clusters” in metadata). Let’s look at cluster sizes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>seurat_clusters)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
  0   1   2   3   4   5   6   7 
164 140 133 102 100  71  70  29 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Plot</span></span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj,<span class="at">label.size =</span> <span class="dv">4</span>,<span class="at">repel =</span> T,<span class="at">label =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>pca_plot <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">"pca"</span>)</span>
<span id="cb132-2"><a href="#cb132-2" aria-hidden="true" tabindex="-1"></a>umap_plot <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">"umap"</span>)</span>
<span id="cb132-3"><a href="#cb132-3" aria-hidden="true" tabindex="-1"></a>tsne_plot <span class="ot">&lt;-</span> <span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">"tsne"</span>)</span>
<span id="cb132-4"><a href="#cb132-4" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(</span>
<span id="cb132-5"><a href="#cb132-5" aria-hidden="true" tabindex="-1"></a>  pca_plot <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>),</span>
<span id="cb132-6"><a href="#cb132-6" aria-hidden="true" tabindex="-1"></a>  umap_plot <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>), </span>
<span id="cb132-7"><a href="#cb132-7" aria-hidden="true" tabindex="-1"></a>  tsne_plot <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>),</span>
<span id="cb132-8"><a href="#cb132-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"PCA PLOT"</span> , <span class="st">"UMAP PLOT"</span>, <span class="st">"t-SNE PLOT"</span>), <span class="at">ncol =</span> <span class="dv">3</span></span>
<span id="cb132-9"><a href="#cb132-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb132-10"><a href="#cb132-10" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(combined_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-23-2.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Cluster Markers</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="co">#*************************************************</span></span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a><span class="co">#*____________Differential Expression and Marker Selection || CLUSTER BIOMARKERS____________</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a><span class="fu">DefaultAssay</span>(seurat_obj) <span class="ot">&lt;-</span> <span class="st">"RNA"</span></span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Normalizing layer: counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(seurat_obj, <span class="at">selection.method =</span> <span class="st">"vst"</span>, <span class="at">nfeatures =</span> <span class="dv">2000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Finding variable features for layer counts</code></pre>
</div>
<div class="sourceCode cell-code" id="cb137"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb137-1"><a href="#cb137-1" aria-hidden="true" tabindex="-1"></a>all.genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(seurat_obj)</span>
<span id="cb137-2"><a href="#cb137-2" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_obj, <span class="at">features =</span> all.genes)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Find markers for every cluster compared to all remaining cells, report only the positive ones</span></span>
<span id="cb139-2"><a href="#cb139-2" aria-hidden="true" tabindex="-1"></a>all.markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(seurat_obj, <span class="at">only.pos =</span> T, <span class="at">min.pct =</span> <span class="fl">0.5</span>, <span class="at">logfc.threshold =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 0</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>For a (much!) faster implementation of the Wilcoxon Rank Sum Test,
(default method for FindMarkers) please install the presto package
--------------------------------------------
install.packages('devtools')
devtools::install_github('immunogenomics/presto')
--------------------------------------------
After installation of presto, Seurat will automatically use the more 
efficient implementation (no further action necessary).
This message will be shown once per session</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 1</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 5</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 6</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Calculating cluster 7</code></pre>
</div>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>all.markers <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">2</span>, <span class="at">wt =</span> avg_log2FC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 16 × 7
# Groups:   cluster [8]
       p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene    
       &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;   
 1 9.39e- 52       1.60 0.957 0.442 1.51e- 47 0       ATP6V1G3
 2 6.12e- 28       1.56 0.677 0.222 9.81e- 24 0       SPX     
 3 6.79e- 45       2.39 0.764 0.25  1.09e- 40 1       SLC26A7 
 4 2.98e- 34       2.45 0.629 0.2   4.78e- 30 1       CLCNKB  
 5 8.44e- 19       1.66 0.586 0.241 1.35e- 14 2       SOX4    
 6 1.57e- 16       1.50 0.556 0.232 2.52e- 12 2       EMX2    
 7 1.08e- 58       3.85 0.598 0.05  1.73e- 54 3       TRIB3   
 8 6.36e- 55       4.17 0.5   0.03  1.02e- 50 3       KCNMB3  
 9 8.06e-100      11.2  0.59  0     1.29e- 95 4       HCK     
10 6.03e- 84      10.9  0.5   0     9.67e- 80 4       CLEC7A  
11 5.99e-117       8.78 0.69  0.001 9.60e-113 5       EHD2    
12 1.83e- 89      11.0  0.521 0     2.93e- 85 5       SMOC2   
13 2.15e- 11       1.64 0.586 0.273 3.45e-  7 6       TRGC1   
14 2.17e- 10       1.71 0.514 0.245 3.48e-  6 6       MT3     
15 8.88e-141      10.3  0.828 0.001 1.42e-136 7       CD3E    
16 4.39e-116       9.89 0.69  0.001 7.04e-112 7       CD3D    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>top5_markers <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(all.markers <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">wt =</span> avg_log2FC))</span>
<span id="cb151-2"><a href="#cb151-2" aria-hidden="true" tabindex="-1"></a>top5_markers</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           p_val avg_log2FC pct.1 pct.2     p_val_adj cluster     gene
1   9.392488e-52   1.598449 0.957 0.442  1.506649e-47       0 ATP6V1G3
2   3.474504e-37   1.552056 0.835 0.318  5.573452e-33       0    MUCL1
3   6.116748e-28   1.563202 0.677 0.222  9.811875e-24       0      SPX
4   1.415834e-24   1.448857 0.640 0.214  2.271139e-20       0    TRGC1
5   2.411213e-24   1.442558 0.677 0.225  3.867826e-20       0   TSPAN7
6   6.930939e-46   2.368421 0.757 0.238  1.111792e-41       1    ITGA6
7   6.786974e-45   2.387674 0.764 0.250  1.088698e-40       1  SLC26A7
8   6.179585e-41   2.270599 0.743 0.256  9.912673e-37       1   THSD7A
9   8.821099e-36   2.300989 0.671 0.230  1.414992e-31       1     ST14
10  2.979319e-34   2.451400 0.629 0.200  4.779126e-30       1   CLCNKB
11  1.837238e-23   1.469435 0.759 0.343  2.947113e-19       2     ELF3
12  1.292205e-20   1.223186 0.782 0.444  2.072827e-16       2  RARRES2
13  8.440120e-19   1.660199 0.586 0.241  1.353880e-14       2     SOX4
14  1.571945e-16   1.495633 0.556 0.232  2.521558e-12       2     EMX2
15  2.071157e-16   1.271370 0.677 0.327  3.322342e-12       2 HIST1H1C
16  5.795142e-67   3.246339 0.863 0.137  9.295987e-63       3     LCN2
17  1.078143e-58   3.849538 0.598 0.050  1.729449e-54       3    TRIB3
18  6.475640e-58   2.887163 0.804 0.123  1.038757e-53       3    CEBPG
19  6.363022e-55   4.172858 0.500 0.030  1.020692e-50       3   KCNMB3
20  1.655296e-41   2.670018 0.657 0.109  2.655260e-37       3    PSAT1
21 2.174203e-116   7.963803 0.710 0.004 3.487639e-112       4   RNASE6
22 2.095430e-110   8.641604 0.660 0.001 3.361279e-106       4     MNDA
23 7.528552e-109   9.213426 0.650 0.001 1.207655e-104       4   LILRB4
24 8.062294e-100  11.227532 0.590 0.000  1.293273e-95       4      HCK
25  6.025749e-84  10.879920 0.500 0.000  9.665904e-80       4   CLEC7A
26 6.305495e-133   8.718914 0.789 0.003 1.011464e-128       5   COL6A2
27 8.385343e-120   8.349824 0.718 0.003 1.345093e-115       5   COX7A1
28 5.985072e-117   8.776074 0.690 0.001 9.600654e-113       5     EHD2
29  1.372370e-94   8.207679 0.577 0.003  2.201418e-90       5     TFPI
30  1.825709e-89  10.997386 0.521 0.000  2.928620e-85       5    SMOC2
31  2.717745e-12   1.616379 0.571 0.263  4.359535e-08       6    MRPL1
32  2.153199e-11   1.640575 0.586 0.273  3.453947e-07       6    TRGC1
33  2.171712e-10   1.706581 0.514 0.245  3.483643e-06       6      MT3
34  1.393172e-09   1.565101 0.514 0.254  2.234787e-05       6    HMGB3
35  4.708872e-09   1.529878 0.514 0.256  7.553502e-05       6    FABP3
36 8.879998e-141  10.296740 0.828 0.001 1.424441e-136       7     CD3E
37 7.326102e-137   9.349649 0.862 0.004 1.175180e-132       7    TRBC1
38 4.387723e-116   9.894490 0.690 0.001 7.038347e-112       7     CD3D
39 2.850665e-112   9.315627 0.724 0.004 4.572751e-108       7      CD7
40 6.599223e-111   9.143521 0.690 0.003 1.058581e-106       7      LCK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a>top5_markers <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(all.markers <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span> <span class="fu">top_n</span>(<span class="at">n =</span> <span class="dv">5</span>, <span class="at">wt =</span> avg_log2FC))</span>
<span id="cb153-2"><a href="#cb153-2" aria-hidden="true" tabindex="-1"></a>top5_markers</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           p_val avg_log2FC pct.1 pct.2     p_val_adj cluster     gene
1   9.392488e-52   1.598449 0.957 0.442  1.506649e-47       0 ATP6V1G3
2   3.474504e-37   1.552056 0.835 0.318  5.573452e-33       0    MUCL1
3   6.116748e-28   1.563202 0.677 0.222  9.811875e-24       0      SPX
4   1.415834e-24   1.448857 0.640 0.214  2.271139e-20       0    TRGC1
5   2.411213e-24   1.442558 0.677 0.225  3.867826e-20       0   TSPAN7
6   6.930939e-46   2.368421 0.757 0.238  1.111792e-41       1    ITGA6
7   6.786974e-45   2.387674 0.764 0.250  1.088698e-40       1  SLC26A7
8   6.179585e-41   2.270599 0.743 0.256  9.912673e-37       1   THSD7A
9   8.821099e-36   2.300989 0.671 0.230  1.414992e-31       1     ST14
10  2.979319e-34   2.451400 0.629 0.200  4.779126e-30       1   CLCNKB
11  1.837238e-23   1.469435 0.759 0.343  2.947113e-19       2     ELF3
12  1.292205e-20   1.223186 0.782 0.444  2.072827e-16       2  RARRES2
13  8.440120e-19   1.660199 0.586 0.241  1.353880e-14       2     SOX4
14  1.571945e-16   1.495633 0.556 0.232  2.521558e-12       2     EMX2
15  2.071157e-16   1.271370 0.677 0.327  3.322342e-12       2 HIST1H1C
16  5.795142e-67   3.246339 0.863 0.137  9.295987e-63       3     LCN2
17  1.078143e-58   3.849538 0.598 0.050  1.729449e-54       3    TRIB3
18  6.475640e-58   2.887163 0.804 0.123  1.038757e-53       3    CEBPG
19  6.363022e-55   4.172858 0.500 0.030  1.020692e-50       3   KCNMB3
20  1.655296e-41   2.670018 0.657 0.109  2.655260e-37       3    PSAT1
21 2.174203e-116   7.963803 0.710 0.004 3.487639e-112       4   RNASE6
22 2.095430e-110   8.641604 0.660 0.001 3.361279e-106       4     MNDA
23 7.528552e-109   9.213426 0.650 0.001 1.207655e-104       4   LILRB4
24 8.062294e-100  11.227532 0.590 0.000  1.293273e-95       4      HCK
25  6.025749e-84  10.879920 0.500 0.000  9.665904e-80       4   CLEC7A
26 6.305495e-133   8.718914 0.789 0.003 1.011464e-128       5   COL6A2
27 8.385343e-120   8.349824 0.718 0.003 1.345093e-115       5   COX7A1
28 5.985072e-117   8.776074 0.690 0.001 9.600654e-113       5     EHD2
29  1.372370e-94   8.207679 0.577 0.003  2.201418e-90       5     TFPI
30  1.825709e-89  10.997386 0.521 0.000  2.928620e-85       5    SMOC2
31  2.717745e-12   1.616379 0.571 0.263  4.359535e-08       6    MRPL1
32  2.153199e-11   1.640575 0.586 0.273  3.453947e-07       6    TRGC1
33  2.171712e-10   1.706581 0.514 0.245  3.483643e-06       6      MT3
34  1.393172e-09   1.565101 0.514 0.254  2.234787e-05       6    HMGB3
35  4.708872e-09   1.529878 0.514 0.256  7.553502e-05       6    FABP3
36 8.879998e-141  10.296740 0.828 0.001 1.424441e-136       7     CD3E
37 7.326102e-137   9.349649 0.862 0.004 1.175180e-132       7    TRBC1
38 4.387723e-116   9.894490 0.690 0.001 7.038347e-112       7     CD3D
39 2.850665e-112   9.315627 0.724 0.004 4.572751e-108       7      CD7
40 6.599223e-111   9.143521 0.690 0.003 1.058581e-106       7      LCK</code></pre>
</div>
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="co"># find all markers of cluster 1</span></span>
<span id="cb155-2"><a href="#cb155-2" aria-hidden="true" tabindex="-1"></a><span class="co">#cluster1.markers &lt;- FindMarkers(seurat_obj, ident.1 = 1)</span></span>
<span id="cb155-3"><a href="#cb155-3" aria-hidden="true" tabindex="-1"></a><span class="co">#head(cluster1.markers, n = 5)</span></span>
<span id="cb155-4"><a href="#cb155-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-5"><a href="#cb155-5" aria-hidden="true" tabindex="-1"></a><span class="co"># find all markers distinguishing cluster 5 from clusters 0 and 3</span></span>
<span id="cb155-6"><a href="#cb155-6" aria-hidden="true" tabindex="-1"></a><span class="co">#cluster5.markers &lt;- FindMarkers(seurat_obj, ident.1 = 5, ident.2 = c(0, 3))</span></span>
<span id="cb155-7"><a href="#cb155-7" aria-hidden="true" tabindex="-1"></a><span class="co">#head(cluster5.markers, n = 5)</span></span>
<span id="cb155-8"><a href="#cb155-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb155-9"><a href="#cb155-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the aggregated markers as a CSV file in the current working directory</span></span>
<span id="cb155-10"><a href="#cb155-10" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(all.markers, <span class="at">file =</span> <span class="st">"seurat_markers.csv"</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>More defined plots</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb156"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="co">#__________________PLOTS_____________________</span></span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="co">#_________DimPlot___________</span></span>
<span id="cb156-3"><a href="#cb156-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(seurat_obj, <span class="at">dims =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>15:57:03 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:57:03 Read 809 rows and found 15 numeric columns</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:57:03 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Found more than one class "dist" in cache; using the first, from namespace 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Also defined by 'spam'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>15:57:03 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>**************************************************|
15:57:03 Writing NN index file to temp file /var/folders/k5/2sqf38tx6cl7vn0xqg7hckb40000gq/T//Rtmp4d6Vv1/filec6ba5db63003
15:57:03 Searching Annoy index using 1 thread, search_k = 3000
15:57:03 Annoy recall = 100%
15:57:04 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
15:57:04 Initializing from normalized Laplacian + noise (using RSpectra)
15:57:04 Commencing optimization for 500 epochs, with 31896 positive edges
15:57:05 Optimization finished</code></pre>
</div>
<div class="sourceCode cell-code" id="cb168"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb168-1"><a href="#cb168-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">reduction =</span> <span class="st">"umap"</span>,<span class="at">label=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb169"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb169-1"><a href="#cb169-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb170"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb170-1"><a href="#cb170-1" aria-hidden="true" tabindex="-1"></a><span class="co">#______VlnPlot________</span></span>
<span id="cb170-2"><a href="#cb170-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"HCK"</span>, <span class="st">"SMOC2"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb171"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb171-1"><a href="#cb171-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"HCK"</span>, <span class="st">"SMOC2"</span>), <span class="at">slot =</span> <span class="st">"counts"</span>, <span class="at">log =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The `slot` argument of `VlnPlot()` is deprecated as of Seurat 5.0.0.
ℹ Please use the `layer` argument instead.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb173"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb173-1"><a href="#cb173-1" aria-hidden="true" tabindex="-1"></a><span class="co">#_____featurePlot______</span></span>
<span id="cb173-2"><a href="#cb173-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"MS4A1"</span>, <span class="st">"GNLY"</span>, <span class="st">"CD3E"</span>, <span class="st">"CD14"</span>, <span class="st">"FCER1A"</span>, <span class="st">"FCGR3A"</span>, <span class="st">"LYZ"</span>, <span class="st">"PPBP"</span>,<span class="st">"CD8A"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The following requested variables were not found: MS4A1, FCER1A, PPBP,
CD8A</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb175"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb175-1"><a href="#cb175-1" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">VariableFeatures</span>(seurat_obj)</span>
<span id="cb175-2"><a href="#cb175-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> features[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb176"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb176-1"><a href="#cb176-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> features[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">blend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-7.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb177"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb177-1"><a href="#cb177-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ___interactive plots --------</span></span>
<span id="cb177-2"><a href="#cb177-2" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> <span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"HCK"</span>))</span>
<span id="cb177-3"><a href="#cb177-3" aria-hidden="true" tabindex="-1"></a><span class="fu">HoverLocator</span>(<span class="at">plot =</span> plot, <span class="at">information =</span> <span class="fu">FetchData</span>(seurat_obj, <span class="at">vars =</span> <span class="fu">c</span>(<span class="st">"ident"</span>, <span class="st">"PC_1"</span>, <span class="st">"nFeature_RNA"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `error_y.color` does not currently support multiple values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `error_x.color` does not currently support multiple values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: `line.color` does not currently support multiple values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The titlefont attribute is deprecated. Use title = list(font = ...)
instead.</code></pre>
</div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-e4d2c5c78f37d06ba907" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-e4d2c5c78f37d06ba907">{"x":{"visdat":{"c6ba640a4505":["function () ","plotlyVisDat"]},"cur_data":"c6ba640a4505","attrs":{"c6ba640a4505":{"x":{},"y":{},"mode":"markers","hoverinfo":"text","text":{},"color":{},"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter"}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"xaxis":{"domain":[0,1],"automargin":true,"title":"umap_1","showgrid":false,"zeroline":false,"showline":true},"yaxis":{"domain":[0,1],"automargin":true,"title":"umap_2","showgrid":false,"zeroline":false,"showline":true},"title":"HCK","titlefont":{"color":"black"},"paper_bgcolor":"white","plot_bgcolor":"white","hovermode":"closest","showlegend":false},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[9.6691077062019755,-6.8720142534843038,-3.4766180208793234,-2.1119153193107199,-4.3705885103812765,-2.8876487902274679,-4.840661256658132,17.156886846674389,-3.6181170633903097,-6.9427387407889913,-2.9712879351249288,-4.6452810457816671,-7.4216043642631124,-5.1789495638480734,-5.3665544680228781,-3.5008938006034445,-6.1587621859183859,-2.286833017217214,-1.4620234659782003,-4.1524612597098898,-7.0627462557426046,-4.5385429552665304,8.6715290852913309,-3.12603828226238,21.508370191706128,-6.2431242159476827,8.5971200772652079,9.5946710416206766,-2.4477560213676046,-5.6809894732108663,21.622235090387768,4.8042212435135294,-1.4961416414847921,16.638273031366772,-0.67586490427165735,-1.7116367510428976,9.2307517835030009,-2.6911365679374288,-1.0255644014945577,-3.2825414827933859,-2.2575814417472433,-4.947695939885671,21.684271604669995,-5.534269540654714,-2.9788648775687765,-5.5889713457694601,-3.2372286013236593,-3.2933528116813253,-2.5759670427909445,-2.1077739885917257,-6.8744766405692648,17.215513975275464,-1.4976412943473409,-5.4088842562308859,-3.0811998537650656,8.5142467328438212,-3.799284189092214,-4.5134794405570577,-2.4201762369742941,-1.1391136339774679,-5.9709922960868429,16.838927061212964,16.5161636182198,16.889560491693921,9.1910083600410868,8.8251197644600321,-7.0830223253837179,-3.4148299387565206,-1.7026097468009542,-3.7344596079459738,-0.63242122446208704,-5.4466335466971945,-6.082670419560964,-2.2475626162162374,-3.2257969072928976,-3.3633644274344991,-1.923274724828298,9.714540273798411,-4.5320818117728781,-2.0354230097404074,-7.0542576006522726,-1.1970970324149679,-5.5872423342338156,-2.6457702807059835,-4.3468057802787374,8.7215359517464091,-1.5103633097282003,-2.7241747072806906,-1.6993906191459249,9.0668547459968973,-6.1130125216117452,8.5853727170357157,-5.156954973088796,-4.775083749639089,-2.8419310740104269,4.9331321426758219,-4.6677896669974874,-3.214043825017507,-1.9897124460807394,-3.9632541826835226,-4.0595104387870382,-2.4379927805533956,-5.2589790514579366,-5.3611309221854757,-5.6809885195365499,-0.66544553552776087,9.3585417577156473,-4.4673320940604757,-0.85567256723552454,16.81031301702351,-6.2864772966971945,-2.2596518686881613,-4.5814935854545187,-6.8237211397758077,-3.4970547846427511,-1.5592190912833761,-2.6166631868949484,-1.0326101473441671,-1.9894997767081808,5.0429900595077921,-2.2744667223563741,-1.9762069872489523,-0.95762178217082727,-2.6019894770255636,-3.4340269259086202,-6.2099401644340109,-3.5798499277701925,-3.3675429514518331,-1.365965097295339,-2.8708183458915304,-5.0593588045707296,16.634856969965405,8.8835671254524637,17.086176664484448,-3.497249334203298,-2.9331452540031027,-5.1089212587943624,-1.0261227778068136,8.2353666135200907,8.568544180048411,-1.9590189150443624,16.854558736933178,-5.4647428682914327,-7.4304849794974874,8.5425679990181376,-7.254148691045339,-2.9985344103446554,-5.2166893175712179,17.278836996210522,-2.3057968309989523,-5.6004831484427999,8.4342220136055399,-6.8470327547660421,8.7037012883552958,-2.434909074651296,-3.0357882670035909,-6.4347288302054952,-5.166410653935964,-2.564578264104421,-5.0090505770316671,-5.5363323382011007,-6.3112833193412374,-4.8056652239432882,-4.4627601793876241,-3.5046298197379659,-0.93682214533000696,-7.5417911699882101,-5.4254705599418234,17.152429372919507,-4.0922109774222921,-5.4418947390189718,-5.6127569368949484,-1.4814026049247335,-4.5793335131278585,17.28559663976521,8.6217382260689188,4.7814033337959696,-2.2211343935599874,-2.3572876146903585,-7.046471803533132,-2.4221632174125265,-2.9461495569816183,-2.3061120203605245,17.161314756525464,-2.1705267122855734,-0.80306836877971399,-6.0601427248588156,17.085800916803784,-7.386869638310964,-1.4088103464713644,-2.9186522654166769,-5.6514475039115499,-5.3641712359061788,-5.4969989946952413,5.1337507077583719,8.4051721402534891,-2.4421050242057394,-2.6858851603141378,-2.2650987795463156,-6.8532106569877218,-6.6090471438041281,21.450152189386792,8.8353412457832743,8.3798129865059305,-3.9755642107596945,-2.4870836428275656,8.8748810597786356,8.8591049023994852,-6.0866062334647726,-5.4285146883597921,-3.5546438387504171,5.0940343448051859,-1.3686415842643331,-5.2730686358085226,-4.9730932406058859,-5.6826345614066671,16.674558431757397,-3.0108334711661886,16.873410970819897,5.1532846399673868,-5.1933767489066671,16.750194341791577,5.0992257901558329,-7.3143045595756124,17.077421934259839,-2.2452809504142355,-1.5975210360160421,-1.7135259798636984,-6.7934066942802023,-1.5874478510490011,9.4822657414802958,-1.7680489710441183,17.347405225885815,-3.3891427210441183,-7.3715774706473898,-4.361601083623464,-1.2245728663077902,5.0222246953377176,-4.221734254705007,-2.784940927373464,21.607785017145581,5.1672420808205057,-3.4455716303458761,-3.1614119700065206,-2.135897844182546,-2.5405361345878195,-1.4191329172721456,8.8995082684883524,-5.8258583239188741,-1.8434350184073995,-1.9365389040580343,17.323413641107983,-2.3729364565482687,-3.017733781682546,-0.91007491861491907,-2.4053642443290304,8.7148897954353739,-7.4083378008475851,-7.1448404482474874,16.74588659490437,-2.8983838251700949,-2.1666633776298116,-5.792351930486257,-2.4420425585380148,-1.7403723887076925,-2.9919669321647238,-6.8329622439017843,-3.9333402804008077,-3.2189438036552023,-4.6054441622367452,-1.3014084986320089,-1.583860605107839,-2.7393438509574484,-6.1878893068900656,-3.8630392244926046,8.6231782742866923,8.4538882085213114,-5.6748773745170187,-3.3350851229301046,-2.6718651941886495,9.5119607755073954,-0.99987384592204798,-3.2508561304679464,-5.7316991976371359,-6.0289671114554952,-3.0854689768424581,-3.9445106676688741,9.3579232999214579,-1.4194175890555929,-7.2895357302299093,-5.6300165346732687,16.984036237848706,-3.6131303003898214,-1.6253153971305441,-3.7290317705741476,-3.3426248720756124,8.3952033826240946,-2.0945861033073019,-3.1072475603690695,17.047389776361889,-1.4658729723563741,-5.6234943560233663,-6.8687393358817648,-2.181310384618337,16.9484145947823,-2.3285600832572531,9.5250537701973368,-4.8077385119071554,-4.015409677373464,-4.4295093706718038,-2.5469338587394308,-2.784121721135671,8.4738953420051981,-4.3559019259086202,-6.1363165072074484,-1.6993205240836691,-7.2521593264213156,-1.3841478518119406,-2.4332835367789816,-5.9991447619071554,-3.3047692469230245,-2.3549472979178976,-5.8794777086844991,-3.8062288454642843,-4.4120084932914327,-1.4446727923026632,21.511847288263745,-6.5492069414725851,-1.1944520166984152,16.762036115778393,-4.555937021123464,-3.8831197908988546,4.9119278499016215,-2.0979439905753683,17.231219083918042,9.2017286130318094,-7.2866375139823507,-4.7130186251273702,-6.4561788729301046,-3.7451159647574972,-2.9393288782706808,17.038081915033764,-0.71680328165202845,-3.4068824938407491,-2.2128188303580831,8.768314630640452,-6.8305380037894796,8.1628358670601298,-3.1965501001945089,-7.5197851351371359,-1.3711774042716574,-2.7278859308829855,-5.5173961809745382,-3.0108549288383077,-4.0236332110038351,16.723946363581128,-2.8210580042472433,-5.8233949831595968,-3.1720406702628683,-2.815697877751882,-4.989342897283132,-6.2098085573783468,-6.2670547655692648,-2.1742932489982199,4.9537160703072001,-2.3252231768241476,5.0802466579803873,9.7966440030464579,8.5850408383736063,-1.8488356760612081,-5.9134914568534445,9.1533410855659891,-7.2281362703910421,-2.87182685648113,21.516009122980542,-1.9479472330680441,-2.4245903185477804,-4.5706064394584249,-3.2228386095634054,-6.2930528811088156,-1.891820638524587,8.9015205212959696,21.665087492121167,5.0614241668114115,-3.5795557192435812,-6.2858898333182882,-1.140081136571462,-3.0925824335685324,-7.4175550631156515,-7.4729921511283468,-1.4272224596610616,-5.8518984011283468,-5.7773124865165304,-4.3182394198051046,-5.8161136797538351,-5.6991827181449484,-3.9430124453177999,-5.5763551882377218,-4.0469133547416281,17.215166837824292,-4.4254257372489523,-6.1449358156791281,-7.3311016253104757,-4.4450685671439718,9.0653112241157938,-3.3358294657340597,-4.1334621599784445,8.4901635953316141,-7.1525842836967062,-2.0024797609916281,16.773068220270581,-2.7096225908866476,-5.5356838396659445,-1.3931486300101827,17.015488416803784,-4.2139837435355734,-3.1367237261405538,16.972171575678296,-7.074374406682546,-1.310484140264089,9.0672009297737528,16.867918760431714,8.801550657404368,17.147824079645581,-3.0100562265983175,-3.2479617289176534,-2.4244453600516866,8.3928859540352274,-2.5375921419730734,-2.0253216913810324,-4.1707670382133077,-1.2744047335258077,-4.4190141848197531,8.7349341222176005,16.512693197382397,-3.6775700739493917,5.204049915922587,-3.1588132074943136,17.256837637079663,-6.2967083147636007,4.9748476454147745,-2.11981269632488,9.775725633753245,-3.1981980494132589,8.8997323819527079,-3.2743575266471456,16.673681051386303,8.587094099176829,-7.6094247988334249,9.4595444508919169,-3.661553590642507,-2.5317098787894796,-4.2277795961967062,-1.8781664064994406,-5.8280346087089132,-2.2612082651725363,-2.5152742556205343,-5.8583795717826437,-2.7784416369071554,-2.5823542765250753,9.5953424283394266,-4.3103372744193624,-2.7716638735404562,-6.2144643953910421,-4.5348074129691671,-1.7229392222037863,8.6126363583931376,16.873684675348706,-0.5794064692130636,-1.0440370730033468,-2.9168851069083761,17.352705747736401,9.178980142725413,-2.2485496691337179,-3.6265937975516866,-1.1899964502921652,-2.2305791071525167,-3.0381643465629171,-4.0370866945853781,-4.4605657747855734,-2.8763339213004659,-1.3758265665641378,16.732841283930249,-2.2831155947318624,-5.0427162340751241,-4.7053301027885031,-1.8088223627677511,-3.3833353212943624,-3.7236239603629659,9.3288595982918192,-1.188561647283132,9.6135518857368876,-5.8859913042655538,9.4076569386848856,-2.7548415354362081,16.80411413396687,-0.80178615366130579,9.6051295110115458,-2.9578778437247824,5.0127109595665384,-1.5572158983817648,-7.3951713732352804,-5.1735546282401632,-2.0794446161857199,8.6690800496467997,9.0318982907661844,-0.90625163828044641,9.4051254102119852,8.6366231747993876,8.7175600835213114,-5.9387981585136007,5.0574770876297404,-1.4417950800529074,8.7120006390938212,17.09161356176228,-0.9624783686271261,-5.2796594790092062,-6.5547859362235616,-2.1118127993217062,-2.7134363344779562,-3.3674175432792257,-2.8686873606315206,16.653323919428296,-6.6541549852958273,17.049897939814038,-6.6311351946464132,-7.1090891054740499,9.0697515317329813,-5.6114895037284445,-3.1805364779105734,-0.81945630823283899,-2.2352029970756124,16.877951414240307,-2.8847644022574972,-0.66038915430217493,-7.1676914385428976,-6.2606069735160421,16.896304876459546,-5.0628368548026632,-2.2530996492972921,-6.2638628176322531,-3.4087607554069113,-3.8096973589530538,-3.1888343981376241,-3.2108423403373312,16.872311384333081,-1.5910474947562765,-3.6609046152701925,-2.4902283838859152,9.0000269719490458,-3.9844429186454366,-1.3435403994193624,-2.7664253405204366,-1.8290946177115988,-6.6984197786918234,9.7028286763557841,-7.6424266985526632,-7.0930072954764913,-2.527654378758962,-3.3091857126823019,-1.0196854761710714,-5.5934469393363546,-1.3750788859000753,-2.6527478388419699,-1.5610882929435324,-5.283471315251882,-2.6659862688651632,-2.5522424868217062,-2.0761048487296652,-4.0424215487113546,-2.531889169560964,-2.8864257029166769,-5.149285524236257,-6.5612022570243429,8.3148200818428446,-1.252838819371755,-2.6816398790946554,5.2543045588859965,-4.0768024614921163,8.7367599316963602,9.2777917691597391,-2.0516946009269308,16.706040174616284,-2.7129118136039327,-4.5557806185355734,4.8421532222160746,-3.0391947916617941,-3.085332601415212,-1.7732636622062277,-5.9077331713309835,-5.7605554751029562,-6.0326015642753195,-2.0993468454948019,-5.9202015093436788,-5.1436102083793234,-5.4019205263724874,-5.5142204455009054,-5.2099191835990499,-3.8148443392386984,-6.029875009404714,-7.2814714602103781,-5.026736467229421,9.0756538220772196,-5.452447145330007,-6.9850217989554952,16.765362531794018,-2.5749523333182882,16.685458929193921,-1.8817927530875753,-2.6769125155082296,-5.4030258349052023,-6.1643173388114523,-3.507414071904714,-5.0766155413261007,-3.1252596071830343,16.841248304499096,-7.5130417040458273,-6.4920217684379171,9.0407440968880106,9.2718098470100809,-4.0304062059989523,-3.049523561345632,-6.6886245897880148,-4.8541176012626241,-4.2656709841361593,17.130881101740307,-2.5894000223746847,-4.0868217638602804,8.6544425794014384,-1.0536286524406027,8.4842641660103251,-3.1998302630057882,-3.1516191652885031,-4.9040424517264913,9.5637071439155985,-3.3635537317863058,-3.0385214975944113,-1.4387762240043234,-2.6238510302177023,-5.2588464907279562,9.3981225797066141,4.8730308123955179,-2.4590704134574484,-2.2628013781180929,-5.4457361391654562,-1.6147601297965597,16.739487440241284,-3.2786857775321554,-2.0088593653312277,5.1296448418030192,-4.0352775744071554,-2.6483914545646261,-1.4679166964164327,-2.5641376665702413,-1.5848834208121847,-1.623744695531423,-5.315366952764089,-4.5333568743339132,-6.3815223864188741,-6.0117857149711202,-7.4689237764945577,-2.0722319773307394,-3.0547353914847921,-2.0747234514823507,-1.2955157450309347,4.9402978130707194,-5.967440812932546,9.3419983693489481,16.949356825006909,-1.1766144922843527,-0.86708852564006555,8.2969396420845438,-1.578029363500173,-6.2283212832084249,-3.1521241358390402,-5.1038810900321554,-1.5842048815360616,4.9957920023331095,8.2679989644417216,-3.1717722109427999,8.7634461232551981,-6.097018449651296,-1.9358775309196066,-2.7421619585624288,-1.2480847529044699,-2.2541758707633566,-2.281367986547048,-1.8398887804618429,-5.0455705813041281,-3.1809198549857687,-5.3729955843558859,-2.9499323061576437,17.309181005609936,-5.1494667223563741,9.5718100377449442,5.2311841555962015,9.2694824048408915,-3.7473976305594991,-6.0969030550590109,-1.9261772326102804,-3.3742282084098409,17.172801763666577,-1.0408217600455831,-5.1624281099906515,-3.7030484369865011,-2.011470525609548,-4.5851337603202413,-2.9024860552421163,-3.6018674067130636,-0.99277707849651087,-4.4613420656791281,-5.9846164873710226,-2.4194290331473898,-5.4232370546927999,-5.5027477434745382,-4.1140901735892843,-1.7230388811698507,-4.7906553438773702,-4.967492311345632,8.7062318631538798,-2.0715648821464132,-1.7582785776725363,21.595217497003979,9.3440482922920634,8.9184677907356669,-2.2871834925284933,-6.0844528368583273,16.629364759577221,-3.1875621965995382,-2.160243718968923,-0.8622882059684347,-7.5045644930472921,-2.9830882242789816,-5.0692808321586202,-3.6383034876456808,-1.7987305811515402,-6.6473381212821554,-3.3781344584098409,9.0869257756599833,8.7872770139107157,-2.8727733782401632,5.0759653755554606,5.1986418672928263,-2.5001933268180441,-4.357481210576589,9.5988524266609598,-1.4175264528861593,17.204159528864331,-2.3420636347404074,-1.0349967173209738,-2.3054082087150167,-6.8987838915458273,9.0866182156929423,-7.1905443361869406,5.0096633740791727,-4.0925972155204366,-1.9694039515128683,-6.1545231035819601,-2.9783956698051046,9.2207148381599833,-1.2907197168937277,8.978390485895579,-4.8748495272269796,-2.8140027216544699,9.3149020978340555,8.5210244962105204,-1.6474716356864523,8.1468627759346415,-7.3644773653617452,-3.8909113100638937,-3.4666349581351827,8.5282819577583719,-5.8039629152885031,-2.1508824518790792,-2.8493239573112081,-7.0615541628470968,-6.4901001146903585,-5.7043382814994406,-5.2464344194999288,-2.7723061731925558,-6.2235033205619406,-2.7546097925773214,-1.5251762560477804,-7.1520139864554952,9.2346599408516337,-4.5642158678641866,-5.6961424044242452,-5.5724661043754171,-3.0198013476005148,-3.2821533373466085,-5.3772279909720968,-2.2918836763969015,17.174904615534253,9.5140269109138895,-2.3168518236747335,-4.3831388643851827,16.659672529352612,5.1140824266800333,-6.1169883898368429,-2.0969164064994406,-1.8243114641776632,-0.89966460977702845,-2.6649348429313253,16.870391637934155,9.2630408116707255,-1.2588321856132101,-0.74131557260661829,-2.5965945414176534,-1.5404126337638449],"y":[7.8100949759758498,-2.3266984586440538,-5.7286633019172166,-4.4823504452430223,-0.97240189358820683,-6.0712202553473924,0.7040671344078564,-2.6579782728873704,1.5617326731956982,-3.3789622549735521,-4.6898556713782762,1.7667569155968212,-2.4914943341933702,-2.2149112586699937,0.22107460453877681,-3.0373034243308519,-0.74322251126398808,2.3986400599754836,-0.77086422488322026,0.95772685959706538,-3.0122767452918504,1.1898998256004834,7.6093560691154982,0.51693930580983394,-6.4936536316596483,-1.8052910928451038,7.5567876334465529,5.9548484320915724,-5.2649694447242235,-1.039603686782024,-6.6038793091498826,12.871877812889911,-1.2303988103591419,-4.2042397026740526,-2.7036279206000779,0.64634671166310542,8.0579911704338567,1.4542209143913769,-2.511948860140941,-2.2248559956275438,0.79497375443349116,1.4485970969475293,-6.5340989594184373,-1.9469386999331928,1.2564708228386425,0.43550481751332515,-6.2092679505073045,1.455228232888081,1.9310783381737255,-5.3079343800269578,-1.7844335500918842,-3.1302307610236619,1.5076920028008007,1.4237352843559765,-5.9439691071235154,8.0878106589592473,-3.9484871391974901,0.2067148919380688,1.4233278270042919,0.39873471215138667,0.14657726242909663,-4.057681179495952,-4.3802119259559129,-3.0894182686530565,5.6305290694511916,8.1082288260735051,-2.5073143009864305,-4.9094289307319139,1.9374302859581494,0.83768238976369136,-2.6814360384665941,-0.011487699004314011,-3.1010735993110154,-4.4276193146430467,1.5205301280296826,-5.2007652287208055,0.42508234932790034,6.7770911689079787,0.76525368645558589,1.5307501788414501,-3.3745364908896898,-0.41322419688334233,0.59200372650990718,-4.5966406826697801,-1.8663246397696949,5.8626057143486525,1.7668148513115429,-6.312986708136699,2.0093468661583449,8.2287656302727239,-1.6772432987414814,8.087671422509052,-0.2840467457496143,-0.53400466487040288,-4.7614890103064988,12.829360150841572,0.69017519905934566,1.9994374747551464,0.64142694428334468,0.44511189415822261,-1.3459113602363086,1.7245790000237011,0.61871209099660152,-1.3597997193061329,-0.12403795764078862,-2.7025320296012376,6.2564827437675978,-1.4963611055933452,-2.8314425711356614,-3.2972930435859178,0.28506603196034663,1.4252478118217968,1.934227370766499,-2.4492109183989976,1.7256156440056347,0.082157396820881301,-5.8113749985419725,-0.44471511885752446,-3.7622691158973192,13.201064252404072,-4.7448292736731981,-4.6368239884101365,-3.1519073013984178,0.60629286721120113,-5.7100566391669725,-2.9588081602775071,1.8540939326561474,1.1162921424187207,-3.3223840956412767,-5.5982011322699998,0.42956271126637691,-2.7908501391135667,5.7561580176628615,-2.5453706149779771,-2.5934975509368394,-5.2813457016669725,-0.88080576703181035,-2.9506857161246751,7.6632787223137404,7.3211089606560256,-5.4958027367316697,-3.8289490704261278,1.0339042182243847,-2.0942240719520067,5.6183411116875197,-2.1604229573928331,-6.1279152397834276,1.2945522303856396,-2.2555256252013658,1.6312226290977978,-1.4603771482430912,7.3176213736809279,-1.3753622834407306,5.8052317137993361,-5.8411677841865037,1.5008687014854931,-2.9944691424094652,0.17078354313740962,-3.7479763989173387,0.35287573292622798,0.48889603569874995,-3.1303157572471116,0.45944275811085933,0.56846680596242183,1.1926607127464794,-0.012418961974284715,-1.9655722145759082,0.96934714272389644,-2.9000113968574022,-2.1044038896285508,-0.095681882353923386,0.6072038646019482,1.7418788905418896,0.75981035187611812,-2.2961025600158189,5.3890825743950392,12.796921872643329,1.2221483226097607,-5.089600182028911,-2.6259041790687059,-3.6105965141974901,1.9372614856041455,0.34366002037892573,-2.8359061245643113,-5.2459236149512742,-0.81174341485132939,-1.2063687507831073,-2.5342238311492418,-2.2172435884200548,-2.8671861653052781,-5.2174552445136522,0.7926911349571728,-1.5522343021654106,-0.68856946274866826,14.095931195763447,7.8195373053825881,0.10559346630940669,-2.5102043513976549,1.3441294665611767,-1.711706264766357,-0.57011876151194341,-6.4486726288520311,6.0987478728569533,7.6524812217033888,2.1085146899498488,0.34549500897298091,8.1285068030632512,8.0425187583244817,-1.3879017595969654,0.74514832451710933,0.29307045891652339,14.042358541039325,-2.9044477705680345,1.3914150233543896,-1.0569635991775013,-0.62622187182535893,-4.2631767754279588,-1.2621159021579242,-2.3325218324386094,14.053078794030048,1.6402129168785595,-2.7311665777884935,13.89502921059499,-1.9215703968726612,-2.7794731621467088,-5.4789687161170457,1.1980389590538525,1.2213286395348095,-1.1919075016700245,1.6837765212334179,6.0715238089836623,1.5510391230858349,-2.3761276964866136,-3.1664609675132249,-2.0632588033400987,2.1120342250145461,-0.32851312205424077,12.97369017556081,0.07491841271291011,-5.722342348548076,-6.5890506272040819,13.887822293785907,-1.7868635122500873,-6.1487954621039842,0.53212680771718257,-5.1735218529426072,1.694272422341206,6.2493688102043654,1.1097342009819531,0.81154456093678706,2.0884765620506789,-2.5071090821944688,0.85980644181142085,-1.1909951333724476,-2.7137735848151658,-6.2974303726874803,8.3146897788322942,-2.3867760662757371,-1.6001725143065724,-4.0615213875495408,-6.2233994965278123,2.1220470900810744,0.18926038697133296,-5.7155683998786424,-1.8109344457112766,-5.6794341568671678,-1.6698953275405384,1.7593080039299511,-4.9576338295661424,0.19329621746907466,-3.2725746397696946,0.79253568604359859,-6.0283211235724901,-0.00066885993113285913,1.4612437720573925,6.6834889884270217,7.9442006583488967,0.74863400414357417,-2.6036723618232225,2.1749292846001174,6.8266479010857131,-3.1776550774299119,-5.4477303986274217,0.85843815758595698,-0.1483961109839893,-6.0260389809333299,1.5967785353935742,5.9075213904655959,-2.6759540800773118,-2.9878128533088182,-0.7562975649558521,-3.3946518664084886,-3.8262041573249315,-2.9293312792502855,-3.0485397343360399,-4.4220934872352098,5.7026702399529006,-1.4150186096154667,-4.8881565098487352,-1.9860264425002552,1.3102566237724804,0.34237506344685786,-2.1400873427115892,-3.7752119068824266,-1.9052075509749866,1.0932132239616894,7.4254752631462599,-2.9949884180747484,1.8463019366539501,-1.4401926313363529,1.0644959445274853,-2.5814038281165574,5.8588682647026564,-0.42985556170573003,-1.6796157722197986,-3.9741159443580125,-1.6956742022954441,1.1731881137169384,1.8301147456444287,-1.2449525956832386,-5.4605647091590379,-6.1365407471381639,0.067993902710773879,1.8546990390099072,0.34076073124776118,1.5964852805412793,-6.450710154029033,-1.1201933686458088,0.031662010697224074,-3.0027856592856859,1.2008687491692089,1.190974139717915,12.982789182213642,-5.6558655266486619,-2.4362329010688279,5.6295362944878127,-1.9620716337882496,1.2689844603813671,-0.15218601271738774,-3.3314137224875902,1.4003619189537548,-2.9231773380958055,-2.8706135515891527,-3.9720757965766404,-6.0479734902106737,6.4154635901726271,-2.6071953539573167,6.9098464484489943,-5.2650300030433153,-2.4374253515922044,1.1005820269859814,-5.2518919472419237,-0.38485703513254887,-4.6917460922919725,0.4639543528831982,-3.8054212097846483,-2.948589063140056,-0.51187512919535405,-2.1024455194198106,-1.9368438486777759,0.50994768097768062,-0.73971626803507573,-0.48122546717753178,0.57449831917653316,12.968003415612079,0.37074413254628413,13.178401135948993,7.6524354453362013,5.9322835440910842,1.9373361106194043,-0.20216629550089604,6.5730792517937209,-2.4403973583899949,-3.5542201761924241,-6.5558203224860643,-5.4873755936347459,-6.2821224693976854,-1.65107267990937,-1.3282957439147449,-1.6886389617644764,0.61322894051442378,6.2084647650993849,-6.6472929482184862,14.027149343041279,1.4107831950462841,0.53623261406789058,-1.2424164180480457,1.222301625756123,-2.1565014486037706,-2.4361853365622972,-3.4882414107047532,-0.050134397002360886,-0.65404675051798589,-1.3439160053454853,-1.1658415262423969,0.6831327910698437,0.78936757996449702,-0.12112400576700932,1.828892373589375,-2.6096049313270067,-2.1755289797507738,-0.91168019816508061,-2.4977345828734849,-1.1530892913066364,6.7889510627068068,0.90027894928822749,-2.3172284726821397,7.5385724540031935,-1.8998564843856312,-4.2369705204688524,-2.3359130983077501,-5.9410451416694139,1.2832409377373242,-2.6151356463157152,-3.1825519566260789,0.55895891144643062,0.85170474007497066,-2.1562198166572069,-2.3145659928046678,1.0527841086662793,6.5488521094597365,-3.5263733629905198,6.8869754309929396,-2.6849142079078172,1.3995040889061474,-6.0002439980231737,-5.4186867241584276,6.1377874846733596,-0.31580815360178716,1.6830505366600537,1.7156168456352734,1.5350693698204541,0.16759076073537105,6.0476218695915724,-4.1226702694617723,1.0398539538658642,14.066576146630146,-3.4512601856910203,-2.0959828262053941,-0.9981227283202625,13.085559033898212,2.1997550959862258,7.5301677222527053,-0.90498266265024907,5.5271793837822463,-5.8029540543281053,-3.466062880011699,6.2796770568169142,-2.0099630419932817,7.9882213111198928,-3.6329060558997606,1.7407738204277539,1.0522541041649365,-4.5402961735450242,-0.38280973479380376,-4.661440945120952,-5.4569021229468797,0.31712760880360835,-0.042038178893230027,-6.0756767754279588,6.8634158606804396,1.6027451987541699,-4.9474903111182664,-3.4289492134772752,1.9684392447746777,-0.38555774733652837,6.1415459151543166,-3.2667850022040819,-2.847371793242595,-3.1202003721915696,-6.2237609390937303,-2.6643469099723314,8.0647012229240911,1.7213262553490185,-1.6586367104970432,-3.2543273930274461,-6.2700851921760057,1.1434165473259472,-1.1100147132598377,0.89199247315297359,2.1526674266136672,-3.0592837099753831,-2.7608170275413011,-3.4332026009284471,1.332826995400288,0.62248721077809566,1.1815473075188183,-5.3715926174842332,-1.7886749272071338,7.8572608466423537,0.080596828011372024,7.2450810904778029,0.37856914952168697,6.0299197669304396,-0.80748382851710088,-2.2207345132552598,-2.8768779997550462,6.267368459252217,-5.8733347420417283,13.087414884117939,0.8395046229637646,-1.8397280339919544,-2.4854874376975511,-4.4433547024451707,5.7269231314934279,5.5908833976067092,-0.8517485623084522,6.3925143714226271,8.2819854254997747,7.8682366843498732,0.72511424973378413,13.867718839196064,1.5415669913567089,8.2231132025993841,-2.0719602172099565,-2.9807499413215135,-0.029700017424724168,-1.49538809702029,-4.0999616150580858,-6.0037220482551072,-4.5130218033515428,-0.62251547619929082,-2.602203941794536,-2.9207215075217698,-3.4465278153144334,-2.4255764846526597,-1.8364111398183323,5.5640946860588576,0.85507359459767573,0.92490043595204585,-2.9287096028052781,2.2638410563744094,-2.3780553106986497,1.9507531161583447,-2.5060757760726426,-3.0293216471396898,-0.81705645367731816,-3.5242713455878709,1.6708473201073193,-6.0134342674933885,-0.047032332869670457,1.3158623214043164,0.53374495461354488,-1.6497329686604953,1.7225760455406689,-2.0619486634456132,-2.460167682620189,-2.4767314557754014,-0.95588169142832524,6.5582019324577834,-1.5551507425331093,0.99816622689137691,0.90508856728444331,-4.578585720511577,-1.6876115863048053,7.7051755423820998,-2.1542582277976488,-3.1638294224463914,0.76276006653676265,1.7256204123772168,-0.96590558335413701,0.21812918141255611,0.81969060852895015,1.2993246550835156,-2.7690359120093797,-0.52195045992960698,-6.1985953812324022,-5.4522393707953904,-6.0045140747748826,0.36688675835499995,1.9945398803032421,-5.6587966446601365,-0.23672447249521977,-0.77360591933359868,7.7634784217155959,-1.1425993208610035,0.86000146820912593,14.102769040612079,1.6227699752129101,5.5132218833244826,6.3236237998283888,-5.7864249710761522,-4.1035059456550096,-5.7021411423408006,0.032917046097614699,12.919602536705829,1.8211356635368847,1.9747168059624218,1.6612578864372753,-0.98551622674097783,-0.43927822157969243,-0.84706077620615727,1.995309495476582,-0.17021784827341802,-1.210004008265636,1.2095233435905957,-1.2878347103320575,-0.075923896285197801,-3.5511979822837327,0.39731886341939204,-2.4319189552985643,-1.6029274199460215,5.8828183646477248,-0.90420643136133916,-2.3041842941962694,-4.1021796230994676,0.76818265869984859,-3.6129146580420946,1.8331362242973828,0.58220615342030757,-0.22075519606699712,-1.4125506941997028,-3.7039509777747606,-1.0909889702521778,-4.5486248497687791,-2.2464101676665758,-2.2944592599593614,-1.0701424245559192,5.7644039626396681,5.9609567160881545,-1.5230544497095562,-6.0080412392341112,-2.0960018996917222,0.85945978119740718,-2.3848679427825425,-2.3852929835044359,-1.2938050095759845,-2.5872140292846177,6.2325612540520217,-2.725369430037639,6.7264682288445021,0.74243941262135738,-5.9626792435370897,-0.34774611041178471,7.760236882714131,-1.5613256756984211,-3.6349941258155321,-2.7129506830893968,-5.3461154465400194,-0.88560632512202031,5.8739134307182814,13.091517591027118,-5.4552021984778856,-5.2357944016181444,-0.051195240470073289,-2.795471764060161,-3.7475901608191942,-5.3541742329322313,1.5592292781151318,14.124385022667743,0.92099657013783687,-5.9309929375373338,-1.8707214061938739,0.34433999016652339,-3.5398657088004564,-2.6374054913245653,-0.68165115162958867,-0.033729291411540574,-3.2923772101127122,0.82927694275746577,-2.0743452672683214,2.2710942263878371,-6.2902215485297654,0.85356774285207027,-2.6883530382834886,13.065776013878681,-0.1466028456412769,6.2342778678215529,-3.0614408020698045,0.087876700905659133,-0.64205781266321904,6.4425155158318068,-3.3567857508384202,-3.28290674731364,-5.7464555268012498,1.5302916999138378,-1.6161186498426414,13.055070066002704,7.2258855338371779,-5.741875982733867,5.6420852179802443,-0.77510706231226689,0.97310390427479976,1.6963035102165722,-2.76105616137614,0.6018637652672314,-5.3982077603064988,-2.8094945912085985,1.7340845103539013,-5.574590301962993,1.0269502635277294,-6.1415923599921678,-2.5073739056312059,0.49953951790700191,6.346141480950215,13.984902524498798,8.2041141028679387,-0.45673975989451177,0.64778389885792964,-5.7709916596137498,-1.6346645046793438,-1.9544703428470112,-2.7096215252601121,1.4452535147942089,-3.6984351639472459,-3.6063886646949266,1.5760840411461376,1.020019435433247,-1.9226898317061878,-0.75482670113672978,1.1235598559654736,-2.9445527558051561,-3.8851340775214647,-0.71859035536875493,0.11748006298909419,1.9998451705254101,1.2762717242516064,-1.4615644578658558,0.76141562416920894,8.1171027655876653,0.61749591782460445,-5.2558022503577684,-6.5778945450507615,6.2461506362236525,8.1936055655754583,-5.8738502030097459,-3.2451840405189012,-4.0240405564033006,-1.9176188175403095,-4.8967364792548631,-0.75900510594477422,-2.211523986312053,2.0122407908714797,0.38249304249654048,1.3448041911400341,0.94322242691884273,-2.771185851546428,-5.1102541451179002,5.9348875518120314,7.9864179129875685,1.7196539874352001,12.886026525048115,13.828011655358173,-5.5758126740180467,1.4615990157402539,7.8245603080070998,-2.4903728728019212,-2.9299942021094774,1.2017818923271679,-2.6313164000236009,1.4677342410362744,-2.6873072151862596,5.7427016730583693,-3.0359129671775316,13.158868932274677,-2.718275285216472,-6.1047423844062303,0.62035121872792476,1.4338039393700146,5.5427724356926467,-0.89973375842203862,5.89402356102834,1.3726572032249951,0.96991743996510738,6.042240285424092,7.1344920630730178,-1.4178576682530857,7.622806691674092,-2.2838003997527574,1.4816068644798779,-4.0721248154365037,8.0155221457756536,-1.0759830836974598,1.0564993853844189,-2.0796260599814866,-3.325568175765178,-1.1940345530234791,-1.6627955575310207,1.5287431712425732,-1.6470925387763,0.34109439804921382,0.80305662110219234,1.3434080119408154,-2.4508759741507982,5.9842645163811232,-0.1171073679648853,-0.47933993384470708,0.10643806412587398,-4.9751158718787645,-5.7199419502936815,-0.84866932675471074,0.74418702080617183,-2.1972703699790452,5.9391943450249221,-4.1489666466437791,0.31588723614583247,-2.6371506218635057,14.124465131310322,-3.1986919407569383,-6.0602863793097947,2.0375126834190871,-0.87153980776896245,-6.1051505570136522,-2.5512099628173326,8.0844193930901067,-1.6432413299404598,-2.8638033632956956,-6.2842148308478807,1.5306264395988964],"mode":"markers","hoverinfo":["text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text","text"],"text":["AAACCTGAGCATGGCA-1<br>ident: 4<br>PC_1: 13.31943891<br>nFeature_RNA: 1177","AAACCTGAGGCATGGT-1<br>ident: 6<br>PC_1: -4.49230905<br>nFeature_RNA: 1707","AAACCTGAGTGACATA-1<br>ident: 3<br>PC_1: -3.25450890<br>nFeature_RNA: 2394","AAAGTAGAGGATATAC-1<br>ident: 3<br>PC_1: -3.26156664<br>nFeature_RNA: 2411","AAATGCCAGCGTGAGT-1<br>ident: 2<br>PC_1: -5.60064367<br>nFeature_RNA: 2049","AACACGTCAAGTTCTG-1<br>ident: 3<br>PC_1: -5.01268402<br>nFeature_RNA: 1998","AACACGTGTTGATTGC-1<br>ident: 0<br>PC_1: -4.50268352<br>nFeature_RNA: 1829","AACACGTTCCACGTTC-1<br>ident: 5<br>PC_1:  5.01030071<br>nFeature_RNA: 1458","AACCATGCACAGAGGT-1<br>ident: 0<br>PC_1: -5.08790126<br>nFeature_RNA: 1803","AACCATGGTCTCCATC-1<br>ident: 6<br>PC_1: -5.52068094<br>nFeature_RNA:  807","AACCGCGGTAAGTGGC-1<br>ident: 3<br>PC_1: -3.28279760<br>nFeature_RNA: 2000","AACGTTGAGATCCGAG-1<br>ident: 0<br>PC_1: -4.75549078<br>nFeature_RNA: 2133","AACTCAGAGGCGCTCT-1<br>ident: 6<br>PC_1: -4.66429070<br>nFeature_RNA: 1443","AACTCCCCACTGTTAG-1<br>ident: 2<br>PC_1: -5.04751054<br>nFeature_RNA: 1604","AACTCCCGTCAGCTAT-1<br>ident: 0<br>PC_1: -4.79111088<br>nFeature_RNA: 2421","AACTCCCTCATAAAGG-1<br>ident: 2<br>PC_1: -4.99127429<br>nFeature_RNA: 1221","AACTCTTCACTGTTAG-1<br>ident: 0<br>PC_1: -5.28782457<br>nFeature_RNA: 1775","AACTCTTCATGTAAGA-1<br>ident: 1<br>PC_1: -4.52568321<br>nFeature_RNA: 1090","AACTGGTAGCGTGAGT-1<br>ident: 2<br>PC_1: -2.98964394<br>nFeature_RNA: 2072","AACTGGTCACGTTGGC-1<br>ident: 0<br>PC_1: -5.25819139<br>nFeature_RNA: 1657","AACTGGTGTTTAGCTG-1<br>ident: 6<br>PC_1: -3.65329660<br>nFeature_RNA: 1422","AAGACCTAGAACAACT-1<br>ident: 0<br>PC_1: -4.24232881<br>nFeature_RNA: 2196","AAGACCTGTCCGAATT-1<br>ident: 4<br>PC_1: 24.13138655<br>nFeature_RNA: 2023","AAGCCGCGTTGTTTGG-1<br>ident: 1<br>PC_1: -5.06512408<br>nFeature_RNA: 1523","AAGGAGCCAGCTTAAC-1<br>ident: 5<br>PC_1:  8.22519626<br>nFeature_RNA: 1925","AAGGCAGAGAATGTTG-1<br>ident: 0<br>PC_1: -4.94451435<br>nFeature_RNA: 1961","AAGGCAGAGCACGCCT-1<br>ident: 4<br>PC_1: 24.20924841<br>nFeature_RNA: 2426","AAGGCAGTCCGCAAGC-1<br>ident: 4<br>PC_1: 21.99200494<br>nFeature_RNA: 1912","AAGGTTCAGGCTCATT-1<br>ident: 3<br>PC_1: -3.57953834<br>nFeature_RNA: 2336","AAGGTTCCATGTTCCC-1<br>ident: 0<br>PC_1: -4.91863354<br>nFeature_RNA: 2141","AAGGTTCTCGTAGATC-1<br>ident: 5<br>PC_1:  8.16381571<br>nFeature_RNA: 1728","AAGTCTGAGACCACGA-1<br>ident: 7<br>PC_1:  4.85919808<br>nFeature_RNA:  985","AAGTCTGCACCGAAAG-1<br>ident: 2<br>PC_1: -2.91346009<br>nFeature_RNA: 1555","AAGTCTGTCGTGACAT-1<br>ident: 5<br>PC_1:  7.21688068<br>nFeature_RNA: 2243","AATCCAGAGTTCCACA-1<br>ident: 2<br>PC_1: -1.05750973<br>nFeature_RNA: 2436","AATCGGTTCAATAAGG-1<br>ident: 1<br>PC_1: -5.70406228<br>nFeature_RNA: 1005","AATCGGTTCGGCGCAT-1<br>ident: 4<br>PC_1: 15.57552584<br>nFeature_RNA:  994","ACACCAACATCACAAC-1<br>ident: 1<br>PC_1: -5.71075929<br>nFeature_RNA: 1696","ACACCGGAGATGTTAG-1<br>ident: 2<br>PC_1: -3.05628495<br>nFeature_RNA: 1549","ACACCGGGTGTCTGAT-1<br>ident: 2<br>PC_1: -4.20880211<br>nFeature_RNA: 2350","ACACCGGTCATCGCTC-1<br>ident: 1<br>PC_1: -5.35489790<br>nFeature_RNA: 1933","ACACTGAAGTTCGCAT-1<br>ident: 0<br>PC_1: -5.30666993<br>nFeature_RNA: 2391","ACACTGACAGCGTTCG-1<br>ident: 5<br>PC_1: 13.20233524<br>nFeature_RNA: 2470","ACACTGAGTTTGTGTG-1<br>ident: 2<br>PC_1: -3.32062550<br>nFeature_RNA: 1939","ACAGCTAAGAGTTGGC-1<br>ident: 1<br>PC_1: -5.66980918<br>nFeature_RNA: 1250","ACAGCTACATGAACCT-1<br>ident: 0<br>PC_1: -4.69346387<br>nFeature_RNA: 2465","ACAGCTAGTCCTCTTG-1<br>ident: 3<br>PC_1: -3.97647320<br>nFeature_RNA: 2391","ACATACGCATACTACG-1<br>ident: 1<br>PC_1: -7.68101600<br>nFeature_RNA:  998","ACATACGTCTAAGCCA-1<br>ident: 1<br>PC_1: -5.57096621<br>nFeature_RNA: 1416","ACATACGTCTCTGAGA-1<br>ident: 3<br>PC_1: -4.46001738<br>nFeature_RNA: 2425","ACATCAGAGTCCGTAT-1<br>ident: 6<br>PC_1: -4.98304686<br>nFeature_RNA: 1709","ACATGGTGTTGGAGGT-1<br>ident: 5<br>PC_1:  7.72556612<br>nFeature_RNA: 2190","ACCCACTCAGGGTTAG-1<br>ident: 1<br>PC_1: -6.37467440<br>nFeature_RNA: 1429","ACCCACTGTCTAGGTT-1<br>ident: 0<br>PC_1: -4.43861093<br>nFeature_RNA: 2461","ACCGTAAAGTACTTGC-1<br>ident: 3<br>PC_1: -3.79866619<br>nFeature_RNA: 2122","ACCGTAAGTGCCTGGT-1<br>ident: 4<br>PC_1: 21.92487468<br>nFeature_RNA: 1783","ACCTTTATCATGCTCC-1<br>ident: 2<br>PC_1: -4.63843866<br>nFeature_RNA: 1980","ACCTTTATCCTCAATT-1<br>ident: 0<br>PC_1: -5.06146504<br>nFeature_RNA: 2262","ACGAGCCAGACATAAC-1<br>ident: 1<br>PC_1: -7.33676802<br>nFeature_RNA: 1361","ACGAGCCTCTTAGAGC-1<br>ident: 1<br>PC_1: -3.63581075<br>nFeature_RNA: 1767","ACGAGCCTCTTATCTG-1<br>ident: 0<br>PC_1: -5.16692926<br>nFeature_RNA: 1422","ACGAGGACAAGAAGAG-1<br>ident: 5<br>PC_1:  5.70676181<br>nFeature_RNA: 1999","ACGATACAGCGCTTAT-1<br>ident: 5<br>PC_1:  6.79514060<br>nFeature_RNA: 2373","ACGATACCACTTAACG-1<br>ident: 5<br>PC_1:  9.92041878<br>nFeature_RNA: 2128","ACGATACCAGCTTAAC-1<br>ident: 4<br>PC_1: 17.41015518<br>nFeature_RNA: 1720","ACGATACGTGTTTGTG-1<br>ident: 4<br>PC_1: 23.17130831<br>nFeature_RNA: 2028","ACGATGTCAGCCAATT-1<br>ident: 6<br>PC_1: -5.34925944<br>nFeature_RNA: 1412","ACGATGTCAGGCGATA-1<br>ident: 2<br>PC_1: -4.59659827<br>nFeature_RNA: 1535","ACGCAGCAGCTAGGCA-1<br>ident: 1<br>PC_1: -4.69635103<br>nFeature_RNA: 2023","ACGCAGCCATTACGAC-1<br>ident: 0<br>PC_1: -6.33359095<br>nFeature_RNA: 1905","ACGCCAGAGATGTGTA-1<br>ident: 2<br>PC_1: -0.99599038<br>nFeature_RNA: 1899","ACGCCAGAGTACGACG-1<br>ident: 0<br>PC_1: -4.76429602<br>nFeature_RNA: 2407","ACGCCAGCAAGTCATC-1<br>ident: 6<br>PC_1: -4.59132716<br>nFeature_RNA: 1568","ACGCCAGGTTGCCTCT-1<br>ident: 2<br>PC_1: -2.82127504<br>nFeature_RNA: 2175","ACGCCGAAGAGTACAT-1<br>ident: 1<br>PC_1: -7.27719873<br>nFeature_RNA: 1171","ACGCCGAAGGACTGGT-1<br>ident: 3<br>PC_1: -4.80650088<br>nFeature_RNA: 2101","ACGGAGAAGTTCGCAT-1<br>ident: 1<br>PC_1: -4.88425886<br>nFeature_RNA: 1781","ACGGAGACAAGTCATC-1<br>ident: 4<br>PC_1: 20.05958119<br>nFeature_RNA: 1994","ACGGAGAGTGACGCCT-1<br>ident: 0<br>PC_1: -3.73465919<br>nFeature_RNA: 2322","ACGGCCAAGCACGCCT-1<br>ident: 1<br>PC_1: -5.38522648<br>nFeature_RNA: 2162","ACGGCCACAGTGACAG-1<br>ident: 6<br>PC_1: -5.49619943<br>nFeature_RNA:  838","ACGGCCATCTCTGAGA-1<br>ident: 6<br>PC_1: -0.11810016<br>nFeature_RNA: 2130","ACGGGCTGTATTACCG-1<br>ident: 0<br>PC_1: -5.09958749<br>nFeature_RNA: 2294","ACGGGTCAGCACACAG-1<br>ident: 3<br>PC_1: -4.59654817<br>nFeature_RNA: 2396","ACGGGTCGTCTCACCT-1<br>ident: 2<br>PC_1: -5.27517855<br>nFeature_RNA: 2145","ACGGGTCTCATACGGT-1<br>ident: 4<br>PC_1: 13.16664088<br>nFeature_RNA: 1304","ACGTCAAGTCCTCTTG-1<br>ident: 1<br>PC_1: -5.09936786<br>nFeature_RNA: 1639","ACTATCTGTATAGTAG-1<br>ident: 3<br>PC_1: -4.64450467<br>nFeature_RNA: 2460","ACTATCTGTTTGGGCC-1<br>ident: 1<br>PC_1: -6.26785592<br>nFeature_RNA: 2042","ACTATCTTCAAGAAGT-1<br>ident: 4<br>PC_1: 19.62397179<br>nFeature_RNA: 1455","ACTGAACGTTATTCTC-1<br>ident: 0<br>PC_1: -3.89306639<br>nFeature_RNA: 2240","ACTGAGTGTTAGATGA-1<br>ident: 4<br>PC_1: 16.08469419<br>nFeature_RNA: 1577","ACTGATGAGGCGACAT-1<br>ident: 0<br>PC_1: -3.15359308<br>nFeature_RNA: 2493","ACTGATGCAATGAAAC-1<br>ident: 0<br>PC_1: -4.90355935<br>nFeature_RNA: 2448","ACTGATGGTCTTGCGG-1<br>ident: 3<br>PC_1: -3.23028696<br>nFeature_RNA: 2259","ACTGCTCAGATGTGGC-1<br>ident: 7<br>PC_1:  4.28145568<br>nFeature_RNA: 1117","ACTGCTCAGCGGCTTC-1<br>ident: 0<br>PC_1: -5.37580392<br>nFeature_RNA: 1308","ACTGCTCGTTACCGAT-1<br>ident: 1<br>PC_1: -6.23843686<br>nFeature_RNA: 1568","ACTTACTCAAAGCAAT-1<br>ident: 1<br>PC_1: -4.60776334<br>nFeature_RNA: 1875","ACTTACTCAGGACGTA-1<br>ident: 0<br>PC_1: -4.69379490<br>nFeature_RNA: 2331","ACTTACTTCCAGAAGG-1<br>ident: 2<br>PC_1: -4.80122247<br>nFeature_RNA: 1912","ACTTGTTCATCCAACA-1<br>ident: 1<br>PC_1: -6.45771225<br>nFeature_RNA: 1186","ACTTTCAGTTCCACGG-1<br>ident: 0<br>PC_1: -5.67202052<br>nFeature_RNA: 1366","AGAATAGCATTCACTT-1<br>ident: 0<br>PC_1: -6.25198791<br>nFeature_RNA: 2193","AGACGTTCATGCATGT-1<br>ident: 0<br>PC_1: -5.17766851<br>nFeature_RNA: 2264","AGAGCTTCATACAGCT-1<br>ident: 2<br>PC_1: -1.73025939<br>nFeature_RNA: 1724","AGAGCTTCATACTCTT-1<br>ident: 4<br>PC_1: 22.23321142<br>nFeature_RNA: 1854","AGAGCTTTCAAACCGT-1<br>ident: 2<br>PC_1: -4.25118142<br>nFeature_RNA: 2498","AGAGTGGAGTCCAGGA-1<br>ident: 2<br>PC_1: -1.57533030<br>nFeature_RNA: 2138","AGAGTGGGTGTAATGA-1<br>ident: 5<br>PC_1:  6.10541569<br>nFeature_RNA: 2018","AGAGTGGGTTATCCGA-1<br>ident: 0<br>PC_1: -5.27245201<br>nFeature_RNA: 2184","AGATCTGAGGGATGGG-1<br>ident: 1<br>PC_1: -5.09861979<br>nFeature_RNA: 1211","AGATCTGCATTAGGCT-1<br>ident: 1<br>PC_1: -5.59221648<br>nFeature_RNA: 1126","AGATCTGTCAGGATCT-1<br>ident: 6<br>PC_1: -4.06199253<br>nFeature_RNA: 2096","AGATTGCAGTGTGAAT-1<br>ident: 1<br>PC_1: -4.90414682<br>nFeature_RNA: 2107","AGATTGCGTCGGCATC-1<br>ident: 2<br>PC_1: -3.60494437<br>nFeature_RNA: 2417","AGCAGCCCAGGCTGAA-1<br>ident: 3<br>PC_1: -3.45894713<br>nFeature_RNA: 2042","AGCAGCCTCCACGTTC-1<br>ident: 6<br>PC_1: -3.64866026<br>nFeature_RNA: 2008","AGCCTAACACCGAAAG-1<br>ident: 2<br>PC_1: -4.88184186<br>nFeature_RNA: 2154","AGCCTAACAGAGTGTG-1<br>ident: 7<br>PC_1:  6.62676534<br>nFeature_RNA: 1255","AGCCTAACATGGGAAC-1<br>ident: 3<br>PC_1: -4.09938848<br>nFeature_RNA: 2352","AGCGGTCGTGCATCTA-1<br>ident: 3<br>PC_1: -3.47072168<br>nFeature_RNA: 1717","AGCGTATAGCCACGCT-1<br>ident: 2<br>PC_1: -2.71477016<br>nFeature_RNA: 1754","AGCGTATTCTGATTCT-1<br>ident: 1<br>PC_1: -7.44537730<br>nFeature_RNA: 1173","AGCGTCGGTGTGGCTC-1<br>ident: 3<br>PC_1: -4.10853186<br>nFeature_RNA: 1404","AGCGTCGTCAACGAAA-1<br>ident: 0<br>PC_1: -3.40981378<br>nFeature_RNA: 2393","AGCTCCTAGAGACTAT-1<br>ident: 1<br>PC_1: -4.84670786<br>nFeature_RNA: 2414","AGCTCCTAGGAGTTGC-1<br>ident: 1<br>PC_1: -6.45712446<br>nFeature_RNA: 2354","AGCTCCTTCGCCATAA-1<br>ident: 2<br>PC_1: -1.80204256<br>nFeature_RNA: 2028","AGCTCCTTCTGGCGAC-1<br>ident: 3<br>PC_1: -2.48741645<br>nFeature_RNA: 2188","AGCTCCTTCTGGCGTG-1<br>ident: 0<br>PC_1: -4.10529726<br>nFeature_RNA: 2447","AGCTCTCCATTCGACA-1<br>ident: 5<br>PC_1:  8.02365482<br>nFeature_RNA: 2252","AGCTTGAAGGGCTTCC-1<br>ident: 4<br>PC_1: 15.26538660<br>nFeature_RNA: 1333","AGCTTGACACCCATGG-1<br>ident: 5<br>PC_1:  7.30756656<br>nFeature_RNA: 2347","AGCTTGATCACCATAG-1<br>ident: 2<br>PC_1: -4.78338505<br>nFeature_RNA: 2333","AGCTTGATCTACTCAT-1<br>ident: 3<br>PC_1: -4.76328821<br>nFeature_RNA: 1795","AGGCCACAGCCACGTC-1<br>ident: 0<br>PC_1: -5.09529684<br>nFeature_RNA: 2007","AGGCCACTCCCGGATG-1<br>ident: 2<br>PC_1: -2.20954565<br>nFeature_RNA: 1456","AGGCCACTCTGTTTGT-1<br>ident: 4<br>PC_1: 27.93500590<br>nFeature_RNA: 2338","AGGCCGTTCGAATGGG-1<br>ident: 4<br>PC_1: 26.40513127<br>nFeature_RNA: 2241","AGGCCGTTCGACCAGC-1<br>ident: 3<br>PC_1: -2.98937424<br>nFeature_RNA: 2285","AGGGAGTAGCGAAGGG-1<br>ident: 5<br>PC_1:  7.66461283<br>nFeature_RNA: 2089","AGGGATGCAAGTACCT-1<br>ident: 0<br>PC_1: -3.90319107<br>nFeature_RNA: 2079","AGGGATGCAGTAACGG-1<br>ident: 6<br>PC_1: -5.54242856<br>nFeature_RNA: 1015","AGGGTGACAGACTCGC-1<br>ident: 4<br>PC_1: 15.67513640<br>nFeature_RNA: 1403","AGGTCATAGTCTCCTC-1<br>ident: 6<br>PC_1: -5.15227019<br>nFeature_RNA: 1448","AGGTCATTCCAAGTAC-1<br>ident: 3<br>PC_1: -3.36655149<br>nFeature_RNA: 2039","AGGTCCGAGCAGCCTC-1<br>ident: 0<br>PC_1: -4.71047766<br>nFeature_RNA: 2461","AGGTCCGAGCGATGAC-1<br>ident: 5<br>PC_1: 15.84576681<br>nFeature_RNA: 2218","AGGTCCGAGGTGCTAG-1<br>ident: 1<br>PC_1: -4.49115593<br>nFeature_RNA: 2467","AGTAGTCCACACCGCA-1<br>ident: 0<br>PC_1: -5.21974266<br>nFeature_RNA: 1204","AGTAGTCTCCGTAGTA-1<br>ident: 4<br>PC_1: 23.34450172<br>nFeature_RNA: 1861","AGTCTTTAGTCATCCA-1<br>ident: 6<br>PC_1: -5.75343413<br>nFeature_RNA: 1149","AGTCTTTCAATCCAAC-1<br>ident: 4<br>PC_1: 15.87388885<br>nFeature_RNA: 1505","AGTCTTTTCCGATATG-1<br>ident: 3<br>PC_1: -4.52428663<br>nFeature_RNA: 2341","AGTCTTTTCGCCGTGA-1<br>ident: 1<br>PC_1: -5.59091967<br>nFeature_RNA: 1308","AGTGAGGAGCTAACTC-1<br>ident: 6<br>PC_1: -4.33532855<br>nFeature_RNA: 1414","AGTGAGGTCTATCCTA-1<br>ident: 0<br>PC_1: -6.31588947<br>nFeature_RNA: 1411","AGTGTCACAATGGAAT-1<br>ident: 2<br>PC_1: -3.94111893<br>nFeature_RNA: 2219","AGTGTCATCCGTAGTA-1<br>ident: 0<br>PC_1: -6.51166260<br>nFeature_RNA: 1929","AGTTGGTTCCAGTATG-1<br>ident: 0<br>PC_1: -4.44516367<br>nFeature_RNA: 2341","ATAACGCCACATCCAA-1<br>ident: 6<br>PC_1: -2.73548136<br>nFeature_RNA: 1465","ATAACGCTCTTGCATT-1<br>ident: 0<br>PC_1: -3.75633422<br>nFeature_RNA: 2171","ATAAGAGGTAGGCTGA-1<br>ident: 0<br>PC_1: -6.19007760<br>nFeature_RNA: 1276","ATAGACCGTTATGTGC-1<br>ident: 1<br>PC_1: -5.89130134<br>nFeature_RNA: 1777","ATCATCTGTTACTGAC-1<br>ident: 1<br>PC_1: -0.10063943<br>nFeature_RNA: 1734","ATCATCTTCCCGACTT-1<br>ident: 6<br>PC_1: -4.38138382<br>nFeature_RNA: 1709","ATCATGGCATCGGGTC-1<br>ident: 0<br>PC_1: -3.82855366<br>nFeature_RNA: 2351","ATCATGGTCCTGCTTG-1<br>ident: 5<br>PC_1: 10.05470365<br>nFeature_RNA: 2153","ATCCACCGTGAAGGCT-1<br>ident: 2<br>PC_1: -4.65296342<br>nFeature_RNA: 2195","ATCCACCGTGCCTGTG-1<br>ident: 0<br>PC_1: -6.26924348<br>nFeature_RNA: 1913","ATCCACCTCAAACGGG-1<br>ident: 0<br>PC_1: -4.49720904<br>nFeature_RNA: 1973","ATCCGAAGTTCGAATC-1<br>ident: 1<br>PC_1: -6.32971158<br>nFeature_RNA: 2289","ATCGAGTAGAGTCGGT-1<br>ident: 0<br>PC_1: -4.62310555<br>nFeature_RNA: 1813","ATCGAGTCACAGCCCA-1<br>ident: 5<br>PC_1: 11.42111165<br>nFeature_RNA: 2328","ATCGAGTCACGTCTCT-1<br>ident: 4<br>PC_1: 15.86416327<br>nFeature_RNA: 1505","ATCGAGTTCGGTTAAC-1<br>ident: 7<br>PC_1:  3.73293175<br>nFeature_RNA: 1053","ATCTACTGTTAGATGA-1<br>ident: 1<br>PC_1: -4.96853415<br>nFeature_RNA: 1379","ATCTGCCCAGCTCGAC-1<br>ident: 3<br>PC_1: -3.00082726<br>nFeature_RNA: 2087","ATCTGCCTCGGTGTCG-1<br>ident: 6<br>PC_1: -4.90247814<br>nFeature_RNA: 1373","ATGAGGGGTCGCTTCT-1<br>ident: 2<br>PC_1: -3.83983896<br>nFeature_RNA: 2450","ATGAGGGTCACATAGC-1<br>ident: 1<br>PC_1: -4.42431730<br>nFeature_RNA: 2142","ATGAGGGTCTACTCAT-1<br>ident: 1<br>PC_1: -4.93211809<br>nFeature_RNA: 2397","ATGCGATGTGGTCTCG-1<br>ident: 5<br>PC_1:  5.18428251<br>nFeature_RNA: 1924","ATGGGAGGTATAGGGC-1<br>ident: 3<br>PC_1: -3.29007681<br>nFeature_RNA: 2119","ATGGGAGGTCCAGTAT-1<br>ident: 6<br>PC_1: -2.53206018<br>nFeature_RNA: 1896","ATGGGAGGTGATGTGG-1<br>ident: 0<br>PC_1: -5.35783043<br>nFeature_RNA: 1595","ATGTGTGCACTCGACG-1<br>ident: 5<br>PC_1:  9.29475174<br>nFeature_RNA: 2396","ATGTGTGGTTTGTTTC-1<br>ident: 6<br>PC_1: -4.05749340<br>nFeature_RNA: 1599","ATTACTCCAATGGACG-1<br>ident: 2<br>PC_1: -4.22478975<br>nFeature_RNA: 1772","ATTACTCTCTTACCGC-1<br>ident: 3<br>PC_1: -4.76544667<br>nFeature_RNA: 2168","ATTATCCCACTATCTT-1<br>ident: 0<br>PC_1: -4.19833342<br>nFeature_RNA: 2258","ATTATCCCAGCTGCTG-1<br>ident: 0<br>PC_1: -4.98158965<br>nFeature_RNA: 1755","ATTATCCTCTTGCAAG-1<br>ident: 0<br>PC_1: -4.06450418<br>nFeature_RNA: 2495","ATTCTACAGGCGACAT-1<br>ident: 7<br>PC_1:  6.87481542<br>nFeature_RNA: 1453","ATTCTACTCATATCGG-1<br>ident: 4<br>PC_1: 24.17001627<br>nFeature_RNA: 1895","ATTGGACCACACCGCA-1<br>ident: 1<br>PC_1: -4.99142870<br>nFeature_RNA: 1162","ATTGGACGTAGCGCAA-1<br>ident: 2<br>PC_1: -4.28266889<br>nFeature_RNA: 2311","ATTGGACGTATAAACG-1<br>ident: 1<br>PC_1: -6.12612742<br>nFeature_RNA: 1107","ATTGGACTCATCGCTC-1<br>ident: 6<br>PC_1: -5.36631868<br>nFeature_RNA: 1353","ATTGGTGGTCAACTGT-1<br>ident: 0<br>PC_1: -4.86831269<br>nFeature_RNA: 1472","ATTTCTGCACCACGTG-1<br>ident: 5<br>PC_1:  7.20320769<br>nFeature_RNA: 1879","ATTTCTGGTGCCTGTG-1<br>ident: 4<br>PC_1: 22.33985737<br>nFeature_RNA: 2232","CAACCAAAGTCGTACT-1<br>ident: 4<br>PC_1: 21.57888916<br>nFeature_RNA: 1632","CAACCAATCTGATTCT-1<br>ident: 1<br>PC_1: -4.65065973<br>nFeature_RNA: 1959","CAACCTCAGTTGCAGG-1<br>ident: 1<br>PC_1: -5.90858041<br>nFeature_RNA: 1354","CAACCTCCAGACGTAG-1<br>ident: 4<br>PC_1: 19.56794770<br>nFeature_RNA: 1840","CAACCTCGTGTAACGG-1<br>ident: 4<br>PC_1: 19.90367399<br>nFeature_RNA: 2120","CAACTAGTCTAACGGT-1<br>ident: 0<br>PC_1: -5.74957188<br>nFeature_RNA: 1680","CAAGAAACATACCATG-1<br>ident: 0<br>PC_1: -3.89472744<br>nFeature_RNA: 1393","CAAGATCAGAGGTTAT-1<br>ident: 1<br>PC_1: -6.56771960<br>nFeature_RNA: 1234","CAAGGCCCAACCGCCA-1<br>ident: 7<br>PC_1:  7.53498317<br>nFeature_RNA: 1334","CAAGGCCCACAGGAGT-1<br>ident: 2<br>PC_1: -2.44641384<br>nFeature_RNA: 1458","CAAGTTGCAAATACAG-1<br>ident: 0<br>PC_1: -5.01713154<br>nFeature_RNA: 2264","CAAGTTGTCCTGCTTG-1<br>ident: 0<br>PC_1: -4.94093153<br>nFeature_RNA: 2419","CACAAACTCGCGATCG-1<br>ident: 0<br>PC_1: -6.01311042<br>nFeature_RNA: 2392","CACAAACTCTTACCGC-1<br>ident: 5<br>PC_1:  7.80113189<br>nFeature_RNA: 2104","CACACAAGTCTCGTTC-1<br>ident: 2<br>PC_1: -4.64680341<br>nFeature_RNA: 2141","CACACAAGTTAGGGTG-1<br>ident: 5<br>PC_1:  8.16206298<br>nFeature_RNA: 2279","CACACAATCAATCTCT-1<br>ident: 7<br>PC_1: 10.48994770<br>nFeature_RNA: 1393","CACACCTGTGTTTGTG-1<br>ident: 0<br>PC_1: -5.89532195<br>nFeature_RNA: 1260","CACACTCCAATCACAC-1<br>ident: 5<br>PC_1:  6.54039152<br>nFeature_RNA: 2339","CACACTCGTACTCAAC-1<br>ident: 7<br>PC_1:  7.05019428<br>nFeature_RNA: 1367","CACAGGCAGTGCAAGC-1<br>ident: 6<br>PC_1: -5.45190212<br>nFeature_RNA: 1145","CACAGGCTCCAAGCCG-1<br>ident: 5<br>PC_1:  3.80623443<br>nFeature_RNA: 1605","CACAGTATCTTGCCGT-1<br>ident: 3<br>PC_1: -3.77070080<br>nFeature_RNA: 2335","CACATTTCATTCACTT-1<br>ident: 1<br>PC_1: -4.59143455<br>nFeature_RNA: 1177","CACCACTAGGATGCGT-1<br>ident: 1<br>PC_1: -5.79842480<br>nFeature_RNA: 1691","CACCAGGCAAACCCAT-1<br>ident: 0<br>PC_1: -4.46202828<br>nFeature_RNA: 2322","CACCAGGCATCGGACC-1<br>ident: 1<br>PC_1: -4.56232387<br>nFeature_RNA: 1625","CACCAGGGTACTTCTT-1<br>ident: 4<br>PC_1: 20.81199962<br>nFeature_RNA: 2245","CACCAGGTCAACCATG-1<br>ident: 1<br>PC_1: -6.16733218<br>nFeature_RNA: 1349","CACCAGGTCGGGAGTA-1<br>ident: 5<br>PC_1:  7.51174645<br>nFeature_RNA: 2075","CACCTTGCAGATGAGC-1<br>ident: 2<br>PC_1: -4.45919992<br>nFeature_RNA: 2254","CACCTTGTCTCCAGGG-1<br>ident: 6<br>PC_1: -4.15709451<br>nFeature_RNA: 1640","CACCTTGTCTGAAAGA-1<br>ident: 1<br>PC_1: -4.67957919<br>nFeature_RNA: 2245","CAGAATCAGAACTCGG-1<br>ident: 6<br>PC_1: -2.11573393<br>nFeature_RNA: 1925","CAGAATCTCACATGCA-1<br>ident: 7<br>PC_1:  5.21681596<br>nFeature_RNA: 1139","CAGAATCTCGGAGGTA-1<br>ident: 0<br>PC_1: -6.18806583<br>nFeature_RNA: 1759","CAGAGAGCACCTCGTT-1<br>ident: 3<br>PC_1: -3.77706120<br>nFeature_RNA: 2491","CAGAGAGGTGTGGCTC-1<br>ident: 5<br>PC_1: 12.00387381<br>nFeature_RNA: 2388","CAGATCAAGCTCCTTC-1<br>ident: 7<br>PC_1:  9.68838102<br>nFeature_RNA: 1398","CAGATCAGTTCCTCCA-1<br>ident: 2<br>PC_1: -2.95623863<br>nFeature_RNA: 2131","CAGATCATCTGTCTCG-1<br>ident: 3<br>PC_1: -3.91629280<br>nFeature_RNA: 2495","CAGCAGCAGTAGATGT-1<br>ident: 1<br>PC_1: -6.79829363<br>nFeature_RNA: 1302","CAGCAGCGTCCCTTGT-1<br>ident: 3<br>PC_1: -5.06607001<br>nFeature_RNA: 2406","CAGCCGACACATGACT-1<br>ident: 1<br>PC_1: -3.47838546<br>nFeature_RNA: 2468","CAGCGACGTTTCGCTC-1<br>ident: 4<br>PC_1: 22.57125835<br>nFeature_RNA: 2031","CAGCGACTCTTGTCAT-1<br>ident: 0<br>PC_1: -4.21429262<br>nFeature_RNA: 2114","CAGCGACTCTTGTTTG-1<br>ident: 1<br>PC_1: -4.29414991<br>nFeature_RNA: 1578","CAGCTAAGTCAATGTC-1<br>ident: 1<br>PC_1: -4.72593504<br>nFeature_RNA: 2356","CAGCTAAGTCTAACGT-1<br>ident: 5<br>PC_1: 10.34223257<br>nFeature_RNA: 2496","CAGGTGCAGGAGCGTT-1<br>ident: 1<br>PC_1: -5.96320883<br>nFeature_RNA: 1335","CAGGTGCCATCCCATC-1<br>ident: 2<br>PC_1: -2.96516465<br>nFeature_RNA: 2382","CAGGTGCTCAGCTCGG-1<br>ident: 2<br>PC_1: -3.17657230<br>nFeature_RNA: 1704","CAGTAACCAATCTACG-1<br>ident: 3<br>PC_1: -4.36597682<br>nFeature_RNA: 2397","CATATGGAGGACAGAA-1<br>ident: 4<br>PC_1: 21.18635909<br>nFeature_RNA: 2169","CATATGGGTGAGGGTT-1<br>ident: 6<br>PC_1: -4.73225228<br>nFeature_RNA: 1418","CATATTCCAAGAAAGG-1<br>ident: 6<br>PC_1: -5.11596777<br>nFeature_RNA: 1170","CATCAAGGTCCGCTGA-1<br>ident: 5<br>PC_1:  6.59410178<br>nFeature_RNA: 2139","CATCAAGGTGCACTTA-1<br>ident: 3<br>PC_1: -3.41866182<br>nFeature_RNA: 2313","CATCAAGTCTTCTGGC-1<br>ident: 1<br>PC_1: -5.64173093<br>nFeature_RNA: 1357","CATCAGAAGAGGTTGC-1<br>ident: 0<br>PC_1: -4.75900681<br>nFeature_RNA: 2383","CATCAGAAGATGGCGT-1<br>ident: 3<br>PC_1: -4.57549836<br>nFeature_RNA: 1083","CATCAGACATGCCCGA-1<br>ident: 2<br>PC_1: -4.36294156<br>nFeature_RNA: 1767","CATCAGATCAGTGCAT-1<br>ident: 3<br>PC_1: -3.62094543<br>nFeature_RNA: 2139","CATCCACCAGTAAGAT-1<br>ident: 6<br>PC_1: -5.76824835<br>nFeature_RNA: 1055","CATCGAACACAGACTT-1<br>ident: 1<br>PC_1: -5.53861453<br>nFeature_RNA: 1474","CATCGAATCGACAGCC-1<br>ident: 3<br>PC_1: -4.24569297<br>nFeature_RNA: 1617","CATCGGGGTGACTCAT-1<br>ident: 0<br>PC_1: -4.98479307<br>nFeature_RNA: 2219","CATGACAAGGGCACTA-1<br>ident: 2<br>PC_1: -4.06291147<br>nFeature_RNA: 2279","CATGACATCCAAGTAC-1<br>ident: 1<br>PC_1: -5.09747482<br>nFeature_RNA: 1231","CATGCCTCACACGCTG-1<br>ident: 3<br>PC_1: -3.55333832<br>nFeature_RNA: 1986","CATGCCTCACGCATCG-1<br>ident: 0<br>PC_1: -6.18067163<br>nFeature_RNA: 1604","CATGCCTGTCATGCAT-1<br>ident: 0<br>PC_1: -4.40686927<br>nFeature_RNA: 2452","CATGGCGAGATGTGGC-1<br>ident: 4<br>PC_1: 22.84078842<br>nFeature_RNA: 1779","CATTATCAGGAGCGAG-1<br>ident: 4<br>PC_1: 21.13987284<br>nFeature_RNA: 1950","CATTATCAGGCATGTG-1<br>ident: 0<br>PC_1: -4.00066048<br>nFeature_RNA: 2469","CATTATCTCACGACTA-1<br>ident: 2<br>PC_1: -5.78439577<br>nFeature_RNA: 2055","CATTCGCAGTTAGCGG-1<br>ident: 1<br>PC_1: -5.93758035<br>nFeature_RNA: 1235","CCAATCCGTAAGTTCC-1<br>ident: 4<br>PC_1: 19.95002061<br>nFeature_RNA: 1824","CCAATCCGTACGAAAT-1<br>ident: 2<br>PC_1: -3.29076398<br>nFeature_RNA: 1428","CCACCTATCCGATATG-1<br>ident: 3<br>PC_1: -4.79554699<br>nFeature_RNA: 1339","CCACGGACATATGAGA-1<br>ident: 0<br>PC_1: -4.45710376<br>nFeature_RNA: 2135","CCACGGAGTAACGACG-1<br>ident: 0<br>PC_1: -5.23257124<br>nFeature_RNA: 2345","CCACGGAGTCAATGTC-1<br>ident: 3<br>PC_1: -4.63995420<br>nFeature_RNA: 2321","CCACGGATCCTAGTGA-1<br>ident: 1<br>PC_1: -5.50520206<br>nFeature_RNA: 2202","CCAGCGAAGAAAGTGG-1<br>ident: 4<br>PC_1: 20.23939516<br>nFeature_RNA: 1969","CCAGCGAGTTACCAGT-1<br>ident: 2<br>PC_1: -2.31746335<br>nFeature_RNA: 1297","CCATGTCCACTCTGTC-1<br>ident: 6<br>PC_1: -4.37997941<br>nFeature_RNA: 1009","CCATTCGAGTAGATGT-1<br>ident: 0<br>PC_1: -5.57010877<br>nFeature_RNA: 1850","CCATTCGCAGGATCGA-1<br>ident: 5<br>PC_1:  5.86808333<br>nFeature_RNA: 2132","CCCAGTTTCTAGCACA-1<br>ident: 3<br>PC_1: -5.67854079<br>nFeature_RNA: 1513","CCCATACCACAGCGTC-1<br>ident: 2<br>PC_1: -2.86346728<br>nFeature_RNA: 2428","CCCATACGTACAAGTA-1<br>ident: 2<br>PC_1: -4.73947276<br>nFeature_RNA: 2002","CCCATACGTCGCGTGT-1<br>ident: 3<br>PC_1: -6.65932608<br>nFeature_RNA: 1331","CCCATACGTCTTCAAG-1<br>ident: 4<br>PC_1: 13.84848270<br>nFeature_RNA: 1511","CCCTCCTCAGCCTTGG-1<br>ident: 2<br>PC_1: -2.52344250<br>nFeature_RNA: 2315","CCCTCCTCATGGTCAT-1<br>ident: 3<br>PC_1: -4.87135900<br>nFeature_RNA: 1547","CCGGGATCAGCGATCC-1<br>ident: 5<br>PC_1:  5.97193462<br>nFeature_RNA: 1551","CCGGTAGGTACAGTTC-1<br>ident: 1<br>PC_1: -5.41077861<br>nFeature_RNA: 1560","CCGGTAGGTCTTGATG-1<br>ident: 0<br>PC_1: -4.29695812<br>nFeature_RNA: 2155","CCGTACTCAACTGCGC-1<br>ident: 6<br>PC_1: -5.31878709<br>nFeature_RNA: 1565","CCGTACTGTCCGAATT-1<br>ident: 2<br>PC_1: -4.23283647<br>nFeature_RNA: 1679","CCGTACTGTTTAGGAA-1<br>ident: 5<br>PC_1: 10.09271595<br>nFeature_RNA: 2331","CCGTACTTCATCTGCC-1<br>ident: 1<br>PC_1: -7.17917621<br>nFeature_RNA: 1162","CCGTGGACATAGAAAC-1<br>ident: 4<br>PC_1: 17.56737981<br>nFeature_RNA: 1525","CCGTTCAGTAGGCATG-1<br>ident: 2<br>PC_1: -2.72916801<br>nFeature_RNA: 1580","CCGTTCATCGTTTGCC-1<br>ident: 1<br>PC_1: -4.85015692<br>nFeature_RNA: 1988","CCGTTCATCTGTCTAT-1<br>ident: 2<br>PC_1: -6.40097184<br>nFeature_RNA: 1575","CCTAAAGAGTTCGCGC-1<br>ident: 1<br>PC_1: -5.67072188<br>nFeature_RNA: 2208","CCTAAAGGTTTGACAC-1<br>ident: 2<br>PC_1: -4.92998572<br>nFeature_RNA: 2440","CCTACACAGAAACGAG-1<br>ident: 4<br>PC_1: 18.60080214<br>nFeature_RNA: 1576","CCTACACCAATGGTCT-1<br>ident: 0<br>PC_1: -5.59235767<br>nFeature_RNA: 2237","CCTACACGTCAACTGT-1<br>ident: 0<br>PC_1: -4.65993253<br>nFeature_RNA: 1467","CCTACACGTGAGGGAG-1<br>ident: 2<br>PC_1: -2.78407604<br>nFeature_RNA: 1903","CCTACACGTGCGCTTG-1<br>ident: 6<br>PC_1: -4.70619875<br>nFeature_RNA: 1181","CCTACCAAGCCCTAAT-1<br>ident: 1<br>PC_1: -5.86790519<br>nFeature_RNA: 1852","CCTACCAGTCAGTGGA-1<br>ident: 1<br>PC_1: -5.81465093<br>nFeature_RNA: 1281","CCTACCATCCTCATTA-1<br>ident: 0<br>PC_1: -5.38213680<br>nFeature_RNA: 1252","CCTAGCTAGTGGAGTC-1<br>ident: 3<br>PC_1: -2.48785615<br>nFeature_RNA: 1850","CCTAGCTCATCCCATC-1<br>ident: 3<br>PC_1: -3.98152106<br>nFeature_RNA: 2445","CCTCAGTTCTGCGTAA-1<br>ident: 0<br>PC_1: -5.37539224<br>nFeature_RNA: 1749","CCTCTGAGTCATCCCT-1<br>ident: 1<br>PC_1: -5.40263373<br>nFeature_RNA: 2493","CCTTACGAGTGCGTGA-1<br>ident: 0<br>PC_1: -5.79885259<br>nFeature_RNA: 1824","CCTTACGGTAGGCATG-1<br>ident: 1<br>PC_1: -5.93041932<br>nFeature_RNA: 2254","CCTTCGAGTTAAGAAC-1<br>ident: 5<br>PC_1: 13.99431493<br>nFeature_RNA: 2431","CCTTTCTGTCACTGGC-1<br>ident: 6<br>PC_1: -6.27220675<br>nFeature_RNA: 1372","CCTTTCTGTCGTTGTA-1<br>ident: 1<br>PC_1: -4.45029432<br>nFeature_RNA: 1359","CCTTTCTGTCTTGTCC-1<br>ident: 5<br>PC_1:  8.94687975<br>nFeature_RNA: 2205","CGAACATTCTAGAGTC-1<br>ident: 0<br>PC_1: -4.38327489<br>nFeature_RNA: 2277","CGAATGTAGCTTATCG-1<br>ident: 1<br>PC_1: -5.85378380<br>nFeature_RNA: 2206","CGAATGTCAGACGTAG-1<br>ident: 7<br>PC_1:  4.28507689<br>nFeature_RNA: 1007","CGACCTTGTTCTGTTT-1<br>ident: 3<br>PC_1: -3.96693205<br>nFeature_RNA: 2431","CGACTTCAGGGCATGT-1<br>ident: 5<br>PC_1:  7.27187923<br>nFeature_RNA: 1968","CGACTTCCAAACGCGA-1<br>ident: 4<br>PC_1: 24.12074602<br>nFeature_RNA: 2497","CGAGAAGAGCGATTCT-1<br>ident: 6<br>PC_1: -4.62601007<br>nFeature_RNA: 1419","CGAGAAGAGTCGAGTG-1<br>ident: 0<br>PC_1: -4.85305482<br>nFeature_RNA: 2091","CGAGAAGGTGCAGTAG-1<br>ident: 0<br>PC_1: -3.76166337<br>nFeature_RNA: 2431","CGAGAAGGTTCTGGTA-1<br>ident: 2<br>PC_1: -5.71630203<br>nFeature_RNA: 2163","CGAGCCAAGTAACCCT-1<br>ident: 1<br>PC_1: -5.54042291<br>nFeature_RNA: 1472","CGAGCCACATGCATGT-1<br>ident: 5<br>PC_1:  2.18398960<br>nFeature_RNA: 1726","CGATCGGCACTACAGT-1<br>ident: 2<br>PC_1: -1.63231938<br>nFeature_RNA: 1788","CGATGTAAGGATATAC-1<br>ident: 3<br>PC_1: -5.64549119<br>nFeature_RNA: 1737","CGATGTACACGGTAAG-1<br>ident: 3<br>PC_1: -3.47042406<br>nFeature_RNA: 1794","CGCCAAGTCGTGGGAA-1<br>ident: 4<br>PC_1: 21.57048982<br>nFeature_RNA: 1913","CGCTGGACAGTATAAG-1<br>ident: 6<br>PC_1: -4.63508375<br>nFeature_RNA: 1287","CGCTTCAGTGGTGTAG-1<br>ident: 4<br>PC_1: 17.73250004<br>nFeature_RNA: 1396","CGGACACTCGCATGGC-1<br>ident: 3<br>PC_1: -3.59463939<br>nFeature_RNA: 2303","CGGACGTCATACTCTT-1<br>ident: 6<br>PC_1: -5.02751502<br>nFeature_RNA: 1367","CGGACGTTCTTACCTA-1<br>ident: 1<br>PC_1: -5.86561317<br>nFeature_RNA: 1974","CGGACTGAGGACGAAA-1<br>ident: 3<br>PC_1: -4.85858828<br>nFeature_RNA: 1912","CGGACTGAGTCGCCGT-1<br>ident: 0<br>PC_1: -6.13977326<br>nFeature_RNA: 2041","CGGACTGTCTTGCCGT-1<br>ident: 3<br>PC_1: -4.69790340<br>nFeature_RNA: 2016","CGGAGCTAGAAGCCCA-1<br>ident: 0<br>PC_1: -5.58761787<br>nFeature_RNA: 2271","CGGAGCTTCAATACCG-1<br>ident: 5<br>PC_1:  4.08480799<br>nFeature_RNA: 1740","CGGAGCTTCAATCTCT-1<br>ident: 2<br>PC_1: -3.41298816<br>nFeature_RNA: 2412","CGGAGTCAGACTAGAT-1<br>ident: 0<br>PC_1: -4.09500452<br>nFeature_RNA: 2062","CGGAGTCGTCCAGTAT-1<br>ident: 2<br>PC_1: -2.78664236<br>nFeature_RNA: 2235","CGGAGTCGTCGTTGTA-1<br>ident: 2<br>PC_1: -5.06242444<br>nFeature_RNA: 1905","CGGAGTCTCCAAGCCG-1<br>ident: 0<br>PC_1: -6.44252675<br>nFeature_RNA: 1306","CGGCTAGGTGCCTGGT-1<br>ident: 0<br>PC_1: -5.26472607<br>nFeature_RNA: 2281","CGGCTAGTCTTCAACT-1<br>ident: 0<br>PC_1: -5.64259649<br>nFeature_RNA: 1721","CGGGTCACAAGGTTTC-1<br>ident: 1<br>PC_1: -4.64494865<br>nFeature_RNA: 2317","CGGGTCATCAGTTTGG-1<br>ident: 7<br>PC_1:  7.09256641<br>nFeature_RNA: 1425","CGGGTCATCTAACTCT-1<br>ident: 2<br>PC_1: -4.05694294<br>nFeature_RNA: 2292","CGGTTAACATTCCTGC-1<br>ident: 7<br>PC_1: 12.15535142<br>nFeature_RNA: 1532","CGGTTAATCCTTGGTC-1<br>ident: 4<br>PC_1: 16.35506551<br>nFeature_RNA: 1461","CGGTTAATCTCTAGGA-1<br>ident: 4<br>PC_1: 18.57052518<br>nFeature_RNA: 1596","CGTAGGCTCCATGAAC-1<br>ident: 1<br>PC_1: -2.89191980<br>nFeature_RNA: 2452","CGTCAGGGTCGGCTCA-1<br>ident: 0<br>PC_1: -4.44093050<br>nFeature_RNA: 2346","CGTCTACGTTCGCGAC-1<br>ident: 4<br>PC_1: 23.99356282<br>nFeature_RNA: 2218","CGTCTACTCTCTAAGG-1<br>ident: 6<br>PC_1: -5.00212423<br>nFeature_RNA: 1375","CGTGAGCAGTTGAGTA-1<br>ident: 2<br>PC_1: -3.48518253<br>nFeature_RNA: 2144","CGTGAGCCACAAGTAA-1<br>ident: 5<br>PC_1:  9.86472058<br>nFeature_RNA: 2260","CGTGAGCCACAGATTC-1<br>ident: 3<br>PC_1: -4.16634434<br>nFeature_RNA: 2467","CGTGTAACAGGGATTG-1<br>ident: 3<br>PC_1: -3.56191026<br>nFeature_RNA: 1846","CGTGTAACATAGAAAC-1<br>ident: 2<br>PC_1: -4.36264644<br>nFeature_RNA: 2116","CGTGTAACATCGGAAG-1<br>ident: 2<br>PC_1: -3.80770928<br>nFeature_RNA: 2445","CGTTAGAAGGGCACTA-1<br>ident: 0<br>PC_1: -4.43719107<br>nFeature_RNA: 2409","CGTTCTGAGACAAAGG-1<br>ident: 1<br>PC_1: -6.56587806<br>nFeature_RNA: 1100","CGTTCTGAGGGTATCG-1<br>ident: 4<br>PC_1: 22.21396922<br>nFeature_RNA: 2447","CGTTGGGAGGGATCTG-1<br>ident: 5<br>PC_1: 14.26584352<br>nFeature_RNA: 2392","CTAACTTCACAGACAG-1<br>ident: 7<br>PC_1:  8.56723773<br>nFeature_RNA: 1474","CTAACTTTCGTAGATC-1<br>ident: 1<br>PC_1: -5.01768597<br>nFeature_RNA: 1998","CTAAGACTCAGGTTCA-1<br>ident: 0<br>PC_1: -3.32669789<br>nFeature_RNA: 2451","CTAATGGAGAGCAATT-1<br>ident: 2<br>PC_1: -3.34087277<br>nFeature_RNA: 1844","CTACACCCAATCTGCA-1<br>ident: 1<br>PC_1: -6.04977953<br>nFeature_RNA: 1132","CTACACCGTATATGAG-1<br>ident: 6<br>PC_1: -5.26543167<br>nFeature_RNA: 1073","CTACATTAGTGCGATG-1<br>ident: 6<br>PC_1: -4.70284465<br>nFeature_RNA: 1363","CTACATTTCCTATGTT-1<br>ident: 2<br>PC_1: -0.90927827<br>nFeature_RNA: 2255","CTACGTCAGGATATAC-1<br>ident: 0<br>PC_1: -5.29249248<br>nFeature_RNA: 2269","CTACGTCCAGTGGGAT-1<br>ident: 0<br>PC_1: -4.94894142<br>nFeature_RNA: 2039","CTAGAGTAGACACTAA-1<br>ident: 2<br>PC_1: -4.44342377<br>nFeature_RNA: 1815","CTAGAGTCACGGCGTT-1<br>ident: 0<br>PC_1: -4.37097938<br>nFeature_RNA: 2444","CTAGCCTCAAGCCTAT-1<br>ident: 0<br>PC_1: -3.69879118<br>nFeature_RNA: 2394","CTAGTGACAGACGCAA-1<br>ident: 1<br>PC_1: -6.22619980<br>nFeature_RNA: 1476","CTAGTGAGTTGTCGCG-1<br>ident: 0<br>PC_1: -4.97368467<br>nFeature_RNA: 1750","CTCGGAGAGAGTGAGA-1<br>ident: 1<br>PC_1: -5.66525968<br>nFeature_RNA: 1700","CTCGGAGGTTCCCGAG-1<br>ident: 5<br>PC_1:  8.34825872<br>nFeature_RNA: 1909","CTCGTCAAGGGAAACA-1<br>ident: 2<br>PC_1: -5.53163777<br>nFeature_RNA: 1744","CTCGTCATCGTTGACA-1<br>ident: 0<br>PC_1: -3.36670754<br>nFeature_RNA: 2430","CTCTAATAGTCCCACG-1<br>ident: 6<br>PC_1: -4.45619401<br>nFeature_RNA: 1240","CTCTACGGTGTGACCC-1<br>ident: 2<br>PC_1: -4.87854735<br>nFeature_RNA: 2037","CTCTACGGTTATGCGT-1<br>ident: 4<br>PC_1: 19.59008915<br>nFeature_RNA: 1828","CTCTGGTAGCATCATC-1<br>ident: 1<br>PC_1: -4.31188466<br>nFeature_RNA: 2429","CTCTGGTAGCTGCAAG-1<br>ident: 2<br>PC_1: -5.20798917<br>nFeature_RNA: 1997","CTGAAGTGTAAGCACG-1<br>ident: 4<br>PC_1: 18.54380738<br>nFeature_RNA: 1474","CTGCCTAAGCAGATCG-1<br>ident: 6<br>PC_1: -4.82647478<br>nFeature_RNA: 1794","CTGCCTACACGGTAGA-1<br>ident: 3<br>PC_1: -4.17347248<br>nFeature_RNA: 2198","CTGCCTAGTTGCCTCT-1<br>ident: 5<br>PC_1:  7.69935368<br>nFeature_RNA: 1884","CTGCGGAGTAGCGCAA-1<br>ident: 3<br>PC_1: -4.66472245<br>nFeature_RNA: 2026","CTGCGGAGTGAGGGAG-1<br>ident: 0<br>PC_1: -5.53248545<br>nFeature_RNA: 2490","CTGCTGTCAGGCGATA-1<br>ident: 2<br>PC_1: -3.08287373<br>nFeature_RNA: 1499","CTGTGCTTCCCTGACT-1<br>ident: 5<br>PC_1:  7.83003454<br>nFeature_RNA: 2117","CTGTTTAAGAACAATC-1<br>ident: 0<br>PC_1: -5.41177532<br>nFeature_RNA: 1426","CTTAACTCATGAAGTA-1<br>ident: 1<br>PC_1: -6.80955417<br>nFeature_RNA: 1087","CTTAACTGTAACGTTC-1<br>ident: 5<br>PC_1:  8.54026982<br>nFeature_RNA: 1875","CTTAACTTCAACGCTA-1<br>ident: 6<br>PC_1: -5.04941644<br>nFeature_RNA: 1255","CTTACCGGTAAGTAGT-1<br>ident: 1<br>PC_1: -5.86060052<br>nFeature_RNA: 1894","CTTAGGACAATCTGCA-1<br>ident: 4<br>PC_1: 22.09648862<br>nFeature_RNA: 1605","CTTAGGACACACGCTG-1<br>ident: 5<br>PC_1:  9.41306431<br>nFeature_RNA: 2428","CTTAGGATCCCAAGAT-1<br>ident: 4<br>PC_1: 23.55709574<br>nFeature_RNA: 2222","CTTCTCTTCAAACCGT-1<br>ident: 5<br>PC_1:  5.59393841<br>nFeature_RNA: 1787","CTTGGCTAGGTAGCCA-1<br>ident: 1<br>PC_1: -5.01887638<br>nFeature_RNA: 1586","GAAACTCAGATAGGAG-1<br>ident: 3<br>PC_1: -4.48626151<br>nFeature_RNA: 1804","GAAACTCTCAATAAGG-1<br>ident: 3<br>PC_1: -4.59769577<br>nFeature_RNA: 2394","GAACATCAGACCTTTG-1<br>ident: 4<br>PC_1: 18.19882367<br>nFeature_RNA: 1421","GAACATCAGAGCTGCA-1<br>ident: 2<br>PC_1: -4.96379827<br>nFeature_RNA: 2370","GAACATCTCAAACCAC-1<br>ident: 1<br>PC_1: -4.51162950<br>nFeature_RNA: 1965","GAACCTAAGTGGTCCC-1<br>ident: 1<br>PC_1: -3.95659514<br>nFeature_RNA: 2476","GAACCTAGTCGCTTCT-1<br>ident: 1<br>PC_1: -4.63352255<br>nFeature_RNA: 1544","GAACGGACAGATCGGA-1<br>ident: 0<br>PC_1: -5.15907345<br>nFeature_RNA: 2355","GAACGGAGTCACTGGC-1<br>ident: 4<br>PC_1: 10.47449253<br>nFeature_RNA: 1275","GAACGGAGTCTCATCC-1<br>ident: 5<br>PC_1: 10.22865047<br>nFeature_RNA: 2315","GAATAAGGTGTGACGA-1<br>ident: 1<br>PC_1: -4.89819642<br>nFeature_RNA: 1255","GAATAAGTCGCAAACT-1<br>ident: 7<br>PC_1: 10.20437793<br>nFeature_RNA: 1894","GAATGAAGTAGTACCT-1<br>ident: 2<br>PC_1: -4.79224565<br>nFeature_RNA: 1929","GAATGAAGTCCCTACT-1<br>ident: 5<br>PC_1:  8.28436809<br>nFeature_RNA: 1700","GACACGCTCCCAAGAT-1<br>ident: 0<br>PC_1: -4.44030540<br>nFeature_RNA: 2290","GACAGAGAGTAGCCGA-1<br>ident: 7<br>PC_1:  6.07328074<br>nFeature_RNA:  995","GACAGAGAGTGGAGTC-1<br>ident: 1<br>PC_1: -4.89049067<br>nFeature_RNA: 2058","GACCAATCATTTGCCC-1<br>ident: 4<br>PC_1: 13.62119066<br>nFeature_RNA: 1160","GACCTGGAGGAATGGA-1<br>ident: 2<br>PC_1: -5.23968744<br>nFeature_RNA: 1319","GACGCGTAGTGTTTGC-1<br>ident: 4<br>PC_1: 20.69648549<br>nFeature_RNA: 1813","GACTACAAGCGCCTTG-1<br>ident: 3<br>PC_1: -3.53290615<br>nFeature_RNA: 1909","GACTACAAGTGTTAGA-1<br>ident: 5<br>PC_1:  7.79942253<br>nFeature_RNA: 2193","GACTACACATCTGGTA-1<br>ident: 4<br>PC_1: 19.42049341<br>nFeature_RNA: 2015","GACTGCGAGCGCTTAT-1<br>ident: 6<br>PC_1: -4.30417209<br>nFeature_RNA: 1173","GACTGCGAGTGTGGCA-1<br>ident: 4<br>PC_1: 16.29105460<br>nFeature_RNA: 2276","GACTGCGCAGTATGCT-1<br>ident: 2<br>PC_1: -4.33712023<br>nFeature_RNA: 1937","GACTGCGTCCAAAGTC-1<br>ident: 1<br>PC_1: -5.15284779<br>nFeature_RNA: 1906","GACTGCGTCTGCAAGT-1<br>ident: 0<br>PC_1: -6.23875399<br>nFeature_RNA: 1277","GAGCAGAAGATATACG-1<br>ident: 3<br>PC_1: -1.34113317<br>nFeature_RNA: 1856","GAGCAGAAGTCATCCA-1<br>ident: 0<br>PC_1: -4.90567552<br>nFeature_RNA: 1665","GAGCAGACAGCCAATT-1<br>ident: 3<br>PC_1: -3.68524222<br>nFeature_RNA: 2086","GAGCAGATCACATACG-1<br>ident: 3<br>PC_1: -3.80666532<br>nFeature_RNA: 2479","GAGGTGAAGCTAGTTC-1<br>ident: 0<br>PC_1: -4.94098677<br>nFeature_RNA: 1788","GAGGTGAAGTAAGTAC-1<br>ident: 1<br>PC_1: -7.84367592<br>nFeature_RNA: 1796","GAGGTGACAATCCAAC-1<br>ident: 3<br>PC_1: -1.87061920<br>nFeature_RNA: 2458","GATCAGTCACATTTCT-1<br>ident: 4<br>PC_1: 19.90930391<br>nFeature_RNA: 1952","GATCAGTCATGACATC-1<br>ident: 0<br>PC_1: -5.04861240<br>nFeature_RNA: 1260","GATCAGTTCAGCTGGC-1<br>ident: 3<br>PC_1: -2.81710195<br>nFeature_RNA: 2258","GATCAGTTCCGCTGTT-1<br>ident: 6<br>PC_1: -1.72137861<br>nFeature_RNA: 1561","GATCGATCAGCCTGTG-1<br>ident: 1<br>PC_1: -5.26495682<br>nFeature_RNA: 1464","GATCGCGGTGCAGTAG-1<br>ident: 2<br>PC_1: -2.81697769<br>nFeature_RNA: 1890","GATCGTAAGGCCGAAT-1<br>ident: 4<br>PC_1: 16.29826581<br>nFeature_RNA: 1445","GATCGTAAGGTGTGGT-1<br>ident: 5<br>PC_1:  4.09221692<br>nFeature_RNA: 1590","GATCGTAAGTCAAGGC-1<br>ident: 2<br>PC_1: -0.02903678<br>nFeature_RNA: 1741","GATCGTATCTGGGCCA-1<br>ident: 2<br>PC_1: -2.85067449<br>nFeature_RNA: 1918","GATCTAGAGTTCGATC-1<br>ident: 3<br>PC_1: -4.68654358<br>nFeature_RNA: 2120","GATCTAGAGTTCGCAT-1<br>ident: 5<br>PC_1:  3.33563515<br>nFeature_RNA: 1607","GATCTAGCAAGTAGTA-1<br>ident: 4<br>PC_1: 15.92379327<br>nFeature_RNA: 1488","GATCTAGCAGGGATTG-1<br>ident: 1<br>PC_1: -5.07525160<br>nFeature_RNA: 2330","GATCTAGGTCGAATCT-1<br>ident: 2<br>PC_1: -5.06339809<br>nFeature_RNA: 2137","GATCTAGGTCTTGATG-1<br>ident: 2<br>PC_1: -4.16204534<br>nFeature_RNA: 2153","GATCTAGGTTGGACCC-1<br>ident: 3<br>PC_1: -3.24107414<br>nFeature_RNA: 2010","GATGAAAGTTAAGGGC-1<br>ident: 1<br>PC_1: -5.59884617<br>nFeature_RNA: 2117","GATGAGGCACACCGAC-1<br>ident: 2<br>PC_1: -5.09826499<br>nFeature_RNA: 1421","GATGAGGGTGTGAAAT-1<br>ident: 0<br>PC_1: -4.61588504<br>nFeature_RNA: 2304","GATGCTACACAACGTT-1<br>ident: 1<br>PC_1: -5.77394155<br>nFeature_RNA: 1089","GATGCTAGTCTAGCCG-1<br>ident: 2<br>PC_1: -3.15594179<br>nFeature_RNA: 2231","GCAAACTAGGGTATCG-1<br>ident: 5<br>PC_1:  4.22363137<br>nFeature_RNA: 1599","GCAATCACAAAGCGGT-1<br>ident: 2<br>PC_1: -3.90112250<br>nFeature_RNA: 1797","GCAATCAGTCCTCCAT-1<br>ident: 0<br>PC_1: -3.25793007<br>nFeature_RNA: 2431","GCACATAAGGTGTTAA-1<br>ident: 0<br>PC_1: -5.38574954<br>nFeature_RNA: 2252","GCACATAAGTAATCCC-1<br>ident: 1<br>PC_1: -5.79495872<br>nFeature_RNA: 2241","GCACATAGTCAGAGGT-1<br>ident: 3<br>PC_1: -4.88062008<br>nFeature_RNA: 1827","GCACTCTAGCAACGGT-1<br>ident: 2<br>PC_1: -5.62105503<br>nFeature_RNA: 1788","GCACTCTCAACTGGCC-1<br>ident: 4<br>PC_1: 20.67945815<br>nFeature_RNA: 1487","GCAGCCAAGCCAGGAT-1<br>ident: 1<br>PC_1: -2.14447796<br>nFeature_RNA: 2418","GCAGCCACAATGAAAC-1<br>ident: 4<br>PC_1: 18.57795281<br>nFeature_RNA: 1625","GCAGTTAGTATGCTTG-1<br>ident: 0<br>PC_1: -4.02994242<br>nFeature_RNA: 2284","GCAGTTAGTCCGAGTC-1<br>ident: 4<br>PC_1: 22.10333472<br>nFeature_RNA: 1995","GCAGTTATCGTCGTTC-1<br>ident: 2<br>PC_1: -6.07205512<br>nFeature_RNA: 1254","GCATGATAGTGAAGTT-1<br>ident: 5<br>PC_1:  6.30920892<br>nFeature_RNA: 1840","GCATGATGTCAAGCGA-1<br>ident: 2<br>PC_1: -2.29876836<br>nFeature_RNA: 2253","GCATGTAAGGGAACGG-1<br>ident: 4<br>PC_1: 18.40442074<br>nFeature_RNA: 1699","GCATGTAGTATCGCAT-1<br>ident: 3<br>PC_1: -2.56445035<br>nFeature_RNA: 2238","GCATGTATCGTCACGG-1<br>ident: 7<br>PC_1:  4.99393249<br>nFeature_RNA:  984","GCGACCAAGCGCCTTG-1<br>ident: 1<br>PC_1: -4.20906756<br>nFeature_RNA: 1315","GCGACCACACCTGGTG-1<br>ident: 6<br>PC_1: -4.41697669<br>nFeature_RNA: 1215","GCGACCAGTCCAGTGC-1<br>ident: 2<br>PC_1: -4.71516320<br>nFeature_RNA: 1509","GCGACCATCACATGCA-1<br>ident: 3<br>PC_1: -4.07144780<br>nFeature_RNA: 2204","GCGAGAACAGACTCGC-1<br>ident: 4<br>PC_1: 17.24424568<br>nFeature_RNA: 1614","GCGAGAACATCAGTAC-1<br>ident: 4<br>PC_1: 18.98330796<br>nFeature_RNA: 1717","GCGAGAAGTTAGGGTG-1<br>ident: 6<br>PC_1: -0.53547593<br>nFeature_RNA: 2204","GCGAGAATCCAACCAA-1<br>ident: 4<br>PC_1: 18.89945504<br>nFeature_RNA: 1557","GCGCAACGTGCCTGCA-1<br>ident: 4<br>PC_1: 20.21111479<br>nFeature_RNA: 2187","GCGCAGTAGACGCTTT-1<br>ident: 4<br>PC_1: 23.19969552<br>nFeature_RNA: 1842","GCGCAGTAGTGATCGG-1<br>ident: 0<br>PC_1: -3.33207603<br>nFeature_RNA: 1971","GCGCAGTTCTTTACAC-1<br>ident: 7<br>PC_1:  9.92259452<br>nFeature_RNA: 1560","GCGCCAACAGATAATG-1<br>ident: 1<br>PC_1: -5.70167401<br>nFeature_RNA: 2241","GCGCCAAGTAAAGTCA-1<br>ident: 4<br>PC_1: 20.94057100<br>nFeature_RNA: 2117","GCGCCAAGTTCCATGA-1<br>ident: 5<br>PC_1:  4.10895657<br>nFeature_RNA: 1487","GCGCGATAGTCTCCTC-1<br>ident: 2<br>PC_1: -3.29015164<br>nFeature_RNA: 1779","GCGGGTTAGGTGCACA-1<br>ident: 0<br>PC_1: -4.31760240<br>nFeature_RNA: 2239","GCGGGTTCAAGTACCT-1<br>ident: 0<br>PC_1: -4.37442029<br>nFeature_RNA: 1840","GCGGGTTCAGGTTTCA-1<br>ident: 3<br>PC_1: -4.07190243<br>nFeature_RNA: 1543","GCTCCTAGTCTTGCGG-1<br>ident: 3<br>PC_1: -4.80746112<br>nFeature_RNA: 2106","GCTCCTATCGTGTAGT-1<br>ident: 3<br>PC_1: -5.99036230<br>nFeature_RNA: 1313","GCTCTGTCAGGCGATA-1<br>ident: 2<br>PC_1: -3.40795126<br>nFeature_RNA: 2493","GCTGCAGAGTGCGATG-1<br>ident: 5<br>PC_1:  8.10721558<br>nFeature_RNA: 2053","GCTGCAGCAATAAGCA-1<br>ident: 6<br>PC_1: -4.31678892<br>nFeature_RNA: 1375","GCTGCGAAGTTAGCGG-1<br>ident: 5<br>PC_1:  5.24809462<br>nFeature_RNA: 1876","GCTGCGACAGCCTGTG-1<br>ident: 6<br>PC_1: -4.48844193<br>nFeature_RNA: 1493","GCTGCTTAGACTCGGA-1<br>ident: 6<br>PC_1: -4.79897064<br>nFeature_RNA: 1176","GCTTGAACACGCATCG-1<br>ident: 4<br>PC_1: 19.06248427<br>nFeature_RNA: 1705","GGAAAGCTCAGGCAAG-1<br>ident: 0<br>PC_1: -4.39980564<br>nFeature_RNA: 1836","GGAACTTGTTATGCGT-1<br>ident: 1<br>PC_1: -6.23225327<br>nFeature_RNA: 1412","GGACAAGAGACCACGA-1<br>ident: 2<br>PC_1: -2.32875362<br>nFeature_RNA: 1644","GGACAAGAGATCACGG-1<br>ident: 1<br>PC_1: -5.37673561<br>nFeature_RNA: 1684","GGACAAGGTCGAGTTT-1<br>ident: 5<br>PC_1:  8.27916380<br>nFeature_RNA: 1857","GGACAGATCATGGTCA-1<br>ident: 1<br>PC_1: -5.70424866<br>nFeature_RNA: 1608","GGACAGATCCTGCAGG-1<br>ident: 2<br>PC_1: -0.58426928<br>nFeature_RNA: 2426","GGACATTTCGTGGACC-1<br>ident: 6<br>PC_1: -5.41713192<br>nFeature_RNA: 1433","GGACGTCTCGGAATCT-1<br>ident: 0<br>PC_1: -5.52498280<br>nFeature_RNA: 1449","GGAGCAACATCTCCCA-1<br>ident: 5<br>PC_1:  7.03729319<br>nFeature_RNA: 1896","GGAGCAATCCATGAGT-1<br>ident: 0<br>PC_1: -4.83546509<br>nFeature_RNA: 1905","GGAGCAATCGAATGCT-1<br>ident: 3<br>PC_1: -3.83162198<br>nFeature_RNA: 2221","GGAGCAATCGAGCCCA-1<br>ident: 0<br>PC_1: -4.18945407<br>nFeature_RNA: 2282","GGATGTTAGACCCACC-1<br>ident: 1<br>PC_1: -5.26928044<br>nFeature_RNA: 1269","GGATGTTAGGCTAGCA-1<br>ident: 1<br>PC_1: -6.63118165<br>nFeature_RNA: 1785","GGATTACCACGTCAGC-1<br>ident: 2<br>PC_1: -3.10247557<br>nFeature_RNA: 2425","GGCCGATAGCAGCCTC-1<br>ident: 1<br>PC_1: -5.52789290<br>nFeature_RNA: 1068","GGCCGATCAGCCAGAA-1<br>ident: 5<br>PC_1:  6.26995202<br>nFeature_RNA: 1798","GGCCGATGTATTCGTG-1<br>ident: 2<br>PC_1: -2.64635433<br>nFeature_RNA: 1474","GGCGACTAGTAGCCGA-1<br>ident: 2<br>PC_1: -6.20124839<br>nFeature_RNA: 1463","GGCGACTTCAACGAAA-1<br>ident: 2<br>PC_1: -3.93002018<br>nFeature_RNA: 2385","GGCGACTTCACGAAGG-1<br>ident: 4<br>PC_1: 24.38793026<br>nFeature_RNA: 2325","GGCGACTTCTCCCTGA-1<br>ident: 2<br>PC_1: -4.67853360<br>nFeature_RNA: 2497","GGCTGGTCATGGAATA-1<br>ident: 1<br>PC_1: -4.71308990<br>nFeature_RNA: 2095","GGCTGGTTCAAACGGG-1<br>ident: 1<br>PC_1: -5.78049542<br>nFeature_RNA: 1224","GGCTGGTTCATCGATG-1<br>ident: 3<br>PC_1: -3.54852405<br>nFeature_RNA: 2208","GGCTGGTTCCGTTGCT-1<br>ident: 0<br>PC_1: -2.89467753<br>nFeature_RNA: 2181","GGCTGGTTCTCAAACG-1<br>ident: 4<br>PC_1: 15.72630948<br>nFeature_RNA: 1258","GGGAATGGTGGTCCGT-1<br>ident: 6<br>PC_1: -4.11501099<br>nFeature_RNA: 1732","GGGACCTAGATGCCAG-1<br>ident: 6<br>PC_1: -4.56682015<br>nFeature_RNA: 1005","GGGACCTCACAAGCCC-1<br>ident: 2<br>PC_1: -3.90085729<br>nFeature_RNA: 2460","GGGACCTGTCATACTG-1<br>ident: 1<br>PC_1: -5.76718297<br>nFeature_RNA: 1080","GGGACCTGTGGTACAG-1<br>ident: 2<br>PC_1: -1.40341447<br>nFeature_RNA: 2172","GGGACCTTCAACGGGA-1<br>ident: 0<br>PC_1: -4.79494953<br>nFeature_RNA: 1756","GGGAGATAGCTGAAAT-1<br>ident: 1<br>PC_1: -4.28770564<br>nFeature_RNA: 2326","GGGAGATCACAGGAGT-1<br>ident: 1<br>PC_1: -5.54225362<br>nFeature_RNA: 1681","GGGAGATCACGAAGCA-1<br>ident: 2<br>PC_1: -4.11454001<br>nFeature_RNA: 1552","GGGAGATTCCAGATCA-1<br>ident: 0<br>PC_1: -4.83467057<br>nFeature_RNA: 1711","GGGCACTAGTTAAGTG-1<br>ident: 3<br>PC_1: -3.78272490<br>nFeature_RNA: 2132","GGGCACTTCTGATTCT-1<br>ident: 3<br>PC_1: -5.16075897<br>nFeature_RNA: 1831","GGGCATCAGCTAGTCT-1<br>ident: 3<br>PC_1: -2.42542446<br>nFeature_RNA: 2115","GGTATTGGTGCAACGA-1<br>ident: 0<br>PC_1: -4.72903277<br>nFeature_RNA: 2372","GGTATTGTCAGCGACC-1<br>ident: 1<br>PC_1: -4.80930225<br>nFeature_RNA: 1906","GGTGAAGGTTAAGGGC-1<br>ident: 3<br>PC_1: -5.06379681<br>nFeature_RNA: 1856","GGTGCGTAGGACGAAA-1<br>ident: 0<br>PC_1: -4.04384667<br>nFeature_RNA: 2373","GGTGCGTAGTTCCACA-1<br>ident: 0<br>PC_1: -5.34585149<br>nFeature_RNA: 1618","GGTGTTACACACCGCA-1<br>ident: 4<br>PC_1: 19.24335333<br>nFeature_RNA: 2012","GTAACGTTCTGTACGA-1<br>ident: 2<br>PC_1: -2.97181748<br>nFeature_RNA: 2219","GTAACTGCACGACTCG-1<br>ident: 1<br>PC_1: -6.31790788<br>nFeature_RNA: 1606","GTAACTGTCAGTTTGG-1<br>ident: 7<br>PC_1:  6.53597187<br>nFeature_RNA: 1332","GTACGTAGTTCTGTTT-1<br>ident: 1<br>PC_1: -4.32770404<br>nFeature_RNA: 1055","GTACTCCCAACGATCT-1<br>ident: 4<br>PC_1: 15.70696845<br>nFeature_RNA: 1658","GTACTCCCAGCGTAAG-1<br>ident: 4<br>PC_1: 24.85539823<br>nFeature_RNA: 2326","GTACTTTAGCGATATA-1<br>ident: 3<br>PC_1: -3.83373349<br>nFeature_RNA: 2146","GTACTTTGTGCGATAG-1<br>ident: 5<br>PC_1:  8.05095772<br>nFeature_RNA: 2384","GTACTTTTCTACTATC-1<br>ident: 3<br>PC_1: -4.03294703<br>nFeature_RNA: 2151","GTAGGCCAGCCTCGTG-1<br>ident: 0<br>PC_1: -4.09630801<br>nFeature_RNA: 2273","GTAGGCCTCTGCCAGG-1<br>ident: 7<br>PC_1:  4.32882006<br>nFeature_RNA:  923","GTAGTCATCGCACTCT-1<br>ident: 1<br>PC_1: -6.58623744<br>nFeature_RNA: 1560","GTAGTCATCGGTTCGG-1<br>ident: 1<br>PC_1: -4.57049247<br>nFeature_RNA: 1855","GTATCTTTCTCTGTCG-1<br>ident: 1<br>PC_1:  2.11251795<br>nFeature_RNA: 1557","GTCAAGTGTCTCACCT-1<br>ident: 0<br>PC_1: -4.42057827<br>nFeature_RNA: 1811","GTCAAGTTCGTGTAGT-1<br>ident: 0<br>PC_1: -4.91787943<br>nFeature_RNA: 1852","GTCACAAGTAGCTAAA-1<br>ident: 0<br>PC_1: -4.50887223<br>nFeature_RNA: 1944","GTCACAATCTTAGCCC-1<br>ident: 1<br>PC_1: -5.24789846<br>nFeature_RNA: 1897","GTCACGGAGTCATCCA-1<br>ident: 0<br>PC_1: -6.16710578<br>nFeature_RNA: 1483","GTCACGGCAATAACGA-1<br>ident: 0<br>PC_1: -4.79149328<br>nFeature_RNA: 2211","GTCATTTTCGAGAACG-1<br>ident: 0<br>PC_1: -5.02795006<br>nFeature_RNA: 2245","GTCATTTTCTACCTGC-1<br>ident: 0<br>PC_1: -5.27921997<br>nFeature_RNA: 2116","GTCCTCACACACTGCG-1<br>ident: 0<br>PC_1: -4.64869378<br>nFeature_RNA: 1896","GTCGGGTAGAGACTAT-1<br>ident: 2<br>PC_1: -5.20447295<br>nFeature_RNA: 1831","GTCGGGTTCTTCCTTC-1<br>ident: 0<br>PC_1: -4.64373085<br>nFeature_RNA: 2388","GTCTCGTAGTTAAGTG-1<br>ident: 6<br>PC_1: -5.18327441<br>nFeature_RNA: 1533","GTCTCGTAGTTCGCAT-1<br>ident: 0<br>PC_1: -3.67450861<br>nFeature_RNA: 2318","GTCTCGTGTCGAGTTT-1<br>ident: 4<br>PC_1: 21.48898261<br>nFeature_RNA: 1886","GTCTCGTTCCACTGGG-1<br>ident: 0<br>PC_1: -3.59960369<br>nFeature_RNA: 2477","GTGAAGGCAGTTAACC-1<br>ident: 6<br>PC_1: -5.16329494<br>nFeature_RNA:  971","GTGAAGGGTAGGGTAC-1<br>ident: 5<br>PC_1:  6.57418320<br>nFeature_RNA: 1896","GTGCAGCAGGGTGTGT-1<br>ident: 1<br>PC_1: -5.70999214<br>nFeature_RNA:  953","GTGCATATCTTGCATT-1<br>ident: 5<br>PC_1:  3.96583368<br>nFeature_RNA: 2018","GTGCGGTAGGATGGTC-1<br>ident: 1<br>PC_1: -6.10053776<br>nFeature_RNA: 2166","GTGCGGTTCATTGCGA-1<br>ident: 1<br>PC_1: -6.22962828<br>nFeature_RNA: 1101","GTGCTTCAGACTTGAA-1<br>ident: 0<br>PC_1: -5.67351211<br>nFeature_RNA: 2235","GTGCTTCAGGAATCGC-1<br>ident: 0<br>PC_1: -4.17048265<br>nFeature_RNA: 1822","GTGCTTCTCTCAACTT-1<br>ident: 2<br>PC_1: -4.88700973<br>nFeature_RNA: 1739","GTGGGTCCAAACCCAT-1<br>ident: 0<br>PC_1: -5.13026663<br>nFeature_RNA: 1437","GTGGGTCGTTGGGACA-1<br>ident: 3<br>PC_1: -3.27782670<br>nFeature_RNA: 1770","GTGTGCGGTTAAGACA-1<br>ident: 5<br>PC_1: 11.28124931<br>nFeature_RNA: 2345","GTTACAGTCTAACCGA-1<br>ident: 6<br>PC_1: -4.84179785<br>nFeature_RNA: 1258","GTTACAGTCTTCCTTC-1<br>ident: 0<br>PC_1: -4.38820667<br>nFeature_RNA: 2233","GTTCATTCAGGGCATA-1<br>ident: 4<br>PC_1: 21.70479348<br>nFeature_RNA: 1744","GTTCATTTCCCTCAGT-1<br>ident: 4<br>PC_1: 18.04154061<br>nFeature_RNA: 1566","GTTCGGGGTTCCGTCT-1<br>ident: 2<br>PC_1: -5.15351048<br>nFeature_RNA: 2244","GTTCGGGTCGCTTGTC-1<br>ident: 3<br>PC_1: -4.29987476<br>nFeature_RNA: 2332","GTTCTCGAGAATAGGG-1<br>ident: 6<br>PC_1: -4.95518515<br>nFeature_RNA: 1463","GTTCTCGGTACCGGCT-1<br>ident: 0<br>PC_1: -5.55670296<br>nFeature_RNA: 1093","GTTCTCGGTTAGAACA-1<br>ident: 2<br>PC_1: -4.10058165<br>nFeature_RNA: 2270","TAAACCGGTTAAGACA-1<br>ident: 5<br>PC_1:  6.95994373<br>nFeature_RNA: 1960","TAAGAGAAGTGGGTTG-1<br>ident: 2<br>PC_1: -4.24558169<br>nFeature_RNA: 1054","TAAGCGTAGAGGACGG-1<br>ident: 2<br>PC_1: -4.88922922<br>nFeature_RNA: 2123","TAAGCGTTCGGTGTCG-1<br>ident: 4<br>PC_1: 23.53003541<br>nFeature_RNA: 2181","TACAGTGGTAGCTTGT-1<br>ident: 2<br>PC_1: -2.53165842<br>nFeature_RNA: 2061","TACAGTGTCCGGGTGT-1<br>ident: 4<br>PC_1: 25.89386721<br>nFeature_RNA: 2206","TACCTATGTCACACGC-1<br>ident: 2<br>PC_1: -5.41231276<br>nFeature_RNA: 1916","TACCTATTCACATAGC-1<br>ident: 3<br>PC_1: -4.62568009<br>nFeature_RNA: 1760","TACCTTACAATGAAAC-1<br>ident: 0<br>PC_1: -4.41880971<br>nFeature_RNA: 2089","TACCTTATCTACTCAT-1<br>ident: 4<br>PC_1: 22.84838166<br>nFeature_RNA: 1471","TACGGATCACACCGAC-1<br>ident: 2<br>PC_1: -4.61257884<br>nFeature_RNA: 2099","TACGGATTCACTATTC-1<br>ident: 2<br>PC_1: -5.03138135<br>nFeature_RNA: 1924","TACGGATTCAGCAACT-1<br>ident: 2<br>PC_1: -3.45982867<br>nFeature_RNA: 1867","TACGGATTCTCGCATC-1<br>ident: 3<br>PC_1: -3.67345167<br>nFeature_RNA: 2356","TACGGGCCAGGGTTAG-1<br>ident: 0<br>PC_1: -4.76969429<br>nFeature_RNA: 2124","TACGGGCTCCACGCAG-1<br>ident: 4<br>PC_1: 23.51761882<br>nFeature_RNA: 2353","TACGGGCTCCCTGACT-1<br>ident: 7<br>PC_1:  4.97976761<br>nFeature_RNA: 1134","TACGGTAAGTGAACGC-1<br>ident: 3<br>PC_1: -4.00333948<br>nFeature_RNA: 2204","TACGGTACAGTGAGTG-1<br>ident: 3<br>PC_1: -2.17993512<br>nFeature_RNA: 2295","TACGGTATCACTCCTG-1<br>ident: 0<br>PC_1: -4.83787481<br>nFeature_RNA: 2371","TACTCATCAATTGCTG-1<br>ident: 2<br>PC_1: -4.25083301<br>nFeature_RNA: 1670","TACTCATGTCGCTTCT-1<br>ident: 5<br>PC_1:  9.35053012<br>nFeature_RNA: 2231","TACTTGTAGGGTCGAT-1<br>ident: 3<br>PC_1: -4.74912136<br>nFeature_RNA: 2293","TACTTGTGTCTGGTCG-1<br>ident: 1<br>PC_1: -7.18665847<br>nFeature_RNA: 1528","TACTTGTTCCTTTCTC-1<br>ident: 7<br>PC_1:  9.34710946<br>nFeature_RNA: 1202","TACTTGTTCGCGATCG-1<br>ident: 0<br>PC_1: -5.70184265<br>nFeature_RNA: 1356","TAGACCAAGATGTTAG-1<br>ident: 3<br>PC_1: -3.13064174<br>nFeature_RNA: 2386","TAGACCATCAAACCGT-1<br>ident: 2<br>PC_1: -3.71362874<br>nFeature_RNA: 2253","TAGAGCTCATTCCTGC-1<br>ident: 1<br>PC_1: -6.08486622<br>nFeature_RNA: 1262","TAGCCGGCACTCTGTC-1<br>ident: 2<br>PC_1: -3.93307411<br>nFeature_RNA: 1988","TAGGCATAGTTTGCGT-1<br>ident: 2<br>PC_1: -3.83892381<br>nFeature_RNA: 1592","TAGGCATCATTCCTCG-1<br>ident: 0<br>PC_1: -2.78948421<br>nFeature_RNA: 2458","TAGTGGTAGCTAGTCT-1<br>ident: 0<br>PC_1: -5.78944597<br>nFeature_RNA: 1305","TAGTGGTCATACGCCG-1<br>ident: 6<br>PC_1: -1.86397704<br>nFeature_RNA: 1898","TAGTTGGTCTAACTGG-1<br>ident: 0<br>PC_1: -4.77391240<br>nFeature_RNA: 2102","TATCTCAAGCTAACAA-1<br>ident: 6<br>PC_1: -5.20816754<br>nFeature_RNA: 1205","TATCTCAGTGACCAAG-1<br>ident: 1<br>PC_1: -3.44498730<br>nFeature_RNA: 2376","TATTACCCAAGTTGTC-1<br>ident: 3<br>PC_1: -4.03262101<br>nFeature_RNA: 2421","TCAACGAAGAGTTGGC-1<br>ident: 1<br>PC_1: -5.50666241<br>nFeature_RNA: 1178","TCAACGAGTCATGCAT-1<br>ident: 2<br>PC_1: -2.54421222<br>nFeature_RNA: 1781","TCAACGAGTTATGCGT-1<br>ident: 7<br>PC_1:  7.87373591<br>nFeature_RNA: 1243","TCACAAGAGACCACGA-1<br>ident: 0<br>PC_1: -5.32099311<br>nFeature_RNA: 1769","TCACAAGAGTGAAGAG-1<br>ident: 4<br>PC_1: 22.08024005<br>nFeature_RNA: 2230","TCACAAGCAACTGCTA-1<br>ident: 5<br>PC_1:  5.81280550<br>nFeature_RNA: 1871","TCACAAGTCTGTCTAT-1<br>ident: 1<br>PC_1:  0.56425438<br>nFeature_RNA: 1997","TCACGAAAGGTGACCA-1<br>ident: 6<br>PC_1: -4.83311738<br>nFeature_RNA: 1108","TCACGAATCTTCGAGA-1<br>ident: 4<br>PC_1: 21.98069821<br>nFeature_RNA: 1877","TCAGATGAGCGATAGC-1<br>ident: 2<br>PC_1: -3.04079212<br>nFeature_RNA: 2356","TCAGATGAGTGGTAAT-1<br>ident: 6<br>PC_1: -4.22953380<br>nFeature_RNA: 1665","TCAGATGTCCGTACAA-1<br>ident: 3<br>PC_1: -5.64576880<br>nFeature_RNA: 1855","TCAGCAACAGTCACTA-1<br>ident: 0<br>PC_1: -4.88535929<br>nFeature_RNA: 1878","TCAGCAAGTCAACATC-1<br>ident: 2<br>PC_1: -2.79966813<br>nFeature_RNA: 1914","TCAGCTCGTACCGCTG-1<br>ident: 7<br>PC_1:  4.15459713<br>nFeature_RNA:  953","TCAGCTCGTAGCTTGT-1<br>ident: 4<br>PC_1: 15.60732963<br>nFeature_RNA: 1321","TCAGGATCAGGAACGT-1<br>ident: 3<br>PC_1: -3.25790635<br>nFeature_RNA: 2361","TCAGGTACATGTCGAT-1<br>ident: 4<br>PC_1: 19.87660641<br>nFeature_RNA: 1646","TCATTTGAGCGATATA-1<br>ident: 0<br>PC_1: -3.16113641<br>nFeature_RNA: 2332","TCCACACGTATGAAAC-1<br>ident: 1<br>PC_1: -5.40448870<br>nFeature_RNA: 1387","TCCCGATCACAGTCGC-1<br>ident: 1<br>PC_1: -7.64535746<br>nFeature_RNA: 1068","TCGAGGCAGTAGTGCG-1<br>ident: 2<br>PC_1: -3.15152826<br>nFeature_RNA: 1386","TCGAGGCGTTCATGGT-1<br>ident: 1<br>PC_1: -6.55119310<br>nFeature_RNA:  914","TCGCGAGTCTGCTGTC-1<br>ident: 3<br>PC_1: -2.90782544<br>nFeature_RNA: 2292","TCGGTAACAAGAAGAG-1<br>ident: 2<br>PC_1: -3.45901474<br>nFeature_RNA: 2006","TCGGTAAGTAGAGGAA-1<br>ident: 0<br>PC_1: -4.83302238<br>nFeature_RNA: 1335","TCGTACCTCCAACCAA-1<br>ident: 3<br>PC_1: -4.61883359<br>nFeature_RNA: 2224","TCGTAGAAGCATCATC-1<br>ident: 0<br>PC_1: -4.11549554<br>nFeature_RNA: 2407","TCGTAGAAGTACGTTC-1<br>ident: 3<br>PC_1: -3.33421679<br>nFeature_RNA: 2460","TCGTAGAAGTTACGGG-1<br>ident: 5<br>PC_1:  5.17973795<br>nFeature_RNA: 1630","TCGTAGAGTTACGTCA-1<br>ident: 0<br>PC_1: -5.85066170<br>nFeature_RNA: 1325","TCGTAGATCGTTGCCT-1<br>ident: 4<br>PC_1: 21.61358301<br>nFeature_RNA: 2041","TCGTAGATCTTCATGT-1<br>ident: 7<br>PC_1: 10.20902242<br>nFeature_RNA: 1745","TCTCATAGTATCTGCA-1<br>ident: 4<br>PC_1: 18.70150723<br>nFeature_RNA: 1917","TCTCTAAAGGGCTCTC-1<br>ident: 2<br>PC_1: -4.70180018<br>nFeature_RNA: 2053","TCTCTAAGTCGTCTTC-1<br>ident: 0<br>PC_1: -4.70324240<br>nFeature_RNA: 2374","TCTCTAAGTGGTTTCA-1<br>ident: 3<br>PC_1: -3.00561623<br>nFeature_RNA: 2487","TCTCTAATCGGATGGA-1<br>ident: 2<br>PC_1: -4.18802047<br>nFeature_RNA: 2340","TCTGAGACACAGGTTT-1<br>ident: 5<br>PC_1:  3.24128884<br>nFeature_RNA: 1568","TCTGAGAGTCCGAAGA-1<br>ident: 2<br>PC_1: -0.54120119<br>nFeature_RNA: 2335","TCTGAGATCTGAGTGT-1<br>ident: 0<br>PC_1: -6.27950194<br>nFeature_RNA: 1550","TCTGGAAAGCAACGGT-1<br>ident: 2<br>PC_1: -4.04780349<br>nFeature_RNA: 2099","TCTGGAACACTTACGA-1<br>ident: 2<br>PC_1: -3.90693750<br>nFeature_RNA: 2358","TCTTCGGAGAAACGCC-1<br>ident: 0<br>PC_1: -4.94088886<br>nFeature_RNA: 2326","TCTTCGGGTCAAGCGA-1<br>ident: 1<br>PC_1: -6.64915838<br>nFeature_RNA:  959","TCTTTCCAGATCCGAG-1<br>ident: 2<br>PC_1: -3.64974208<br>nFeature_RNA: 1654","TGAAAGAGTTCCACAA-1<br>ident: 6<br>PC_1: -3.42919435<br>nFeature_RNA: 1678","TGACGGCGTTGATTGC-1<br>ident: 0<br>PC_1: -4.62774156<br>nFeature_RNA: 2489","TGACTAGAGGGTCGAT-1<br>ident: 6<br>PC_1: -3.97694192<br>nFeature_RNA: 1499","TGACTTTAGGCATGGT-1<br>ident: 2<br>PC_1: -4.03225057<br>nFeature_RNA: 2276","TGACTTTTCAATCTCT-1<br>ident: 0<br>PC_1: -5.21709192<br>nFeature_RNA: 1981","TGACTTTTCGTCACGG-1<br>ident: 0<br>PC_1: -4.81769800<br>nFeature_RNA: 2112","TGAGCATGTCACAAGG-1<br>ident: 1<br>PC_1: -4.59093900<br>nFeature_RNA: 1596","TGAGCATGTTGTTTGG-1<br>ident: 1<br>PC_1: -4.62331992<br>nFeature_RNA: 1753","TGAGCCGAGTGGGTTG-1<br>ident: 0<br>PC_1: -4.56338418<br>nFeature_RNA: 2056","TGAGCCGGTGTATGGG-1<br>ident: 0<br>PC_1: -5.82938897<br>nFeature_RNA: 1423","TGAGGGACATTGCGGC-1<br>ident: 4<br>PC_1: 23.25151725<br>nFeature_RNA: 1989","TGAGGGATCAACACAC-1<br>ident: 1<br>PC_1: -5.08475478<br>nFeature_RNA: 1174","TGAGGGATCAATCTCT-1<br>ident: 3<br>PC_1: -1.10374334<br>nFeature_RNA: 2398","TGAGGGATCCTAAGTG-1<br>ident: 5<br>PC_1: 11.61271933<br>nFeature_RNA: 2342","TGATTTCTCGCTAGCG-1<br>ident: 4<br>PC_1: 21.91900671<br>nFeature_RNA: 1698","TGCACCTAGTTGAGTA-1<br>ident: 4<br>PC_1: 23.91711681<br>nFeature_RNA: 1930","TGCACCTGTTCCAACA-1<br>ident: 3<br>PC_1: -4.94441328<br>nFeature_RNA: 2385","TGCACCTTCGAACGGA-1<br>ident: 6<br>PC_1:  0.27081016<br>nFeature_RNA: 2334","TGCACCTTCTAGAGTC-1<br>ident: 5<br>PC_1:  5.66478438<br>nFeature_RNA: 1992","TGCCAAATCAAAGTAG-1<br>ident: 2<br>PC_1: -2.91960908<br>nFeature_RNA: 2132","TGCCCATTCACATAGC-1<br>ident: 3<br>PC_1: -4.22182580<br>nFeature_RNA: 2372","TGCCCATTCAGCTCGG-1<br>ident: 6<br>PC_1: -1.15552371<br>nFeature_RNA: 2075","TGCCCTAAGTTGAGAT-1<br>ident: 6<br>PC_1: -4.44361915<br>nFeature_RNA: 1472","TGCCCTAGTCGGATCC-1<br>ident: 1<br>PC_1: -4.63827493<br>nFeature_RNA: 1858","TGCCCTATCACAACGT-1<br>ident: 0<br>PC_1: -4.47290385<br>nFeature_RNA: 1238","TGCGCAGAGCCGATTT-1<br>ident: 1<br>PC_1: -5.15109098<br>nFeature_RNA: 1438","TGCGCAGCAGTCGTGC-1<br>ident: 1<br>PC_1: -4.69850033<br>nFeature_RNA: 1377","TGCGCAGGTCCGACGT-1<br>ident: 6<br>PC_1: -3.23960256<br>nFeature_RNA: 1943","TGCGCAGTCCAGAAGG-1<br>ident: 2<br>PC_1: -2.52280312<br>nFeature_RNA: 2450","TGCGGGTAGAGATGAG-1<br>ident: 4<br>PC_1: 20.72598210<br>nFeature_RNA: 1460","TGCGGGTAGTGGAGTC-1<br>ident: 4<br>PC_1: 18.11925170<br>nFeature_RNA: 1809","TGCGGGTAGTTACGGG-1<br>ident: 1<br>PC_1: -5.03199506<br>nFeature_RNA: 1705","TGCGGGTTCACCAGGC-1<br>ident: 7<br>PC_1:  5.68433154<br>nFeature_RNA: 1083","TGCGTGGCACGGACAA-1<br>ident: 7<br>PC_1:  8.05656038<br>nFeature_RNA: 1684","TGCGTGGGTCTTCAAG-1<br>ident: 3<br>PC_1: -4.29339416<br>nFeature_RNA: 1619","TGCGTGGTCTCAAGTG-1<br>ident: 1<br>PC_1: -4.89891017<br>nFeature_RNA: 1464","TGCTGCTTCTACCTGC-1<br>ident: 4<br>PC_1: 15.13857685<br>nFeature_RNA: 1257","TGGACGCAGGAATCGC-1<br>ident: 2<br>PC_1: -4.70644246<br>nFeature_RNA: 2272","TGGCCAGCAGCGTAAG-1<br>ident: 5<br>PC_1:  1.77546438<br>nFeature_RNA: 1354","TGGCCAGCAGGACCCT-1<br>ident: 1<br>PC_1: -5.62042148<br>nFeature_RNA: 1554","TGGCCAGTCGTCACGG-1<br>ident: 2<br>PC_1: -3.35373254<br>nFeature_RNA: 1930","TGGCGCATCACCTTAT-1<br>ident: 1<br>PC_1: -6.40541424<br>nFeature_RNA: 1410","TGGCGCATCACGCGGT-1<br>ident: 6<br>PC_1: -3.59945428<br>nFeature_RNA: 1182","TGGCGCATCTGGCGAC-1<br>ident: 4<br>PC_1: 17.99265150<br>nFeature_RNA: 1615","TGGCTGGGTCGGGTCT-1<br>ident: 6<br>PC_1: -4.25663083<br>nFeature_RNA: 1592","TGGCTGGGTGTTGGGA-1<br>ident: 7<br>PC_1:  5.44015050<br>nFeature_RNA:  939","TGGCTGGTCCTGCTTG-1<br>ident: 2<br>PC_1: -5.48415651<br>nFeature_RNA: 1550","TGGGAAGCAGTGGGAT-1<br>ident: 3<br>PC_1: -2.82509562<br>nFeature_RNA: 2111","TGGGCGTCAATGGATA-1<br>ident: 0<br>PC_1: -3.21019961<br>nFeature_RNA: 2322","TGGGCGTGTTTGACAC-1<br>ident: 1<br>PC_1: -5.46388173<br>nFeature_RNA: 1094","TGGGCGTTCATACGGT-1<br>ident: 4<br>PC_1: 17.52073291<br>nFeature_RNA: 1433","TGGGCGTTCTCTTGAT-1<br>ident: 0<br>PC_1: -3.34025405<br>nFeature_RNA: 2374","TGGTTAGAGAAGGACA-1<br>ident: 4<br>PC_1: 19.90544934<br>nFeature_RNA: 2223","TGTCCCAGTAGTGAAT-1<br>ident: 0<br>PC_1: -6.02071804<br>nFeature_RNA: 1275","TGTCCCAGTCTTCTCG-1<br>ident: 1<br>PC_1: -5.37195045<br>nFeature_RNA: 1304","TGTCCCATCAGCACAT-1<br>ident: 4<br>PC_1: 24.72882635<br>nFeature_RNA: 2453","TGTGTTTAGCTGTCTA-1<br>ident: 4<br>PC_1: 22.75882752<br>nFeature_RNA: 2101","TGTTCCGAGCTGCCCA-1<br>ident: 2<br>PC_1: -4.02931464<br>nFeature_RNA: 2036","TTAACTCGTAGAAAGG-1<br>ident: 4<br>PC_1: 16.77183192<br>nFeature_RNA: 1455","TTAACTCTCGTCTGAA-1<br>ident: 6<br>PC_1: -5.16272058<br>nFeature_RNA: 1694","TTAGGACGTAACGTTC-1<br>ident: 1<br>PC_1: -6.38653302<br>nFeature_RNA: 1087","TTAGGCAAGGCAGGTT-1<br>ident: 3<br>PC_1: -5.39693332<br>nFeature_RNA: 2034","TTAGGCAAGGTTACCT-1<br>ident: 4<br>PC_1: 22.72213192<br>nFeature_RNA: 1869","TTAGTTCTCGGTTCGG-1<br>ident: 0<br>PC_1: -3.98789455<br>nFeature_RNA: 2180","TTAGTTCTCTGGCGTG-1<br>ident: 1<br>PC_1: -5.90451581<br>nFeature_RNA: 1618","TTATGCTGTGCTGTAT-1<br>ident: 2<br>PC_1: -3.66776131<br>nFeature_RNA: 2469","TTATGCTTCATTGCGA-1<br>ident: 6<br>PC_1: -5.44390097<br>nFeature_RNA:  943","TTCCCAGCACTACAGT-1<br>ident: 0<br>PC_1: -4.39261920<br>nFeature_RNA: 2300","TTCGAAGAGGAGTAGA-1<br>ident: 2<br>PC_1: -4.63856078<br>nFeature_RNA: 2005","TTCGAAGAGTACGCGA-1<br>ident: 0<br>PC_1: -4.54116211<br>nFeature_RNA: 2151","TTCGAAGCATTAGCCA-1<br>ident: 2<br>PC_1: -3.24303406<br>nFeature_RNA: 2182","TTCTACAAGTTCGATC-1<br>ident: 0<br>PC_1: -3.81679846<br>nFeature_RNA: 2320","TTCTCAACATGATCCA-1<br>ident: 1<br>PC_1: -5.83066682<br>nFeature_RNA: 1308","TTCTCCTCAGTCTTCC-1<br>ident: 1<br>PC_1: -5.89048162<br>nFeature_RNA: 1591","TTCTTAGCACGAAGCA-1<br>ident: 6<br>PC_1: -4.82389027<br>nFeature_RNA: 1445","TTCTTAGCATGGGAAC-1<br>ident: 4<br>PC_1: 22.67209040<br>nFeature_RNA: 2106","TTGAACGAGCAATATG-1<br>ident: 0<br>PC_1: -6.22540386<br>nFeature_RNA: 2311","TTGAACGGTCCAAGTT-1<br>ident: 0<br>PC_1: -4.83115043<br>nFeature_RNA: 2198","TTGAACGGTGAAGGCT-1<br>ident: 0<br>PC_1: -5.04809153<br>nFeature_RNA: 1750","TTGAACGTCACGATGT-1<br>ident: 3<br>PC_1: -5.21829532<br>nFeature_RNA:  942","TTGACTTCAGATCCAT-1<br>ident: 3<br>PC_1: -4.81009801<br>nFeature_RNA: 2456","TTGCCGTCACGCCAGT-1<br>ident: 0<br>PC_1: -5.81335907<br>nFeature_RNA: 1920","TTGCGTCCAATAGAGT-1<br>ident: 1<br>PC_1: -5.68842131<br>nFeature_RNA: 1164","TTGGAACAGTCGAGTG-1<br>ident: 5<br>PC_1:  7.63623545<br>nFeature_RNA: 1827","TTGGCAAAGACTGTAA-1<br>ident: 4<br>PC_1: 19.05490657<br>nFeature_RNA: 1466","TTGGCAACATATACCG-1<br>ident: 3<br>PC_1: -4.57930986<br>nFeature_RNA: 2444","TTGGCAACATTGGTAC-1<br>ident: 0<br>PC_1: -4.69792679<br>nFeature_RNA: 2445","TTGTAGGAGAACAACT-1<br>ident: 5<br>PC_1:  6.73547980<br>nFeature_RNA: 1889","TTTACTGCAATCCAAC-1<br>ident: 7<br>PC_1:  3.46308202<br>nFeature_RNA: 1436","TTTATGCAGCAGGTCA-1<br>ident: 6<br>PC_1: -2.18056311<br>nFeature_RNA: 1871","TTTATGCTCCAGATCA-1<br>ident: 3<br>PC_1: -2.21301975<br>nFeature_RNA: 2035","TTTCCTCAGCTAACTC-1<br>ident: 1<br>PC_1: -4.34851490<br>nFeature_RNA: 1809","TTTCCTCCAGCCTGTG-1<br>ident: 2<br>PC_1: -4.06744074<br>nFeature_RNA: 1138","TTTCCTCCATCCTTGC-1<br>ident: 3<br>PC_1: -3.81115073<br>nFeature_RNA: 2077","TTTGCGCGTGCCTGGT-1<br>ident: 5<br>PC_1: 11.12365811<br>nFeature_RNA: 2446","TTTGGTTCAATGTAAG-1<br>ident: 4<br>PC_1: 16.39114194<br>nFeature_RNA: 1111","TTTGGTTCAGTCGATT-1<br>ident: 2<br>PC_1: -2.79245131<br>nFeature_RNA: 2089","TTTGGTTTCTGCCCTA-1<br>ident: 2<br>PC_1: -0.11109894<br>nFeature_RNA: 2159","TTTGTCAGTTGTGGAG-1<br>ident: 3<br>PC_1: -4.59395280<br>nFeature_RNA: 2317","TTTGTCATCCGCATCT-1<br>ident: 2<br>PC_1: -3.66450841<br>nFeature_RNA: 1548"],"type":"scatter","marker":{"color":["rgba(30,11,254,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(140,99,240,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,49,250,1)","rgba(150,110,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(110,68,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(152,113,237,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(84,46,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(133,90,242,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(146,105,239,1)","rgba(109,67,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(127,84,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(106,64,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(148,108,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(109,67,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(91,52,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(140,99,240,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(51,23,253,1)","rgba(126,83,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(82,45,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(152,113,237,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(95,55,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(130,87,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(63,31,252,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(106,64,247,1)","rgba(211,211,211,1)","rgba(105,64,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(99,58,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(86,48,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,48,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(95,55,249,1)","rgba(128,85,243,1)","rgba(100,59,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(131,89,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(134,92,242,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(0,0,255,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(128,85,243,1)","rgba(128,86,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,49,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(129,86,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(92,52,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(116,73,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(127,84,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(119,77,245,1)","rgba(211,211,211,1)","rgba(99,58,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(147,106,239,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(124,82,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(147,107,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(90,51,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(76,40,251,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(148,108,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(146,106,239,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(124,81,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)"],"line":{"color":["rgba(30,11,254,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(140,99,240,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,49,250,1)","rgba(150,110,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(110,68,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(152,113,237,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(84,46,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(133,90,242,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(146,105,239,1)","rgba(109,67,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(127,84,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(106,64,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(148,108,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(109,67,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(91,52,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(140,99,240,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(51,23,253,1)","rgba(126,83,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(82,45,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(152,113,237,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(95,55,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(130,87,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(63,31,252,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(106,64,247,1)","rgba(211,211,211,1)","rgba(105,64,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(99,58,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(86,48,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,48,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(95,55,249,1)","rgba(128,85,243,1)","rgba(100,59,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(131,89,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(134,92,242,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(0,0,255,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(128,85,243,1)","rgba(128,86,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,49,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(129,86,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(92,52,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(116,73,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(127,84,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(119,77,245,1)","rgba(211,211,211,1)","rgba(99,58,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(147,106,239,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(124,82,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(147,107,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(90,51,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(76,40,251,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(148,108,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(146,106,239,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(124,81,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)"]}},"textfont":{"color":["rgba(30,11,254,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(140,99,240,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,49,250,1)","rgba(150,110,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(110,68,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(152,113,237,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(84,46,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(133,90,242,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(146,105,239,1)","rgba(109,67,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(127,84,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(106,64,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(148,108,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(109,67,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(91,52,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(140,99,240,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(51,23,253,1)","rgba(126,83,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(82,45,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(152,113,237,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(95,55,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(130,87,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(63,31,252,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(106,64,247,1)","rgba(211,211,211,1)","rgba(105,64,247,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(99,58,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(86,48,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,48,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(95,55,249,1)","rgba(128,85,243,1)","rgba(100,59,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(131,89,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(134,92,242,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(0,0,255,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(128,85,243,1)","rgba(128,86,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(111,69,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(87,49,250,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(129,86,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(92,52,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(116,73,246,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(127,84,243,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(119,77,245,1)","rgba(211,211,211,1)","rgba(99,58,248,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(147,106,239,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(124,82,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(147,107,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(90,51,249,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(76,40,251,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(148,108,238,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(146,106,239,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(124,81,244,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(118,75,245,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)","rgba(211,211,211,1)"]},"xaxis":"x","yaxis":"y","frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb182"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb182-1"><a href="#cb182-1" aria-hidden="true" tabindex="-1"></a><span class="co">#___Heatmap_____</span></span>
<span id="cb182-2"><a href="#cb182-2" aria-hidden="true" tabindex="-1"></a>all.markers <span class="sc">%&gt;%</span></span>
<span id="cb182-3"><a href="#cb182-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cluster) <span class="sc">%&gt;%</span></span>
<span id="cb182-4"><a href="#cb182-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb182-5"><a href="#cb182-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">slice_head</span>(<span class="at">n =</span> <span class="dv">10</span>) <span class="sc">%&gt;%</span></span>
<span id="cb182-6"><a href="#cb182-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="ot">-&gt;</span> top10</span>
<span id="cb182-7"><a href="#cb182-7" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(seurat_obj, <span class="at">features =</span> top10<span class="sc">$</span>gene) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-9.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb183"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb183-1"><a href="#cb183-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DoHeatmap</span>(seurat_obj, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(seurat_obj)[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>], <span class="at">cells =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">500</span>, <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb183-2"><a href="#cb183-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">angle =</span> <span class="dv">90</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-25-10.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Automating visualization of top5 marker genes</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb184"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb184-1"><a href="#cb184-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)  <span class="co"># Required for visualization functions</span></span>
<span id="cb184-2"><a href="#cb184-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-3"><a href="#cb184-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate violin plots for top gene markers</span></span>
<span id="cb184-4"><a href="#cb184-4" aria-hidden="true" tabindex="-1"></a>plot_top_markers <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_obj, top_markers) {</span>
<span id="cb184-5"><a href="#cb184-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (gene <span class="cf">in</span> top5_markers) {</span>
<span id="cb184-6"><a href="#cb184-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> gene)</span>
<span id="cb184-7"><a href="#cb184-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb184-8"><a href="#cb184-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb184-9"><a href="#cb184-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-10"><a href="#cb184-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate feature plots for top gene markers</span></span>
<span id="cb184-11"><a href="#cb184-11" aria-hidden="true" tabindex="-1"></a>plot_feature_markers <span class="ot">&lt;-</span> <span class="cf">function</span>(seurat_obj, top_markers) {</span>
<span id="cb184-12"><a href="#cb184-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> top_markers)</span>
<span id="cb184-13"><a href="#cb184-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb184-14"><a href="#cb184-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-15"><a href="#cb184-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Example: Assuming you have already identified top gene markers and stored them in 'top5_markers'</span></span>
<span id="cb184-16"><a href="#cb184-16" aria-hidden="true" tabindex="-1"></a>top5_markers <span class="ot">&lt;-</span> <span class="fu">as.character</span>(top5_markers<span class="sc">$</span>gene)  <span class="co"># Convert to character vector if needed</span></span>
<span id="cb184-17"><a href="#cb184-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-18"><a href="#cb184-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate violin plots for top gene markers</span></span>
<span id="cb184-19"><a href="#cb184-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_top_markers</span>(seurat_obj, top5_markers)</span>
<span id="cb184-20"><a href="#cb184-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb184-21"><a href="#cb184-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate feature plots for top gene markers</span></span>
<span id="cb184-22"><a href="#cb184-22" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_feature_markers</span>(seurat_obj, top5_markers)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Cell cycle sorting</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb185"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb185-1"><a href="#cb185-1" aria-hidden="true" tabindex="-1"></a><span class="co">#*************************************************</span></span>
<span id="cb185-2"><a href="#cb185-2" aria-hidden="true" tabindex="-1"></a><span class="co">#*____________Cell-Cycle Scoring_________________</span></span>
<span id="cb185-3"><a href="#cb185-3" aria-hidden="true" tabindex="-1"></a>cc.genes.updated<span class="fl">.2019</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$s.genes
 [1] "MCM5"     "PCNA"     "TYMS"     "FEN1"     "MCM7"     "MCM4"    
 [7] "RRM1"     "UNG"      "GINS2"    "MCM6"     "CDCA7"    "DTL"     
[13] "PRIM1"    "UHRF1"    "CENPU"    "HELLS"    "RFC2"     "POLR1B"  
[19] "NASP"     "RAD51AP1" "GMNN"     "WDR76"    "SLBP"     "CCNE2"   
[25] "UBR7"     "POLD3"    "MSH2"     "ATAD2"    "RAD51"    "RRM2"    
[31] "CDC45"    "CDC6"     "EXO1"     "TIPIN"    "DSCC1"    "BLM"     
[37] "CASP8AP2" "USP1"     "CLSPN"    "POLA1"    "CHAF1B"   "MRPL36"  
[43] "E2F8"    

$g2m.genes
 [1] "HMGB2"   "CDK1"    "NUSAP1"  "UBE2C"   "BIRC5"   "TPX2"    "TOP2A"  
 [8] "NDC80"   "CKS2"    "NUF2"    "CKS1B"   "MKI67"   "TMPO"    "CENPF"  
[15] "TACC3"   "PIMREG"  "SMC4"    "CCNB2"   "CKAP2L"  "CKAP2"   "AURKB"  
[22] "BUB1"    "KIF11"   "ANP32E"  "TUBB4B"  "GTSE1"   "KIF20B"  "HJURP"  
[29] "CDCA3"   "JPT1"    "CDC20"   "TTK"     "CDC25C"  "KIF2C"   "RANGAP1"
[36] "NCAPD2"  "DLGAP5"  "CDCA2"   "CDCA8"   "ECT2"    "KIF23"   "HMMR"   
[43] "AURKA"   "PSRC1"   "ANLN"    "LBR"     "CKAP5"   "CENPE"   "CTCF"   
[50] "NEK2"    "G2E3"    "GAS2L3"  "CBX5"    "CENPA"  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb187"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb187-1"><a href="#cb187-1" aria-hidden="true" tabindex="-1"></a>s.genes <span class="ot">&lt;-</span> cc.genes.updated<span class="fl">.2019</span><span class="sc">$</span>s.genes</span>
<span id="cb187-2"><a href="#cb187-2" aria-hidden="true" tabindex="-1"></a>g2m.genes <span class="ot">&lt;-</span> cc.genes.updated<span class="fl">.2019</span><span class="sc">$</span>g2m.genes</span>
<span id="cb187-3"><a href="#cb187-3" aria-hidden="true" tabindex="-1"></a><span class="co">#assign Cell Score</span></span>
<span id="cb187-4"><a href="#cb187-4" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">CellCycleScoring</span>(seurat_obj, <span class="at">s.features =</span> s.genes, <span class="at">g2m.features =</span> g2m.genes, <span class="at">set.ident =</span> <span class="cn">TRUE</span>,<span class="at">use.synonyms =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The following features are not present in the object: PIMREG, JPT1,
not searching for symbol synonyms</code></pre>
</div>
<div class="sourceCode cell-code" id="cb189"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb189-1"><a href="#cb189-1" aria-hidden="true" tabindex="-1"></a><span class="co"># view cell cycle scores and phase assignments</span></span>
<span id="cb189-2"><a href="#cb189-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(seurat_obj[[]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                   orig.ident nCount_RNA nFeature_RNA percent.mt percent.rb
AAGTCTGAGACCACGA-1     TISCH2   1703.077          985   2.248466   18.36875
ACTGCTCAGATGTGGC-1     TISCH2   1775.858         1117   2.154970   17.81111
AGCCTAACAGAGTGTG-1     TISCH2   2009.721         1255   1.544596   14.88541
ATCGAGTTCGGTTAAC-1     TISCH2   1747.531         1053   2.697471   17.23857
ATTCTACAGGCGACAT-1     TISCH2   2247.777         1453   1.786211   12.30224
CAAGGCCCAACCGCCA-1     TISCH2   2183.734         1334   1.113512   12.61882
                   nCount_SCT nFeature_SCT SCT_snn_res.0.5 seurat_clusters
AAGTCTGAGACCACGA-1       2540          994               7               7
ACTGCTCAGATGTGGC-1       2441         1120               7               7
AGCCTAACAGAGTGTG-1       2358         1252               7               7
ATCGAGTTCGGTTAAC-1       2643         1061               7               7
ATTCTACAGGCGACAT-1       2313         1448               7               7
CAAGGCCCAACCGCCA-1       2412         1334               7               7
                       S.Score   G2M.Score Phase old.ident
AAGTCTGAGACCACGA-1 -0.01920376 -0.07624254    G1         7
ACTGCTCAGATGTGGC-1 -0.03083518 -0.04594581    G1         7
AGCCTAACAGAGTGTG-1  0.01695347 -0.06056703     S         7
ATCGAGTTCGGTTAAC-1 -0.01559862  0.08454327   G2M         7
ATTCTACAGGCGACAT-1 -0.09172761  0.08540350   G2M         7
CAAGGCCCAACCGCCA-1 -0.09615458 -0.10367870    G1         7</code></pre>
</div>
<div class="sourceCode cell-code" id="cb191"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb191-1"><a href="#cb191-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(seurat_obj[[]]<span class="sc">$</span>Phase)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 G1 G2M   S 
429  83 297 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb193"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb193-1"><a href="#cb193-1" aria-hidden="true" tabindex="-1"></a><span class="co">#PLOT- Visualize the distribution of cell cycle markers across</span></span>
<span id="cb193-2"><a href="#cb193-2" aria-hidden="true" tabindex="-1"></a><span class="fu">RidgePlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"MCM5"</span>, <span class="st">"PCNA"</span>, <span class="st">"TYMS"</span>, <span class="st">"FEN1"</span>), <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 3.45e-06</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.081</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 3.4e-06</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 3.31e-06</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb198"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb198-1"><a href="#cb198-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Running a PCA on cell cycle genes reveals, unsurprisingly, that cells separate entirely by</span></span>
<span id="cb198-2"><a href="#cb198-2" aria-hidden="true" tabindex="-1"></a><span class="co"># phase</span></span>
<span id="cb198-3"><a href="#cb198-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(s.genes, g2m.genes))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in PrepDR5(object = object, features = features, layer = layer, : The
following features were not available: PIMREG, JPT1.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in irlba(A = t(x = object), nv = npcs, ...): You're computing too large
a percentage of total singular values, use a standard svd instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  CDC20, HMMR, CENPA, CCNB2, CENPF, ANLN, MKI67, GTSE1, AURKA, BIRC5 
       TOP2A, TPX2, CDCA3, DLGAP5, CENPE, KIF2C, NUSAP1, CDK1, TACC3, CDCA8 
       RRM2, KIF23, NDC80, UBE2C, CENPU, CKAP2, ECT2, NEK2, TTK, HJURP 
Negative:  TIPIN, POLR1B, POLA1, GMNN, CDCA2, MCM6, RFC2, UBR7, DTL, CDCA7 
       CHAF1B, FEN1, CDC6, CBX5, CASP8AP2, MSH2, GINS2, UNG, G2E3, E2F8 
       USP1, SLBP, MRPL36, MCM4, CTCF, WDR76, RRM1, NASP, DSCC1, BLM 
PC_ 2 
Positive:  CENPA, CCNB2, KIF23, GTSE1, CKAP2, TPX2, MKI67, NEK2, ANLN, ECT2 
       CDC20, CKAP5, AURKA, TACC3, CENPE, BUB1, BIRC5, CDCA3, NDC80, PSRC1 
       RANGAP1, MRPL36, CENPF, UBE2C, HMMR, CDK1, G2E3, POLR1B, CDC25C, HJURP 
Negative:  UHRF1, CLSPN, CENPU, KIF11, ATAD2, PCNA, POLD3, TYMS, CCNE2, USP1 
       CDC45, HELLS, MCM5, WDR76, MCM7, RAD51AP1, CBX5, RRM1, UNG, MCM4 
       SLBP, GINS2, CASP8AP2, CHAF1B, DSCC1, MCM6, KIF2C, TTK, RRM2, CDCA7 
PC_ 3 
Positive:  KIF11, CLSPN, DLGAP5, RRM2, CENPU, E2F8, WDR76, CDC45, NASP, CDCA8 
       UBE2C, HJURP, NDC80, CDCA3, BIRC5, CDK1, AURKB, TYMS, CENPE, DSCC1 
       BLM, AURKA, PCNA, EXO1, CKS1B, UBR7, SLBP, ANP32E, PRIM1, KIF23 
Negative:  CCNE2, TTK, UHRF1, ANLN, RAD51, CASP8AP2, LBR, RAD51AP1, MKI67, CTCF 
       GINS2, NUSAP1, NCAPD2, MCM7, MSH2, MCM6, CHAF1B, KIF20B, CKAP2, HELLS 
       CKAP5, KIF2C, PSRC1, CDC20, ECT2, CENPF, GMNN, POLD3, CDCA7, TPX2 
PC_ 4 
Positive:  PRIM1, ECT2, TPX2, CCNB2, TYMS, CDCA8, RAD51AP1, GTSE1, GINS2, CDC6 
       CENPA, CKS2, UNG, CDK1, FEN1, SLBP, UBR7, TMPO, HMGB2, PCNA 
       POLD3, DLGAP5, AURKA, WDR76, ANP32E, HELLS, TIPIN, HMMR, CCNE2, RANGAP1 
Negative:  CENPE, GAS2L3, CDCA3, RRM2, LBR, AURKB, MCM4, E2F8, TTK, CLSPN 
       MKI67, KIF2C, UHRF1, ATAD2, BLM, CENPF, KIF11, BUB1, NASP, CKAP2L 
       NUSAP1, HJURP, MCM5, UBE2C, PSRC1, MCM6, RFC2, CKAP2, CDCA7, TOP2A 
PC_ 5 
Positive:  UBE2C, CDK1, CKS1B, GINS2, POLD3, RAD51, KIF2C, DLGAP5, MCM7, CENPE 
       TUBB4B, FEN1, NUSAP1, NDC80, CCNE2, CASP8AP2, MKI67, PCNA, KIF20B, ECT2 
       RAD51AP1, NEK2, PRIM1, MRPL36, CTCF, CDCA3, HMMR, CENPF, AURKB, ATAD2 
Negative:  PSRC1, NUF2, NCAPD2, GTSE1, MSH2, CENPU, CDCA8, MCM4, MCM6, TACC3 
       SMC4, MCM5, UHRF1, ANP32E, CDCA7, RANGAP1, CKAP2L, CKAP2, CKAP5, CDC25C 
       CDC45, AURKA, CLSPN, RRM2, BUB1, POLR1B, UBR7, HJURP, CDC20, WDR76 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb202"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb202-1"><a href="#cb202-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb203"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb203-1"><a href="#cb203-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Regress out cell cycle scores</span></span>
<span id="cb203-2"><a href="#cb203-2" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">$</span>CC.Difference <span class="ot">&lt;-</span> seurat_obj<span class="sc">$</span>S.Score <span class="sc">-</span> seurat_obj<span class="sc">$</span>G2M.Score</span>
<span id="cb203-3"><a href="#cb203-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(seurat_obj, <span class="at">vars.to.regress =</span> <span class="st">"CC.Difference"</span>, <span class="at">features =</span> <span class="fu">rownames</span>(seurat_obj))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Regressing out CC.Difference</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Centering and scaling data matrix</code></pre>
</div>
<div class="sourceCode cell-code" id="cb206"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb206-1"><a href="#cb206-1" aria-hidden="true" tabindex="-1"></a><span class="co"># When running a PCA on cell cycle genes, actively proliferating cells remain distinct from G1</span></span>
<span id="cb206-2"><a href="#cb206-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Cells; however, within actively proliferating cells, G2M and S phase cells group together</span></span>
<span id="cb206-3"><a href="#cb206-3" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(s.genes, g2m.genes))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in PrepDR5(object = object, features = features, layer = layer, : The
following features were not available: PIMREG, JPT1.

Warning in PrepDR5(object = object, features = features, layer = layer, :
You're computing too large a percentage of total singular values, use a
standard svd instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>PC_ 1 
Positive:  POLA1, G2E3, POLR1B, CDCA2, CTCF, TUBB4B, CBX5, LBR, CKAP5, CDC25C 
       TIPIN, CKS2, RFC2, DTL, BLM, ANP32E, NCAPD2, GMNN, E2F8, AURKB 
       MSH2, CDCA7, CKAP2L, RANGAP1, UBR7, MCM6, KIF20B, EXO1, RAD51, PSRC1 
Negative:  CDC20, HMMR, ANLN, CENPA, CENPF, CCNB2, MKI67, TOP2A, GTSE1, AURKA 
       BIRC5, KIF2C, CENPU, UHRF1, DLGAP5, CDCA3, TPX2, RRM2, NUSAP1, CENPE 
       CDCA8, CDK1, TYMS, TACC3, KIF23, CLSPN, PCNA, RAD51AP1, HELLS, TTK 
PC_ 2 
Positive:  CENPA, CCNB2, GTSE1, TPX2, MRPL36, KIF23, AURKA, UBR7, CDCA8, CDC20 
       CDK1, ANLN, ECT2, TIPIN, HMMR, CDC6, TACC3, PRIM1, DLGAP5, NDC80 
       FEN1, RANGAP1, CDC25C, SLBP, NEK2, BIRC5, CKAP2, POLR1B, NUF2, HJURP 
Negative:  UHRF1, CLSPN, ATAD2, CBX5, NUSAP1, TTK, MCM4, KIF11, CCNE2, KIF2C 
       GAS2L3, TUBB4B, LBR, CENPE, CENPU, RRM2, MCM5, AURKB, SMC4, WDR76 
       NCAPD2, CKS1B, HELLS, MCM7, CDCA3, POLD3, CDCA7, USP1, CDC45, BLM 
PC_ 3 
Positive:  CCNE2, ANLN, TTK, MKI67, UHRF1, CTCF, RAD51, CASP8AP2, RAD51AP1, GTSE1 
       GINS2, LBR, CKAP2, NCAPD2, CHAF1B, ECT2, CDC20, PSRC1, CKAP5, MSH2 
       MCM6, MCM7, TPX2, POLD3, RANGAP1, CENPF, KIF20B, HMMR, CENPA, NUF2 
Negative:  DLGAP5, KIF11, CDCA3, UBE2C, CENPE, CLSPN, RRM2, BIRC5, NDC80, HJURP 
       E2F8, CENPU, AURKB, KIF23, CDCA8, CDK1, NASP, CDC45, WDR76, BLM 
       AURKA, CKS1B, NEK2, DSCC1, ANP32E, EXO1, GAS2L3, BUB1, PCNA, G2E3 
PC_ 4 
Positive:  CENPE, MKI67, CDCA3, GAS2L3, ANLN, CENPF, BUB1, CDC20, RRM2, E2F8 
       NASP, NEK2, AURKB, LBR, MRPL36, TTK, HJURP, CKAP2L, CKAP2, KIF2C 
       BLM, RFC2, TOP2A, CHAF1B, CASP8AP2, NDC80, KIF23, UBE2C, TIPIN, MCM4 
Negative:  CENPU, CDCA8, WDR76, CBX5, CKS2, RAD51AP1, HELLS, CDC45, PRIM1, TYMS 
       TMPO, HMGB2, DLGAP5, NUF2, ANP32E, POLD3, TPX2, UNG, MSH2, CKS1B 
       TUBB4B, CLSPN, NCAPD2, CCNE2, DSCC1, ECT2, AURKA, SMC4, EXO1, RRM1 
PC_ 5 
Positive:  DLGAP5, CDK1, KIF2C, UBE2C, CKS1B, RAD51, POLD3, HMMR, CCNE2, RAD51AP1 
       GINS2, NEK2, MCM7, KIF20B, ECT2, PRIM1, NUSAP1, HELLS, KIF23, BIRC5 
       FEN1, CASP8AP2, TUBB4B, TOP2A, MRPL36, MKI67, CTCF, LBR, NDC80, TTK 
Negative:  PSRC1, GTSE1, NUF2, NCAPD2, CDCA8, MCM4, TACC3, CENPU, MSH2, HJURP 
       RRM2, RANGAP1, CKAP2, CDC20, CLSPN, MCM6, SMC4, BUB1, CKAP2L, MCM5 
       E2F8, ANP32E, CDC25C, KIF11, ANLN, TYMS, UHRF1, BLM, NASP, GAS2L3 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb209"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb209-1"><a href="#cb209-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb210"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb210-1"><a href="#cb210-1" aria-hidden="true" tabindex="-1"></a><span class="co">#clusters defined by any of the technical differences</span></span>
<span id="cb210-2"><a href="#cb210-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj,<span class="at">features =</span> <span class="st">"percent.mt"</span>,<span class="at">label.size =</span> <span class="dv">4</span>,<span class="at">repel =</span> T,<span class="at">label =</span> T) <span class="sc">&amp;</span> </span>
<span id="cb210-3"><a href="#cb210-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb211"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb211-1"><a href="#cb211-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj,<span class="at">features =</span> <span class="st">"percent.mt"</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb212"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj,<span class="at">features =</span> <span class="st">"percent.rb"</span>,<span class="at">label.size =</span> <span class="dv">4</span>,<span class="at">repel =</span> T,<span class="at">label =</span> T) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb213"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb213-1"><a href="#cb213-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj,<span class="at">features =</span> <span class="st">"percent.rb"</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-7.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb214"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb214-1"><a href="#cb214-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj,<span class="at">features =</span> <span class="fu">c</span>(<span class="st">"nCount_RNA"</span>,<span class="st">"nFeature_RNA"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb214-2"><a href="#cb214-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-8.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb215"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb215-1"><a href="#cb215-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj,<span class="at">features =</span> <span class="fu">c</span>(<span class="st">"S.Score"</span>,<span class="st">"G2M.Score"</span>),<span class="at">label.size =</span> <span class="dv">4</span>,<span class="at">repel =</span> T,<span class="at">label =</span> T) <span class="sc">&amp;</span> </span>
<span id="cb215-2"><a href="#cb215-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-9.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb216"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb216-1"><a href="#cb216-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj,<span class="at">features =</span> <span class="fu">c</span>(<span class="st">"S.Score"</span>,<span class="st">"G2M.Score"</span>)) <span class="sc">&amp;</span> </span>
<span id="cb216-2"><a href="#cb216-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-27-10.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb217"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb217-1"><a href="#cb217-1" aria-hidden="true" tabindex="-1"></a><span class="co">#SCTransform as an alternative as Single SCTransform command replaces NormalizeData, ScaleData, and FindVariableFeatures.</span></span>
<span id="cb217-2"><a href="#cb217-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb217-3"><a href="#cb217-3" aria-hidden="true" tabindex="-1"></a><span class="co">#seurat_obj &lt;- SCTransform(seurat_obj, method = "glmGamPoi", ncells = 8824,vars.to.regress = c("percent.mt","S.Score","G2M.Score"), verbose = F)</span></span>
<span id="cb217-4"><a href="#cb217-4" aria-hidden="true" tabindex="-1"></a><span class="co">#seurat_obj</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="cell-type-annotation-using-singler" class="level2">
<h2 class="anchored" data-anchor-id="cell-type-annotation-using-singler">Cell type annotation using SingleR</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb218"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb218-1"><a href="#cb218-1" aria-hidden="true" tabindex="-1"></a>monaco.ref <span class="ot">&lt;-</span> celldex<span class="sc">::</span><span class="fu">MonacoImmuneData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="sourceCode cell-code" id="cb223"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb223-1"><a href="#cb223-1" aria-hidden="true" tabindex="-1"></a>hpca.ref <span class="ot">&lt;-</span> celldex<span class="sc">::</span><span class="fu">HumanPrimaryCellAtlasData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation
loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
<div class="sourceCode cell-code" id="cb227"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb227-1"><a href="#cb227-1" aria-hidden="true" tabindex="-1"></a>dice.ref <span class="ot">&lt;-</span> celldex<span class="sc">::</span><span class="fu">DatabaseImmuneCellExpressionData</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation
loading from cache</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>see ?celldex and browseVignettes('celldex') for documentation</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>loading from cache</code></pre>
</div>
</div>
<p>Let’s convert our Seurat object to single cell experiment (SCE) for convenience. After this, using <code>SingleR</code> becomes very easy:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb231"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb231-1"><a href="#cb231-1" aria-hidden="true" tabindex="-1"></a>sce <span class="ot">&lt;-</span> <span class="fu">as.SingleCellExperiment</span>(<span class="fu">DietSeurat</span>(seurat_obj))</span>
<span id="cb231-2"><a href="#cb231-2" aria-hidden="true" tabindex="-1"></a>sce</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: SingleCellExperiment 
dim: 16041 809 
metadata(0):
assays(3): counts logcounts scaledata
rownames(16041): FO538757.2 AP006222.2 ... AL354822.1 AC004556.1
rowData names(0):
colnames(809): AAGTCTGAGACCACGA-1 ACTGCTCAGATGTGGC-1 ...
  TTGTAGGAGAACAACT-1 TTTGCGCGTGCCTGGT-1
colData names(15): orig.ident nCount_RNA ... CC.Difference ident
reducedDimNames(0):
mainExpName: RNA
altExpNames(1): SCT</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb233"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb233-1"><a href="#cb233-1" aria-hidden="true" tabindex="-1"></a>monaco.main <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> sce,<span class="at">assay.type.test =</span> <span class="dv">1</span>,<span class="at">ref =</span> monaco.ref,<span class="at">labels =</span> monaco.ref<span class="sc">$</span>label.main)</span>
<span id="cb233-2"><a href="#cb233-2" aria-hidden="true" tabindex="-1"></a>monaco.fine <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> sce,<span class="at">assay.type.test =</span> <span class="dv">1</span>,<span class="at">ref =</span> monaco.ref,<span class="at">labels =</span> monaco.ref<span class="sc">$</span>label.fine)</span>
<span id="cb233-3"><a href="#cb233-3" aria-hidden="true" tabindex="-1"></a>hpca.main <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> sce,<span class="at">assay.type.test =</span> <span class="dv">1</span>,<span class="at">ref =</span> hpca.ref,<span class="at">labels =</span> hpca.ref<span class="sc">$</span>label.main)</span>
<span id="cb233-4"><a href="#cb233-4" aria-hidden="true" tabindex="-1"></a>hpca.fine <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> sce,<span class="at">assay.type.test =</span> <span class="dv">1</span>,<span class="at">ref =</span> hpca.ref,<span class="at">labels =</span> hpca.ref<span class="sc">$</span>label.fine)</span>
<span id="cb233-5"><a href="#cb233-5" aria-hidden="true" tabindex="-1"></a>dice.main <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> sce,<span class="at">assay.type.test =</span> <span class="dv">1</span>,<span class="at">ref =</span> dice.ref,<span class="at">labels =</span> dice.ref<span class="sc">$</span>label.main)</span>
<span id="cb233-6"><a href="#cb233-6" aria-hidden="true" tabindex="-1"></a>dice.fine <span class="ot">&lt;-</span> <span class="fu">SingleR</span>(<span class="at">test =</span> sce,<span class="at">assay.type.test =</span> <span class="dv">1</span>,<span class="at">ref =</span> dice.ref,<span class="at">labels =</span> dice.ref<span class="sc">$</span>label.fine)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb234"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb234-1"><a href="#cb234-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(monaco.main<span class="sc">$</span>pruned.labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
        B cells       Basophils    CD4+ T cells    CD8+ T cells Dendritic cells 
            108              15              43               4              77 
      Monocytes        NK cells     Progenitors         T cells 
             85               5             450              15 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb236"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb236-1"><a href="#cb236-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(hpca.main<span class="sc">$</span>pruned.labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
           Astrocyte                   DC Embryonic_stem_cells 
                   1                   17                    1 
   Endothelial_cells     Epithelial_cells          Fibroblasts 
                   9                  588                    8 
           iPS_cells           Macrophage             Monocyte 
                   6                   46                   36 
             Neurons              NK_cell  Smooth_muscle_cells 
                   3                    7                    5 
             T_cells    Tissue_stem_cells 
                  20                   48 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb238"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb238-1"><a href="#cb238-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(dice.main<span class="sc">$</span>pruned.labels)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
      B cells     Monocytes      NK cells T cells, CD4+ T cells, CD8+ 
          133           422            20           118           116 </code></pre>
</div>
</div>
<p>Add annotations to the metadata</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb240"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb240-1"><a href="#cb240-1" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>monaco.main <span class="ot">&lt;-</span> monaco.main<span class="sc">$</span>pruned.labels</span>
<span id="cb240-2"><a href="#cb240-2" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>monaco.fine <span class="ot">&lt;-</span> monaco.fine<span class="sc">$</span>pruned.labels</span>
<span id="cb240-3"><a href="#cb240-3" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>hpca.main   <span class="ot">&lt;-</span> hpca.main<span class="sc">$</span>pruned.labels</span>
<span id="cb240-4"><a href="#cb240-4" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>dice.main   <span class="ot">&lt;-</span> dice.main<span class="sc">$</span>pruned.labels</span>
<span id="cb240-5"><a href="#cb240-5" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>hpca.fine   <span class="ot">&lt;-</span> hpca.fine<span class="sc">$</span>pruned.labels</span>
<span id="cb240-6"><a href="#cb240-6" aria-hidden="true" tabindex="-1"></a>seurat_obj<span class="sc">@</span>meta.data<span class="sc">$</span>dice.fine   <span class="ot">&lt;-</span> dice.fine<span class="sc">$</span>pruned.labels</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Visulaize</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb241"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb241-1"><a href="#cb241-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(seurat_obj, <span class="at">value =</span> <span class="st">"monaco.fine"</span>)</span>
<span id="cb241-2"><a href="#cb241-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">label =</span> T , <span class="at">repel =</span> T, <span class="at">label.size =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 23 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-35-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb243"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb243-1"><a href="#cb243-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(seurat_obj, <span class="at">value =</span> <span class="st">"hpca.fine"</span>)</span>
<span id="cb243-2"><a href="#cb243-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">label =</span> T , <span class="at">repel =</span> T, <span class="at">label.size =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 51 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb245"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb245-1"><a href="#cb245-1" aria-hidden="true" tabindex="-1"></a>seurat_obj <span class="ot">&lt;-</span> <span class="fu">SetIdent</span>(seurat_obj, <span class="at">value =</span> <span class="st">"dice.fine"</span>)</span>
<span id="cb245-2"><a href="#cb245-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">label =</span> T , <span class="at">repel =</span> T, <span class="at">label.size =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 15 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb247"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb247-1"><a href="#cb247-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj,<span class="st">"CD38"</span>) <span class="sc">+</span> <span class="fu">scale_colour_gradientn</span>(<span class="at">colours =</span> <span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">11</span>, <span class="at">name =</span> <span class="st">"Spectral"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Scale for colour is already present.
Adding another scale for colour, which will replace the existing scale.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-38-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb249"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb249-1"><a href="#cb249-1" aria-hidden="true" tabindex="-1"></a><span class="co">#****************************************************************</span></span>
<span id="cb249-2"><a href="#cb249-2" aria-hidden="true" tabindex="-1"></a><span class="co">#*__________________VIZUALIZATION PLOTS________________________</span></span>
<span id="cb249-3"><a href="#cb249-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb249-4"><a href="#cb249-4" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Formal class 'Seurat' [package "SeuratObject"] with 13 slots
  ..@ assays      :List of 2
  .. ..$ RNA:Formal class 'Assay5' [package "SeuratObject"] with 8 slots
  .. .. .. ..@ layers    :List of 3
  .. .. .. .. ..$ counts    :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. .. ..@ i       : int [1:1501544] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. .. ..@ p       : int [1:810] 0 985 2102 3357 4410 5863 7197 8590 9957 11096 ...
  .. .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. ..@ x       : num [1:1501544] 1.29 1.29 1.29 3.21 1.29 ...
  .. .. .. .. .. .. ..@ factors : list()
  .. .. .. .. ..$ data      :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. .. ..@ i       : int [1:1501544] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. .. ..@ p       : int [1:810] 0 985 2102 3357 4410 5863 7197 8590 9957 11096 ...
  .. .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. .. ..$ : NULL
  .. .. .. .. .. .. ..@ x       : num [1:1501544] 2.15 2.15 2.15 2.99 2.15 ...
  .. .. .. .. .. .. ..@ factors : list()
  .. .. .. .. ..$ scale.data: num [1:16041, 1:809] -0.4318 -0.3428 -0.0345 -0.1359 -0.1997 ...
  .. .. .. ..@ cells     :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:809, 1:3] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. .. ..$ : chr [1:3] "counts" "data" "scale.data"
  .. .. .. .. .. ..$ dim     : int [1:2] 809 3
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. .. ..$ : chr [1:3] "counts" "data" "scale.data"
  .. .. .. ..@ features  :Formal class 'LogMap' [package "SeuratObject"] with 1 slot
  .. .. .. .. .. ..@ .Data: logi [1:16041, 1:3] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. .. ..$ : chr [1:3] "counts" "data" "scale.data"
  .. .. .. .. .. ..$ dim     : int [1:2] 16041 3
  .. .. .. .. .. ..$ dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr [1:3] "counts" "data" "scale.data"
  .. .. .. ..@ default   : int 1
  .. .. .. ..@ assay.orig: chr(0) 
  .. .. .. ..@ meta.data :'data.frame': 16041 obs. of  8 variables:
  .. .. .. .. ..$ vf_vst_counts_mean                 : num [1:16041] 0.142317 0.095976 0.000781 0.014704 0.032001 ...
  .. .. .. .. ..$ vf_vst_counts_variance             : num [1:16041] 0.116484 0.085235 0.000493 0.013091 0.028126 ...
  .. .. .. .. ..$ vf_vst_counts_variance.expected    : num [1:16041] 0.128831 0.088432 0.000594 0.01419 0.030607 ...
  .. .. .. .. ..$ vf_vst_counts_variance.standardized: num [1:16041] 0.904 0.964 0.831 0.923 0.919 ...
  .. .. .. .. ..$ vf_vst_counts_variable             : logi [1:16041] FALSE FALSE FALSE FALSE FALSE FALSE ...
  .. .. .. .. ..$ vf_vst_counts_rank                 : int [1:16041] NA NA NA NA NA NA NA 1162 262 226 ...
  .. .. .. .. ..$ var.features                       : chr [1:16041] NA NA NA NA ...
  .. .. .. .. ..$ var.features.rank                  : int [1:16041] NA NA NA NA NA NA NA 1162 262 226 ...
  .. .. .. ..@ misc      : Named list()
  .. .. .. ..@ key       : chr "rna_"
  .. ..$ SCT:Formal class 'SCTAssay' [package "Seurat"] with 9 slots
  .. .. .. ..@ SCTModel.list:List of 1
  .. .. .. .. ..$ counts:Formal class 'SCTModel' [package "Seurat"] with 7 slots
  .. .. .. .. .. .. ..@ feature.attributes:'data.frame':    16041 obs. of  12 variables:
  .. .. .. .. .. .. .. ..$ detection_rate       : num [1:16041] 0.25404 0.17965 0.00632 0.03544 0.05649 ...
  .. .. .. .. .. .. .. ..$ gmean                : num [1:16041] 0.1263 0.08541 0.00269 0.01573 0.0251 ...
  .. .. .. .. .. .. .. ..$ variance             : num [1:16041] 0.08595 0.06204 0.00182 0.01234 0.01952 ...
  .. .. .. .. .. .. .. ..$ residual_mean        : num [1:16041] 3.19e-03 -5.03e-03 -3.55e-19 1.31e-04 2.64e-03 ...
  .. .. .. .. .. .. .. ..$ residual_variance    : num [1:16041] 0.591 0.572 0.09 0.585 0.663 ...
  .. .. .. .. .. .. .. ..$ theta                : num [1:16041] Inf Inf Inf Inf Inf ...
  .. .. .. .. .. .. .. ..$ (Intercept)          : num [1:16041] -9.76 -10.14 -13.6 -11.82 -11.35 ...
  .. .. .. .. .. .. .. ..$ log_umi              : num [1:16041] 2.3 2.3 2.3 2.3 2.3 ...
  .. .. .. .. .. .. .. ..$ genes_log_gmean_step1: logi [1:16041] FALSE FALSE FALSE FALSE FALSE FALSE ...
  .. .. .. .. .. .. .. ..$ step1_theta          : num [1:16041] NA NA NA NA NA ...
  .. .. .. .. .. .. .. ..$ step1_(Intercept)    : num [1:16041] NA NA NA NA NA ...
  .. .. .. .. .. .. .. ..$ step1_log_umi        : num [1:16041] NA NA NA NA NA ...
  .. .. .. .. .. .. ..@ cell.attributes   :'data.frame':    809 obs. of  3 variables:
  .. .. .. .. .. .. .. ..$ umi        : num [1:809] 1703 1776 2010 1748 2248 ...
  .. .. .. .. .. .. .. ..$ log_umi    : num [1:809] 3.23 3.25 3.3 3.24 3.35 ...
  .. .. .. .. .. .. .. ..$ cells_step1: logi [1:809] TRUE TRUE TRUE TRUE TRUE TRUE ...
  .. .. .. .. .. .. ..@ clips             :List of 2
  .. .. .. .. .. .. .. ..$ vst: num [1:2] -53.4 53.4
  .. .. .. .. .. .. .. ..$ sct: num [1:2] -9.75 9.75
  .. .. .. .. .. .. ..@ umi.assay         : chr "RNA"
  .. .. .. .. .. .. ..@ model             : chr "y ~ log_umi"
  .. .. .. .. .. .. ..@ arguments         :List of 33
  .. .. .. .. .. .. .. ..$ glmGamPoi_check          : logi TRUE
  .. .. .. .. .. .. .. ..$ latent_var               : chr "log_umi"
  .. .. .. .. .. .. .. ..$ batch_var                : NULL
  .. .. .. .. .. .. .. ..$ latent_var_nonreg        : NULL
  .. .. .. .. .. .. .. ..$ n_genes                  : num 2000
  .. .. .. .. .. .. .. ..$ n_cells                  : num 2850
  .. .. .. .. .. .. .. ..$ method                   : chr "glmGamPoi_offset"
  .. .. .. .. .. .. .. ..$ do_regularize            : logi TRUE
  .. .. .. .. .. .. .. ..$ theta_regularization     : chr "od_factor"
  .. .. .. .. .. .. .. ..$ res_clip_range           : num [1:2] -53.4 53.4
  .. .. .. .. .. .. .. ..$ bin_size                 : num 500
  .. .. .. .. .. .. .. ..$ min_cells                : num 5
  .. .. .. .. .. .. .. ..$ residual_type            : chr "pearson"
  .. .. .. .. .. .. .. ..$ return_cell_attr         : logi TRUE
  .. .. .. .. .. .. .. ..$ return_gene_attr         : logi TRUE
  .. .. .. .. .. .. .. ..$ return_corrected_umi     : logi TRUE
  .. .. .. .. .. .. .. ..$ min_variance             : chr "umi_median"
  .. .. .. .. .. .. .. ..$ bw_adjust                : num 3
  .. .. .. .. .. .. .. ..$ gmean_eps                : num 1
  .. .. .. .. .. .. .. ..$ theta_estimation_fun     : chr "theta.ml"
  .. .. .. .. .. .. .. ..$ theta_given              : NULL
  .. .. .. .. .. .. .. ..$ exclude_poisson          : logi TRUE
  .. .. .. .. .. .. .. ..$ use_geometric_mean       : logi TRUE
  .. .. .. .. .. .. .. ..$ use_geometric_mean_offset: logi FALSE
  .. .. .. .. .. .. .. ..$ fix_intercept            : logi FALSE
  .. .. .. .. .. .. .. ..$ fix_slope                : logi FALSE
  .. .. .. .. .. .. .. ..$ scale_factor             : logi NA
  .. .. .. .. .. .. .. ..$ vst.flavor               : chr "v2"
  .. .. .. .. .. .. .. ..$ verbosity                : num 0
  .. .. .. .. .. .. .. ..$ verbose                  : NULL
  .. .. .. .. .. .. .. ..$ show_progress            : NULL
  .. .. .. .. .. .. .. ..$ set_min_var              : num 0.0202
  .. .. .. .. .. .. .. ..$ sct.clip.range           : num [1:2] -9.75 9.75
  .. .. .. .. .. .. ..@ median_umi        : num 2813
  .. .. .. ..@ counts       :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. ..@ i       : int [1:1497441] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. ..@ p       : int [1:810] 0 994 2114 3366 4427 5875 7209 8602 9967 11109 ...
  .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..@ x       : num [1:1497441] 2 2 2 5 2 2 2 2 6 2 ...
  .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ data         :Formal class 'dgCMatrix' [package "Matrix"] with 6 slots
  .. .. .. .. .. ..@ i       : int [1:1497441] 26 27 58 81 122 212 220 230 238 256 ...
  .. .. .. .. .. ..@ p       : int [1:810] 0 994 2114 3366 4427 5875 7209 8602 9967 11109 ...
  .. .. .. .. .. ..@ Dim     : int [1:2] 16041 809
  .. .. .. .. .. ..@ Dimnames:List of 2
  .. .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..@ x       : num [1:1497441] 2.18 2.18 2.18 3.03 2.18 ...
  .. .. .. .. .. ..@ factors : list()
  .. .. .. ..@ scale.data   : num [1:16041, 1:809] -0.429 -0.339 0 -0.132 -0.195 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. ..@ assay.orig   : chr "RNA"
  .. .. .. ..@ var.features : chr [1:2000] "WAS" "LAT" "FMNL1" "SERPINA1" ...
  .. .. .. ..@ meta.features:'data.frame':  16041 obs. of  0 variables
  .. .. .. ..@ misc         : Named list()
  .. .. .. ..@ key          : chr "sct_"
  ..@ meta.data   :'data.frame':    809 obs. of  20 variables:
  .. ..$ orig.ident     : Factor w/ 1 level "TISCH2": 1 1 1 1 1 1 1 1 1 1 ...
  .. ..$ nCount_RNA     : num [1:809] 1703 1776 2010 1748 2248 ...
  .. ..$ nFeature_RNA   : int [1:809] 985 1117 1255 1053 1453 1334 1393 1367 1139 1398 ...
  .. ..$ percent.mt     : num [1:809] 2.25 2.15 1.54 2.7 1.79 ...
  .. ..$ percent.rb     : num [1:809] 18.4 17.8 14.9 17.2 12.3 ...
  .. ..$ nCount_SCT     : num [1:809] 2540 2441 2358 2643 2313 ...
  .. ..$ nFeature_SCT   : int [1:809] 994 1120 1252 1061 1448 1334 1393 1365 1142 1397 ...
  .. ..$ SCT_snn_res.0.5: Factor w/ 8 levels "0","1","2","3",..: 8 8 8 8 8 8 8 8 8 8 ...
  .. ..$ seurat_clusters: Factor w/ 8 levels "0","1","2","3",..: 8 8 8 8 8 8 8 8 8 8 ...
  .. ..$ S.Score        : num [1:809] -0.0192 -0.0308 0.017 -0.0156 -0.0917 ...
  .. ..$ G2M.Score      : num [1:809] -0.0762 -0.0459 -0.0606 0.0845 0.0854 ...
  .. ..$ Phase          : chr [1:809] "G1" "G1" "S" "G2M" ...
  .. ..$ old.ident      : Factor w/ 8 levels "0","1","2","3",..: 8 8 8 8 8 8 8 8 8 8 ...
  .. ..$ CC.Difference  : num [1:809] 0.057 0.0151 0.0775 -0.1001 -0.1771 ...
  .. ..$ monaco.main    : chr [1:809] "CD4+ T cells" "CD4+ T cells" "CD4+ T cells" "CD4+ T cells" ...
  .. ..$ monaco.fine    : chr [1:809] "Naive CD4 T cells" "Th17 cells" "Th2 cells" "Th1 cells" ...
  .. ..$ hpca.main      : chr [1:809] "T_cells" "T_cells" "T_cells" "T_cells" ...
  .. ..$ dice.main      : chr [1:809] "T cells, CD4+" "T cells, CD4+" "T cells, CD4+" "T cells, CD4+" ...
  .. ..$ hpca.fine      : chr [1:809] "T_cell:CD4+_Naive" "T_cell:CD4+_effector_memory" "T_cell:CD4+_central_memory" "T_cell:CD4+_central_memory" ...
  .. ..$ dice.fine      : chr [1:809] "T cells, CD4+, TFH" "T cells, CD4+, Th17" "T cells, CD4+, Th2" "T cells, CD4+, Th2" ...
  ..@ active.assay: chr "RNA"
  ..@ active.ident: Factor w/ 15 levels "T cells, CD4+, TFH",..: 1 2 3 3 4 4 4 5 1 4 ...
  .. ..- attr(*, "names")= chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  ..@ graphs      :List of 2
  .. ..$ SCT_nn :Formal class 'Graph' [package "SeuratObject"] with 7 slots
  .. .. .. ..@ assay.used: chr "SCT"
  .. .. .. ..@ i         : int [1:16180] 0 1 2 3 8 10 11 12 16 19 ...
  .. .. .. ..@ p         : int [1:810] 0 15 33 62 78 92 121 130 154 176 ...
  .. .. .. ..@ Dim       : int [1:2] 809 809
  .. .. .. ..@ Dimnames  :List of 2
  .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. ..@ x         : num [1:16180] 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. ..@ factors   : list()
  .. ..$ SCT_snn:Formal class 'Graph' [package "SeuratObject"] with 7 slots
  .. .. .. ..@ assay.used: chr "SCT"
  .. .. .. ..@ i         : int [1:58241] 0 1 2 3 4 5 6 7 8 9 ...
  .. .. .. ..@ p         : int [1:810] 0 29 58 87 116 145 174 203 232 261 ...
  .. .. .. ..@ Dim       : int [1:2] 809 809
  .. .. .. ..@ Dimnames  :List of 2
  .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. ..@ x         : num [1:58241] 1 0.818 0.818 0.818 0.379 ...
  .. .. .. ..@ factors   : list()
  ..@ neighbors   : list()
  ..@ reductions  :List of 3
  .. ..$ pca :Formal class 'DimReduc' [package "SeuratObject"] with 9 slots
  .. .. .. ..@ cell.embeddings           : num [1:809, 1:50] 0.862 1.569 0.257 1.733 4.193 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..$ : chr [1:50] "PC_1" "PC_2" "PC_3" "PC_4" ...
  .. .. .. ..@ feature.loadings          : num [1:95, 1:50] -0.0825 -0.1098 -0.1249 -0.0505 -0.0961 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:95] "MCM5" "PCNA" "TYMS" "FEN1" ...
  .. .. .. .. .. ..$ : chr [1:50] "PC_1" "PC_2" "PC_3" "PC_4" ...
  .. .. .. ..@ feature.loadings.projected: num[0 , 0 ] 
  .. .. .. ..@ assay.used                : chr "RNA"
  .. .. .. ..@ global                    : logi FALSE
  .. .. .. ..@ stdev                     : num [1:50] 2.94 1.6 1.54 1.48 1.41 ...
  .. .. .. ..@ jackstraw                 :Formal class 'JackStrawData' [package "SeuratObject"] with 4 slots
  .. .. .. .. .. ..@ empirical.p.values     : num[0 , 0 ] 
  .. .. .. .. .. ..@ fake.reduction.scores  : num[0 , 0 ] 
  .. .. .. .. .. ..@ empirical.p.values.full: num[0 , 0 ] 
  .. .. .. .. .. ..@ overall.p.values       : num[0 , 0 ] 
  .. .. .. ..@ misc                      :List of 1
  .. .. .. .. ..$ total.variance: num 85.1
  .. .. .. ..@ key                       : chr "PC_"
  .. ..$ umap:Formal class 'DimReduc' [package "SeuratObject"] with 9 slots
  .. .. .. ..@ cell.embeddings           : num [1:809, 1:2] 4.8 4.93 5.04 4.78 5.13 ...
  .. .. .. .. ..- attr(*, "scaled:center")= num [1:2] -4.23 1.6
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..$ : chr [1:2] "umap_1" "umap_2"
  .. .. .. ..@ feature.loadings          : num[0 , 0 ] 
  .. .. .. ..@ feature.loadings.projected: num[0 , 0 ] 
  .. .. .. ..@ assay.used                : chr "SCT"
  .. .. .. ..@ global                    : logi TRUE
  .. .. .. ..@ stdev                     : num(0) 
  .. .. .. ..@ jackstraw                 :Formal class 'JackStrawData' [package "SeuratObject"] with 4 slots
  .. .. .. .. .. ..@ empirical.p.values     : num[0 , 0 ] 
  .. .. .. .. .. ..@ fake.reduction.scores  : num[0 , 0 ] 
  .. .. .. .. .. ..@ empirical.p.values.full: num[0 , 0 ] 
  .. .. .. .. .. ..@ overall.p.values       : num[0 , 0 ] 
  .. .. .. ..@ misc                      : list()
  .. .. .. ..@ key                       : chr "umap_"
  .. ..$ tsne:Formal class 'DimReduc' [package "SeuratObject"] with 9 slots
  .. .. .. ..@ cell.embeddings           : num [1:809, 1:2] 8.67 8.63 7.92 8.53 6.32 ...
  .. .. .. .. ..- attr(*, "dimnames")=List of 2
  .. .. .. .. .. ..$ : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. .. ..$ : chr [1:2] "tSNE_1" "tSNE_2"
  .. .. .. ..@ feature.loadings          : num[0 , 0 ] 
  .. .. .. ..@ feature.loadings.projected: num[0 , 0 ] 
  .. .. .. ..@ assay.used                : chr "SCT"
  .. .. .. ..@ global                    : logi TRUE
  .. .. .. ..@ stdev                     : num(0) 
  .. .. .. ..@ jackstraw                 :Formal class 'JackStrawData' [package "SeuratObject"] with 4 slots
  .. .. .. .. .. ..@ empirical.p.values     : num[0 , 0 ] 
  .. .. .. .. .. ..@ fake.reduction.scores  : num[0 , 0 ] 
  .. .. .. .. .. ..@ empirical.p.values.full: num[0 , 0 ] 
  .. .. .. .. .. ..@ overall.p.values       : num[0 , 0 ] 
  .. .. .. ..@ misc                      : list()
  .. .. .. ..@ key                       : chr "tSNE_"
  ..@ images      : list()
  ..@ project.name: chr "TISCH2"
  ..@ misc        : list()
  ..@ version     :Classes 'package_version', 'numeric_version'  hidden list of 1
  .. ..$ : int [1:3] 5 0 1
  ..@ commands    :List of 13
  .. ..$ SCTransform.RNA         :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "SCTransform.RNA"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:48"
  .. .. .. ..@ assay.used : chr "RNA"
  .. .. .. ..@ call.string: chr [1:2] "SCTransform(seurat_obj, vars.to.regress = \"percent.mt\", " "    verbose = FALSE)"
  .. .. .. ..@ params     :List of 15
  .. .. .. .. ..$ assay                  : chr "RNA"
  .. .. .. .. ..$ new.assay.name         : chr "SCT"
  .. .. .. .. ..$ do.correct.umi         : logi TRUE
  .. .. .. .. ..$ ncells                 : num 5000
  .. .. .. .. ..$ variable.features.n    : num 3000
  .. .. .. .. ..$ variable.features.rv.th: num 1.3
  .. .. .. .. ..$ vars.to.regress        : chr "percent.mt"
  .. .. .. .. ..$ do.scale               : logi FALSE
  .. .. .. .. ..$ do.center              : logi TRUE
  .. .. .. .. ..$ clip.range             : num [1:2] -9.75 9.75
  .. .. .. .. ..$ vst.flavor             : chr "v2"
  .. .. .. .. ..$ conserve.memory        : logi FALSE
  .. .. .. .. ..$ return.only.var.genes  : logi TRUE
  .. .. .. .. ..$ seed.use               : num 1448145
  .. .. .. .. ..$ verbose                : logi FALSE
  .. ..$ NormalizeData.SCT       :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "NormalizeData.SCT"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:50"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr [1:2] "NormalizeData(seurat_obj, normalization.method = \"LogNormalize\", " "    scale.factor = 10000)"
  .. .. .. ..@ params     :List of 5
  .. .. .. .. ..$ assay               : chr "SCT"
  .. .. .. .. ..$ normalization.method: chr "LogNormalize"
  .. .. .. .. ..$ scale.factor        : num 10000
  .. .. .. .. ..$ margin              : num 1
  .. .. .. .. ..$ verbose             : logi TRUE
  .. ..$ FindVariableFeatures.SCT:Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "FindVariableFeatures.SCT"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:50"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr [1:2] "FindVariableFeatures(seurat_obj, selection.method = \"vst\", " "    nfeatures = 2000)"
  .. .. .. ..@ params     :List of 12
  .. .. .. .. ..$ assay              : chr "SCT"
  .. .. .. .. ..$ selection.method   : chr "vst"
  .. .. .. .. ..$ loess.span         : num 0.3
  .. .. .. .. ..$ clip.max           : chr "auto"
  .. .. .. .. ..$ mean.function      :function (mat, display_progress)  
  .. .. .. .. ..$ dispersion.function:function (mat, display_progress)  
  .. .. .. .. ..$ num.bin            : num 20
  .. .. .. .. ..$ binning.method     : chr "equal_width"
  .. .. .. .. ..$ nfeatures          : num 2000
  .. .. .. .. ..$ mean.cutoff        : num [1:2] 0.1 8
  .. .. .. .. ..$ dispersion.cutoff  : num [1:2] 1 Inf
  .. .. .. .. ..$ verbose            : logi TRUE
  .. ..$ ScaleData.SCT           :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "ScaleData.SCT"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:51"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr "ScaleData(seurat_obj, features = all.genes)"
  .. .. .. ..@ params     :List of 10
  .. .. .. .. ..$ features          : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. ..$ assay             : chr "SCT"
  .. .. .. .. ..$ model.use         : chr "linear"
  .. .. .. .. ..$ use.umi           : logi FALSE
  .. .. .. .. ..$ do.scale          : logi TRUE
  .. .. .. .. ..$ do.center         : logi TRUE
  .. .. .. .. ..$ scale.max         : num 10
  .. .. .. .. ..$ block.size        : num 1000
  .. .. .. .. ..$ min.cells.to.block: num 809
  .. .. .. .. ..$ verbose           : logi TRUE
  .. ..$ FindNeighbors.SCT.pca   :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "FindNeighbors.SCT.pca"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:54"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr "FindNeighbors(seurat_obj, dims = 1:15)"
  .. .. .. ..@ params     :List of 16
  .. .. .. .. ..$ reduction      : chr "pca"
  .. .. .. .. ..$ dims           : int [1:15] 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. .. ..$ assay          : chr "SCT"
  .. .. .. .. ..$ k.param        : num 20
  .. .. .. .. ..$ return.neighbor: logi FALSE
  .. .. .. .. ..$ compute.SNN    : logi TRUE
  .. .. .. .. ..$ prune.SNN      : num 0.0667
  .. .. .. .. ..$ nn.method      : chr "annoy"
  .. .. .. .. ..$ n.trees        : num 50
  .. .. .. .. ..$ annoy.metric   : chr "euclidean"
  .. .. .. .. ..$ nn.eps         : num 0
  .. .. .. .. ..$ verbose        : logi TRUE
  .. .. .. .. ..$ do.plot        : logi FALSE
  .. .. .. .. ..$ graph.name     : chr [1:2] "SCT_nn" "SCT_snn"
  .. .. .. .. ..$ l2.norm        : logi FALSE
  .. .. .. .. ..$ cache.index    : logi FALSE
  .. ..$ FindClusters            :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "FindClusters"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:54"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr "FindClusters(seurat_obj, resolution = 0.5)"
  .. .. .. ..@ params     :List of 11
  .. .. .. .. ..$ graph.name      : chr "SCT_snn"
  .. .. .. .. ..$ cluster.name    : chr "SCT_snn_res.0.5"
  .. .. .. .. ..$ modularity.fxn  : num 1
  .. .. .. .. ..$ resolution      : num 0.5
  .. .. .. .. ..$ method          : chr "matrix"
  .. .. .. .. ..$ algorithm       : num 1
  .. .. .. .. ..$ n.start         : num 10
  .. .. .. .. ..$ n.iter          : num 10
  .. .. .. .. ..$ random.seed     : num 0
  .. .. .. .. ..$ group.singletons: logi TRUE
  .. .. .. .. ..$ verbose         : logi TRUE
  .. ..$ RunPCA.SCT              :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "RunPCA.SCT"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:55"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr "RunPCA(seurat_obj, verbose = F)"
  .. .. .. ..@ params     :List of 10
  .. .. .. .. ..$ assay          : chr "SCT"
  .. .. .. .. ..$ npcs           : num 50
  .. .. .. .. ..$ rev.pca        : logi FALSE
  .. .. .. .. ..$ weight.by.var  : logi TRUE
  .. .. .. .. ..$ verbose        : logi FALSE
  .. .. .. .. ..$ ndims.print    : int [1:5] 1 2 3 4 5
  .. .. .. .. ..$ nfeatures.print: num 30
  .. .. .. .. ..$ reduction.name : chr "pca"
  .. .. .. .. ..$ reduction.key  : chr "PC_"
  .. .. .. .. ..$ seed.use       : num 42
  .. ..$ RunTSNE                 :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "RunTSNE"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:56:59"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr "RunTSNE(seurat_obj, dims = 1:15)"
  .. .. .. ..@ params     :List of 8
  .. .. .. .. ..$ reduction     : chr "pca"
  .. .. .. .. ..$ cells         : chr [1:809] "AAGTCTGAGACCACGA-1" "ACTGCTCAGATGTGGC-1" "AGCCTAACAGAGTGTG-1" "ATCGAGTTCGGTTAAC-1" ...
  .. .. .. .. ..$ dims          : int [1:15] 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. .. ..$ seed.use      : num 1
  .. .. .. .. ..$ tsne.method   : chr "Rtsne"
  .. .. .. .. ..$ dim.embed     : num 2
  .. .. .. .. ..$ reduction.name: chr "tsne"
  .. .. .. .. ..$ reduction.key : chr "tSNE_"
  .. ..$ NormalizeData.RNA       :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "NormalizeData.RNA"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:57:00"
  .. .. .. ..@ assay.used : chr "RNA"
  .. .. .. ..@ call.string: chr "NormalizeData(seurat_obj)"
  .. .. .. ..@ params     :List of 5
  .. .. .. .. ..$ assay               : chr "RNA"
  .. .. .. .. ..$ normalization.method: chr "LogNormalize"
  .. .. .. .. ..$ scale.factor        : num 10000
  .. .. .. .. ..$ margin              : num 1
  .. .. .. .. ..$ verbose             : logi TRUE
  .. ..$ FindVariableFeatures.RNA:Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "FindVariableFeatures.RNA"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:57:00"
  .. .. .. ..@ assay.used : chr "RNA"
  .. .. .. ..@ call.string: chr [1:2] "FindVariableFeatures(seurat_obj, selection.method = \"vst\", " "    nfeatures = 2000)"
  .. .. .. ..@ params     :List of 12
  .. .. .. .. ..$ assay              : chr "RNA"
  .. .. .. .. ..$ selection.method   : chr "vst"
  .. .. .. .. ..$ loess.span         : num 0.3
  .. .. .. .. ..$ clip.max           : chr "auto"
  .. .. .. .. ..$ mean.function      :function (mat, display_progress)  
  .. .. .. .. ..$ dispersion.function:function (mat, display_progress)  
  .. .. .. .. ..$ num.bin            : num 20
  .. .. .. .. ..$ binning.method     : chr "equal_width"
  .. .. .. .. ..$ nfeatures          : num 2000
  .. .. .. .. ..$ mean.cutoff        : num [1:2] 0.1 8
  .. .. .. .. ..$ dispersion.cutoff  : num [1:2] 1 Inf
  .. .. .. .. ..$ verbose            : logi TRUE
  .. ..$ RunUMAP.SCT.pca         :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "RunUMAP.SCT.pca"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:57:05"
  .. .. .. ..@ assay.used : chr "SCT"
  .. .. .. ..@ call.string: chr "RunUMAP(seurat_obj, dims = 1:15)"
  .. .. .. ..@ params     :List of 25
  .. .. .. .. ..$ dims                : int [1:15] 1 2 3 4 5 6 7 8 9 10 ...
  .. .. .. .. ..$ reduction           : chr "pca"
  .. .. .. .. ..$ assay               : chr "SCT"
  .. .. .. .. ..$ slot                : chr "data"
  .. .. .. .. ..$ umap.method         : chr "uwot"
  .. .. .. .. ..$ return.model        : logi FALSE
  .. .. .. .. ..$ n.neighbors         : int 30
  .. .. .. .. ..$ n.components        : int 2
  .. .. .. .. ..$ metric              : chr "cosine"
  .. .. .. .. ..$ learning.rate       : num 1
  .. .. .. .. ..$ min.dist            : num 0.3
  .. .. .. .. ..$ spread              : num 1
  .. .. .. .. ..$ set.op.mix.ratio    : num 1
  .. .. .. .. ..$ local.connectivity  : int 1
  .. .. .. .. ..$ repulsion.strength  : num 1
  .. .. .. .. ..$ negative.sample.rate: int 5
  .. .. .. .. ..$ uwot.sgd            : logi FALSE
  .. .. .. .. ..$ seed.use            : int 42
  .. .. .. .. ..$ angular.rp.forest   : logi FALSE
  .. .. .. .. ..$ densmap             : logi FALSE
  .. .. .. .. ..$ dens.lambda         : num 2
  .. .. .. .. ..$ dens.frac           : num 0.3
  .. .. .. .. ..$ dens.var.shift      : num 0.1
  .. .. .. .. ..$ verbose             : logi TRUE
  .. .. .. .. ..$ reduction.name      : chr "umap"
  .. ..$ ScaleData.RNA           :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "ScaleData.RNA"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:58:20"
  .. .. .. ..@ assay.used : chr "RNA"
  .. .. .. ..@ call.string: chr [1:2] "ScaleData(seurat_obj, vars.to.regress = \"CC.Difference\", " "    features = rownames(seurat_obj))"
  .. .. .. ..@ params     :List of 11
  .. .. .. .. ..$ features          : chr [1:16041] "FO538757.2" "AP006222.2" "RP4-669L17.10" "RP11-206L10.9" ...
  .. .. .. .. ..$ assay             : chr "RNA"
  .. .. .. .. ..$ vars.to.regress   : chr "CC.Difference"
  .. .. .. .. ..$ model.use         : chr "linear"
  .. .. .. .. ..$ use.umi           : logi FALSE
  .. .. .. .. ..$ do.scale          : logi TRUE
  .. .. .. .. ..$ do.center         : logi TRUE
  .. .. .. .. ..$ scale.max         : num 10
  .. .. .. .. ..$ block.size        : num 1000
  .. .. .. .. ..$ min.cells.to.block: num 3000
  .. .. .. .. ..$ verbose           : logi TRUE
  .. ..$ RunPCA.RNA              :Formal class 'SeuratCommand' [package "SeuratObject"] with 5 slots
  .. .. .. ..@ name       : chr "RunPCA.RNA"
  .. .. .. ..@ time.stamp : POSIXct[1:1], format: "2024-02-11 15:58:21"
  .. .. .. ..@ assay.used : chr "RNA"
  .. .. .. ..@ call.string: chr "RunPCA(seurat_obj, features = c(s.genes, g2m.genes))"
  .. .. .. ..@ params     :List of 11
  .. .. .. .. ..$ assay          : chr "RNA"
  .. .. .. .. ..$ features       : chr [1:97] "MCM5" "PCNA" "TYMS" "FEN1" ...
  .. .. .. .. ..$ npcs           : num 50
  .. .. .. .. ..$ rev.pca        : logi FALSE
  .. .. .. .. ..$ weight.by.var  : logi TRUE
  .. .. .. .. ..$ verbose        : logi TRUE
  .. .. .. .. ..$ ndims.print    : int [1:5] 1 2 3 4 5
  .. .. .. .. ..$ nfeatures.print: num 30
  .. .. .. .. ..$ reduction.name : chr "pca"
  .. .. .. .. ..$ reduction.key  : chr "PC_"
  .. .. .. .. ..$ seed.use       : num 42
  ..@ tools       : list()</code></pre>
</div>
<div class="sourceCode cell-code" id="cb251"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb251-1"><a href="#cb251-1" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> seurat_obj<span class="sc">@</span>commands<span class="sc">$</span>RunPCA.RNA<span class="sc">$</span>features</span>
<span id="cb251-2"><a href="#cb251-2" aria-hidden="true" tabindex="-1"></a>features <span class="ot">&lt;-</span> <span class="fu">VariableFeatures</span>(seurat_obj)</span>
<span id="cb251-3"><a href="#cb251-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb251-4"><a href="#cb251-4" aria-hidden="true" tabindex="-1"></a>metadata_columns <span class="ot">&lt;-</span> <span class="fu">colnames</span>(seurat_obj<span class="sc">@</span>meta.data)</span>
<span id="cb251-5"><a href="#cb251-5" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(metadata_columns)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "orig.ident"      "nCount_RNA"      "nFeature_RNA"    "percent.mt"     
 [5] "percent.rb"      "nCount_SCT"      "nFeature_SCT"    "SCT_snn_res.0.5"
 [9] "seurat_clusters" "S.Score"         "G2M.Score"       "Phase"          
[13] "old.ident"       "CC.Difference"   "monaco.main"     "monaco.fine"    
[17] "hpca.main"       "dice.main"       "hpca.fine"       "dice.fine"      </code></pre>
</div>
<div class="sourceCode cell-code" id="cb253"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb253-1"><a href="#cb253-1" aria-hidden="true" tabindex="-1"></a><span class="co">#_____DimPLot______</span></span>
<span id="cb253-2"><a href="#cb253-2" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj, <span class="at">label =</span> T , <span class="at">repel =</span> T, <span class="at">label.size =</span> <span class="dv">3</span>) <span class="sc">+</span> <span class="fu">NoLegend</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 15 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb255"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb255-1"><a href="#cb255-1" aria-hidden="true" tabindex="-1"></a><span class="fu">DimPlot</span>(seurat_obj)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-2.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb256"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb256-1"><a href="#cb256-1" aria-hidden="true" tabindex="-1"></a><span class="co">#______VlnPlot________</span></span>
<span id="cb256-2"><a href="#cb256-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"HCK"</span>, <span class="st">"SMOC2"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-3.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb257"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb257-1"><a href="#cb257-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="fu">c</span>(<span class="st">"HCK"</span>, <span class="st">"SMOC2"</span>), <span class="at">slot =</span> <span class="st">"counts"</span>, <span class="at">log =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-4.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb258"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb258-1"><a href="#cb258-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> features) <span class="co">#Visualize single cell expression distributions in each cluster</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-5.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb259"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb259-1"><a href="#cb259-1" aria-hidden="true" tabindex="-1"></a><span class="fu">VlnPlot</span>(seurat_obj, <span class="at">features =</span> <span class="st">"percent.mt"</span>, <span class="at">split.by =</span> <span class="st">"orig.ident"</span>)<span class="co">#Need to be fixed split by groups nnot Found</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>The default behaviour of split.by has changed.
Separate violin plots are now plotted side-by-side.
To restore the old behaviour of a single split violin,
set split.plot = TRUE.
      
This message will be shown once per session.</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-6.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb261"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb261-1"><a href="#cb261-1" aria-hidden="true" tabindex="-1"></a><span class="co">#_____featurePlot______</span></span>
<span id="cb261-2"><a href="#cb261-2" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> features[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-7.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb262"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb262-1"><a href="#cb262-1" aria-hidden="true" tabindex="-1"></a><span class="fu">FeaturePlot</span>(seurat_obj, <span class="at">features =</span> features[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">blend =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-8.png" class="img-fluid" width="672"></p>
</div>
<div class="sourceCode cell-code" id="cb263"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb263-1"><a href="#cb263-1" aria-hidden="true" tabindex="-1"></a><span class="co">#____RidgePlot_________</span></span>
<span id="cb263-2"><a href="#cb263-2" aria-hidden="true" tabindex="-1"></a><span class="fu">RidgePlot</span>(seurat_obj, <span class="at">features =</span> features[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.0977</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.0469</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.107</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.0317</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Picking joint bandwidth of 0.042</code></pre>
</div>
<div class="cell-output-display">
<p><img src="scrnaseq-analysis_files/figure-html/unnamed-chunk-39-9.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>